---
title: "ROHR_07_Gut_stats_Alpha_div"
author: "LLL"
date: "2022-11-18"
output: html_document
---

make sure to start w/ working directory set to correct filepath - in this case:
setwd("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed")
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```

Project summary

- Fish were collected from 2 sites in the Tropical Eastern Pacific: Las Perlas & Coiba. 
- Sampling during both upwelling & non-upwelling seasons (collections done 2021-22). 
- 12 target species, 10 of which are represented in this dataset. 
- Gut = 8 per species.
- Control swabs (swabs exposed to air in field) collected at each site and season
- Swabs were stored in liquid nitrogen for transport from field to lab, then at -80ÂºC
- DNA was extracted using ZymoBIOMICS 96 well DNA kit - 4 plates total (per sample type = skin & gut)
- DNA was amplified following a modified version of the Earth Microbiome Project 16S protocol using phased primers (1 set per plate)
- Negative PCR controls and extraction controls included in each plate
- Index PCR performed to attach unique barcodes, then all four plates (384 samples) pooled for sequencing on the Illumina MiSeq sequencing platform of the Smithsonian Tropical Research Institute's (STRI) Naos facilities in Panama (courtesy of Marta Vargas)
- Sequenced libraries were demultiplexed at STRI by Marta Vargas using the MiSeq Reporter Software
- These libraries were sent to me (Laura Lardinois) and trimmed with Cutadapt to remove primers before being read into R to start the DADA2 pipeline (following tutorial v 1.16) 
- Silva version 138.1 (mar 10, 2021 update) used for taxonomic assignments
- See "Gut Pre Processing RRR Fish 16S Sequences Summer 2022" document for pipeline and processing up to the creation of the initial phyloseq object

- This document follows the steps outlined in "OPEN & REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)", created by Sudarshan A. Shetty, Leo Lahti, and Gerben DA. Hermes, as well as other sources (linked below)

Shetty Sudarshan A, Lahti Leo, Hermes Gerben DA, & Hauke Smidt. (2020, April 11). Microbial bioinformatics introductory course material 2018 (Version v3.0). Zenodo. http://doi.org/10.5281/zenodo.1436630

```{r load required packages, echo=FALSE}
library(microbiome) # data analysis and visualisation
library(phyloseq) # also the basis of data object. Data analysis and visualisation
library(microbiomeutilities) # some utility tools 
library(RColorBrewer) # nice color options
library(ggpubr) # publication quality figures, based on ggplot2
library(DT) # interactive tables in html and markdown
library(data.table) # alternative to data.frame
library(dplyr) # data handling 

library(phylosmith) # package to create merged meta-data column

library(DECIPHER) # multiple-alignment of ASVs to construct phylogenetic tree
library(phangorn) # used to construct phylogenetic tree

library(picante) # calculating phylogenetic (alpha) diversity

library(microViz) # analysis and visualization of microbiome data (used to add sample_data column about tropic group)
library(forcats) #part of the tidyverse, the forcats library allows reordering and modifying factors in R (needed to cluster fish by trophic group)

library(vegan) # community ecology package: tools for diversity analysis, ordination methods, analysis of dissimilarities
```

Loading our phyloseq object = cleaned TEP fish skin microbiome ASVs, with metadata, tax table, and otu table (no tree yet)
```{r read phyloseq object, echo=FALSE}
ROHR_07_obj_clean = readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean.rds")
```

Before working with the full dataset, given that there are both distinct sampling regions (Coiba & Las Perlas) and sampling seasons (upwelling and non-upwelling), we want to be able to split the dataset into these four groups as needed. To do this, we need a single column that specifies both season and region, rather than 2 columns.
```{r add column to phyloseq metadata file that includes both region and season}

ROHR_07_obj_clean <- merge_treatments(ROHR_07_obj_clean, c("region", "season"))

# Save new metadata
saveRDS(ROHR_07_obj_clean, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean.rds")

#read new metadata
ROHR_07_obj_clean <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean.rds")

```

Quick look at cleaned dataset before removing rare ASVs
```{r examining phyloseq object - #reads, variables...}

#look at phyloseq object features (using microbiome package)
summarize_phyloseq(ROHR_07_obj_clean)

#min reads = 7, max reads = 65984, total reads = 7619622, average reads = 20482.85, median reads = 19831.5
##note that sparsity is high (0.9907) --> indicates data has many 0s --> common in microbiome data from sequencing-based techniques (will remove these in later steps)

#number of singletons = 53, % OTUs that are singletons = 0.396

```

We can use the coefficient of variation (C.V. = sd(x)/mean(x)) to measure the variability in our ASV abundance across samples (shows the amount of variability in relation to the population mean). The plot below shows CV (calculated for each ASV across samples) versus mean relative abundance. See that there are many ASVs with high CV, low mean; high heterogeneity/variance in ASV abundance - makes sense considering that we are pooling reads from different species, regions, and season together here. Note that there are many reads still unassigned at the Phylum level (Phylum = NA).
```{r looking at heterogeneity (variability) in ASV abundance across our dataset}

# the plot_taxa_cv will first convert the counts to relative abundances and then calculate the C.V.

CV.ROHR_07 <- plot_taxa_cv(ROHR_07_obj_clean, plot.type = "scatter")
CV.ROHR_07 + scale_x_log10()

```

Look at distribution of number of sequences kept after ASV picking using DADA2
```{r sequence distribution}
p_seqdepth.ROHR_07 <- plot_read_distribution(ROHR_07_obj_clean, "species", "density")
print(p_seqdepth.ROHR_07)
```
```{r Distribution of ASV counts}
# Make a data table with information on the ASVs
ROHR_07_df_taxa <- data.table(tax_table(ROHR_07_obj_clean),
                          ASVabundance = taxa_sums(ROHR_07_obj_clean),
                          ASV = taxa_names(ROHR_07_obj_clean))

ROHR_07_tax_plot <- ggplot(ROHR_07_df_taxa, aes(ASVabundance)) +
  geom_histogram() + ggtitle("Histogram of ASVs (unique sequence) counts (DADA2)") +
  theme_bw() + scale_x_log10() + ylab("Frequency of ASVs") + xlab("Abundance (raw counts)")

print(ROHR_07_tax_plot)

```

Can also look at the distribution of different microbial taxa among the ASVs, plotting average count abundance against taxa prevalence. Taxa that are low prevalence (i.e., rare) but represented by a high abundance of reads are likely to be real observations

```{r distribution of microbial taxa - abundance vs. prevalence}

######currently get error from this code
#taxa_prevalence_ROHR_07 <- plot_taxa_prevalence(ROHR_07_obj_clean, "Phylum")
#taxa_prevalence_ROHR_07
```

```{r examining number of reads, reads/ASV, singletons, doubletons }

#total number of reads in the dataset
reads_per_asv <- taxa_sums(ROHR_07_obj_clean)
print(sum(reads_per_asv)) #7619622

#total number of reads in across all samples 
reads_per_sample <- sample_sums(ROHR_07_obj_clean)
print(sum(reads_per_sample)) #7619622 (sum should be same as above = total number of reads)

#how many ASVs with fewer than 10 reads?
print(length(reads_per_asv[reads_per_asv < 10])) #4769 (out of 13389 ASVs --> around 35.6% of ASVs contain fewer than 10 reads)

#how many reads do these ASVs contain?
print(sum(reads_per_asv[reads_per_asv < 10])) #24840

#how many samples with fewer than 10 reads?
print(length(reads_per_sample[reads_per_sample < 10])) #1

#how many reads do these samples contain?
print(sum(reads_per_sample[reads_per_sample < 10])) #7

#how many samples with fewer than 1000 reads?
print(length(reads_per_sample[reads_per_sample < 1000])) #8 samples with fewer than 1000 reads (out of 372 = 2% of samples had fewer than 1,000 reads)

#how many reads do these samples contain?
print(sum(reads_per_sample[reads_per_sample < 1000])) #4882 

#how many singletons?
length(which(taxa_sums(ROHR_07_obj_clean) <= 1)) #53 

#how many doubletons?
length(which(taxa_sums(ROHR_07_obj_clean) == 2)) #551 doubletons

```
From the code above, we can see that our full swab dataset contains quite a few rare ASVs (35.6% have fewer than 10 reads). In addition, 8 of our samples have fewer than 1,000 reads. We can further visualize this by plotting out the distribution of read counts in our samples.

```{r visualize number of reads per sample}
#calculate total number of reads per sample and sort from smallest # to largest #
sort(sample_sums(ROHR_07_obj_clean))

#visualize distribution of read counts
hist(sample_sums(ROHR_07_obj_clean), main="Histogram: Read Counts", xlab="Total Reads", 
     border="blue", col="green", las=1, breaks=50)
```
Looking at the histogram above, we can see that the majority of our samples have between 0 and ~12,000 reads, with fewer samples having higher numbers of reads (up to 48,585). To remove potentially spurious ASVs, we will want to filter these reads. We will begin by removing samples that have fewer than 1,000 reads, then further filter ASVs that are not present in at least 2 samples (these are relatively low-stringency filters, given that we expect the skin microbiome to harbor rare taxa and want to avoid filtering these out of the dataset if possible). 

```{r removing samples w/ <1,000 reads}
# Remove samples with fewer than 1000 reads from phyloseq object
ROHR_07_obj_clean.1000 <- subset_samples(ROHR_07_obj_clean, sample_sums(ROHR_07_obj_clean) > 1000)

#revove ASVs not present in any sample (occurs after having removed samples)
ROHR_07_obj_clean.1000 <- prune_taxa(taxa_sums(ROHR_07_obj_clean.1000) > 0, ROHR_07_obj_clean.1000)
ROHR_07_obj_clean.1000 # down to 13384 taxa in 364 samples

#check that this worked & visualize distribution of read counts
hist(sample_sums(ROHR_07_obj_clean.1000), main="Histogram: Read Counts", xlab="Total Reads", 
     border="blue", col="green", las=1, breaks=50)

```

``` {r filter ASVs found in <2 samples}

####### Filter ASVs that are not present in at least 2 samples
condition <- function(x) { sum(x > 0) >= 2 }
taxaToKeep <- filter_taxa(ROHR_07_obj_clean.1000, condition)

ROHR_07_obj_clean_updated <- prune_taxa(taxaToKeep, ROHR_07_obj_clean.1000)
ROHR_07_obj_clean_updated #this brings us down to 3982 taxa in 364 samples

#Save phyloseq object with filtered taxa
saveRDS(ROHR_07_obj_clean_updated, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean_updated.rds")

#read in if not starting from top
ROHR_07_obj_clean_updated <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean_updated.rds")
```

```{r filter ASVs unassigned at phylum level}

##Removing ASVs unassigned (NA) at the phylum level
ROHR_07_obj_clean_updated_P <- subset_taxa(ROHR_07_obj_clean_updated, Phylum != "NA")
ROHR_07_obj_clean_updated_P #down to 3218 taxa in 364 samples  --> potentially add these back in as needed, depending on analyses to be run

#Save phyloseq object with filtered taxa
saveRDS(ROHR_07_obj_clean_updated_P, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean_updated_phyla.rds")

#read in if not starting from top
ROHR_07_obj_clean_updated_P <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_obj_clean_updated_phyla.rds")

```
######################################################
Split dataset into Gut (TEP) & Skin (Caribbean)

The ROHR_07 dataset contains both a set of gut samples from Coiba and Las Perlas (8/fish/region/season) that match (for the most part) the skin samples in ROHR_06, and a set of skin swabs from Caribbean sister species which will serve as a preliminary study of transisthmian microbiome comparisons. To look at this dataset further, we will begin by splitting it into Gut and Skin samples, with future comparisons across sample types and across oceans.

```{r Split dataset into Gut and Skin microbiome}
ROHR_07_Gut <- prune_samples(sample_data(ROHR_07_obj_clean_updated_P)$sample == "Gut", ROHR_07_obj_clean_updated_P) #have 312 gut samples

ROHR_07_Skin <- prune_samples(sample_data(ROHR_07_obj_clean_updated_P)$sample == "Swab", ROHR_07_obj_clean_updated_P) #have 52 skin samples

#save subsetted data
saveRDS(ROHR_07_Gut, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_gut_samples_only.rds")
saveRDS(ROHR_07_Skin, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_skin_swabs_only.rds")

#read subsetted data
ROHR_07_Gut <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_gut_samples_only.rds")

ROHR_07_Skin <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_skin_swabs_only.rds")
```


######################################################
ALPHA DIVERSITY
- Measures within individual taxa richness & evenness
- common metrics/indices = Shannon, Inverse Simpson, Simpson, Gini, Observed and Chao1 (do not take into account phylogeny of taxa)
- Faith's PD (phylogenetic diversity) --> uses phylogenetic distance

****"It is important to note that, alpha diversity indices are sensitive to noise that is inherent to application of polymerase chain reaction and the sequencing errors"

*** Need to consider sequencing depth - that is, how many taxa did we actually capture - for each sample, and, as needed, normalize samples to equal depth before running alpha diversity indices

Alpha diversity lecture: https://evolution.unibas.ch/walser/bacteria_community_analysis/2015-02-10_MBM_tutorial_combined.pdf

```{r look at number of reads}
summary(sample_sums(ROHR_07_obj_clean_updated_P)) #see that there is a large difference in number of reads (727 to 65240) --> although we filtered out samples with fewer than 1,000 reads earlier, the next steps removed a considerable number of ASVs, thus reducing the number of reads/sample 
```
```{r look at number of reads - Gut only}
summary(sample_sums(ROHR_07_Gut)) #see that there is a large difference in number of reads (766 to 65240) --> although we filtered out samples with fewer than 1,000 reads earlier, the next steps removed a considerable number of ASVs, thus reducing the number of reads/sample 
```
```{r look at number of reads - Skin only}
summary(sample_sums(ROHR_07_Skin)) #see that there is a large difference in number of reads (727 to 20834) --> although we filtered out samples with fewer than 1,000 reads earlier, the next steps removed a considerable number of ASVs, thus reducing the number of reads/sample --> generally fewer skin reads than gut reads
```

Plot rarefaction curve for the ASV data for the full dataset. This allows us to see if/how microbial richness has been captured by the sequencing effort. All samples appear to reach a plateau, indicating sufficient sampling depth to capture most diversity. 

```{r rarefaction curves for full (trimmed) dataset, echo=FALSE}
otu_tab_ROHR07 <- t(abundances(ROHR_07_obj_clean_updated_P))
rarcurve_ROHR07 <- vegan::rarecurve(otu_tab_ROHR07, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab_ROHR07), 
                                   col = "blue", cex = 0.6))
```
```{r rarefaction curves for Gut dataset, echo=FALSE}
otu_tab_ROHR07.gut <- t(abundances(ROHR_07_Gut))
rarcurve_ROHR07.gut <- vegan::rarecurve(otu_tab_ROHR07.gut, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab_ROHR07.gut), 
                                   col = "blue", cex = 0.6))
```

```{r rarefaction curves for Skin dataset, echo=FALSE}
otu_tab_ROHR07.skin <- t(abundances(ROHR_07_Skin))
rarcurve_ROHR07.skin <- vegan::rarecurve(otu_tab_ROHR07.skin, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab_ROHR07.skin), 
                                   col = "blue", cex = 0.6))
```

For the a-diversity analyses, we will begin by rarefying (normalizing) our data to at least 1,000 reads. This cutoff can be modified to exclude fewer/more samples with low sequencing depth

```{r normalizing data - Gut}
#check current reads/sample
reads_per_sample <- sample_sums(ROHR_07_Gut)

#how many samples with fewer than 1000 reads?
print(length(reads_per_sample[reads_per_sample < 1000])) #2 samples will be excluded under this criterion

set.seed(9242)  # This will help in reproducing the filtering and normalisation. 

ROHR07.rar.gut <- rarefy_even_depth(ROHR_07_Gut, sample.size = 1000) 
#1149OTUs were removed because they are no longer present in any sample after random subsampling

```
```{r normalizing data - Skin}
#check current reads/sample
reads_per_sample <- sample_sums(ROHR_07_Skin)

#how many samples with fewer than 1000 reads?
print(length(reads_per_sample[reads_per_sample < 1000])) #2 samples will be excluded under this criterion

set.seed(9242)  # This will help in reproducing the filtering and normalisation. 

ROHR07.rar.skin <- rarefy_even_depth(ROHR_07_Skin, sample.size = 1000) 
#1589OTUs were removed because they are no longer present in any sample after random subsampling

```

```{r save rarefied data as new phyloseq object}

####Gut####
# Save rarefied data
saveRDS(ROHR07.rar.gut, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Gut_rarefied_1000.rds")

#read rarefied data
ROHR07.rar.gut <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Gut_rarefied_1000.rds")

#check how much data post-rarefaction
print(ROHR07.rar.gut) # 2069 taxa, 310 samples

####Skin####
# Save rarefied data
saveRDS(ROHR07.rar.skin, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Skin_rarefied_1000.rds")

#read rarefied data
ROHR07.rar.skin <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Skin_rarefied_1000.rds")

#check how much data post-rarefaction
print(ROHR07.rar.skin) # 1629 taxa, 50 samples
```

```{r check data post-rarefaction}
 # quick check for sampling depth

barplot(sample_sums(ROHR07.rar.gut), las =2) #all samples rarefied to 1,000 reads
barplot(sample_sums(ROHR07.rar.skin), las =2) #all samples rarefied to 1,000 reads
```

Alpha diversity - non-phylogenetic diversities

- Alpha diversity metrics/indices give us information about the variation of microbes within a single sample

The "alpha" function within the microbiome package gives "global ecosystem state variables", including richness, evenness, and diversity. 
```{r Gut: calculate a-div - observed richness, other forms of richness, diversity, evenness, dominance, rarity...}
ROHR07.div.gut <- alpha(ROHR07.rar.gut, index = "all")

datatable(ROHR07.div.gut)
```
```{r Gut plot a-div data (one option) - shannon diversity}
# get the metadata out as separate object
ROHR07.meta.gut <- meta(ROHR07.rar.gut)

# Add the rownames as a new column for easy integration later.
ROHR07.meta.gut$sam_name <- rownames(ROHR07.meta.gut)

# Add the rownames to diversity table
ROHR07.div.gut$sam_name <- rownames(ROHR07.div.gut)

# merge these two data frames into one
div.df.gut <- merge(ROHR07.div.gut,ROHR07.meta.gut, by = "sam_name")

# check the tables
colnames(div.df.gut)

#reorder samples so that fish are grouped by diet
div.df.gut <- div.df.gut %>%
  mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Cephalopholis colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

# Now use this data frame to plot 
ROHR07.shannon.gut <- ggboxplot(div.df.gut, 
               x = "species", 
               y = "diversity_shannon",
              fill = "species", 
              palette = "rickandmorty",
              title = "Shannon Diversity of TEP Fish Gut Microbiome",
              xlab = "Host species",
              ylab = "Diversity (Shannon)")

ROHR07.shannon.gut <- ROHR07.shannon.gut + rotate_x_text()

print(ROHR07.shannon.gut)

# facet by region & season
ROHR07.shannon.RS.gut <- ggboxplot(div.df.gut, 
               x = "species", 
               y = "diversity_shannon",
              fill = "species", 
              palette = "rickandmorty",
              title = "Shannon Diversity of TEP Fish Gut Microbiome",
              xlab = "Host species",
              ylab = "Diversity (Shannon)",
              facet.by = "region_season")

ROHR07.shannon.RS.gut <- ROHR07.shannon.RS.gut + rotate_x_text()

print(ROHR07.shannon.RS.gut)


```
Plotting a-diversity (in this case using Shannon Diversity) by host species, shows that diversity is high across all hosts (close to upper range, as Shannon Index goes from 0-5), with some variation in amount of within-sample diversity depending on the species (e.g., A. concolor, C. humeralis, and (to a lesser extent) C. panamensis appear to generally have lower alpha diversity than most other fish, regardless of sampling season or region). Faceting by season & region shows that, although some species (e.g. J. nigrirostris) appear to host differentially diverse microbiomes by season, most remain relatively consistent across seasons and regions, for the most part having scores ~2-3 (lower than skin a-div, which was around 4). These high scores make sense given that Shannon Diversity measures both the number of species (in this case, microbial taxa) and inequality between species abundances, both of which are high in microbial datasets (and can be further visualized with the stacked barplots of relative abundances in the b-diversity section below).

```{r Skin: calculate a-div - observed richness, other forms of richness, diversity, evenness, dominance, rarity...}
ROHR07.div.s <- alpha(ROHR07.rar.skin, index = "all")

datatable(ROHR07.div.s)
```
```{r Skin plot a-div data (one option) - shannon diversity}
# get the metadata out as seprate object
ROHR07.meta.s <- meta(ROHR07.rar.skin)

# Add the rownames as a new column for easy integration later.
ROHR07.meta.s$sam_name <- rownames(ROHR07.meta.s)

# Add the rownames to diversity table
ROHR07.div.s$sam_name <- rownames(ROHR07.div.s)

# merge these two data frames into one
div.df.s <- merge(ROHR07.div.s,ROHR07.meta.s, by = "sam_name")

# check the tables
colnames(div.df.s)

#reorder samples so that fish are grouped by diet - update for Caribbean species
div.df.s <- div.df.s %>%
  mutate(species = fct_relevel(species, 
           "Abudefduf taurus", "Microspathodon chrysurus", "Chaetodon capistratus", "Chaetodon ocellatus", "Chaetodon striatus", "Abudefduf saxatilis", "Anisotremus virginicus"))

# Now use this data frame to plot 
ROHR07.shannon.s <- ggboxplot(div.df.s, 
               x = "species", 
               y = "diversity_shannon",
              fill = "species", 
              palette = "rickandmorty",
              title = "Shannon Diversity of Caribbean Fish Skin Microbiome",
              xlab = "Host species",
              ylab = "Diversity (Shannon)")

ROHR07.shannon.s <- ROHR07.shannon.s + rotate_x_text()

print(ROHR07.shannon.s)


```
Unsurprisingly, the 3 chaetodon species cluster together and again have the lowest a-diversity of the 7 Caribbean species, with the highest a-div found in A. virginicus, A. taurus, and A. saxatilis



--> explanation + formulas & calculations of Shannon-Wiener & Simpson's Indices: https://www.webpages.uidaho.edu/veg_measure/Modules/Lessons/Module%209(Composition&Diversity)/Detailed_Calculations.htm

We can gain a bit more insight on the distribution of the alpha diversity metrics by using violin plots, instead of simple box plots, following code outlined in: https://rpubs.com/lgschaerer/515637

***note that in the "plot_richness" documentation, they specifically advise against working with trimmed, normalized count data to obtain meaningful results, but the data I'm currently working with HAS been trimmed and normalized --> "You must use untrimmed, non-normalized count data for meaningful results, as many of these estimates are highly dependent on the number of singletons. You can always trim the data later on if needed, just not before using this function." 

Currently have 2 versions of the plot, one with the filtered and rarefied data (ROHR07.rar) and one with the non-rarefied and filtered data (samples with fewer than 1,000 reads + ASVs found in only 2 samples + ASVs unassigned at phylum level removed - ROHR_07_obj_clean_updated_P). Pick the one that is most appropriate to accurately display alpha diversity. 
```{r Gut plotting alpha diversity (another option using observed + shannon & violin plots)}
#reorder samples so that fish are grouped by diet
ROHR07.rar.gut <- ROHR07.rar.gut %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Cephalopholis colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

###Assign each category a color, define sample types and create sample labels

#picked colors by trophic group, with herbivores as different shades of green (territorial = aqua, roving = bright green), carnivores brown, planktivore purple, invertivores blue, and omnivore golden yellow

sample_colors <- c("Abudefduf concolor" = "aquamarine4", "Microspathodon dorsalis" = "aquamarine", "Acanthurus xanthopterus" = "chartreuse3", "Prionurus laticlavius" = "darkgreen", "Cephalopholis panamensis" = "brown4", "Epinephelus labriformis" = "chocolate2", "Cephalopholis colonus" = "darkmagenta", "Johnrandallia nigrirostris" = "darkslategray1", "Chaetodon humeralis" = "deepskyblue2", "Abudefduf troschelii" = "darkgoldenrod1")

sample_types <- c("Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Cephalopholis panamensis", "Epinephelus labriformis", "Cephalopholis colonus", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Abudefduf troschelii")

sample_labels <- c("Abudefduf concolor" = "A. concolor", "Microspathodon dorsalis" = "M. dorsalis", "Acanthurus xanthopterus" = "A. xanthopterus", "Prionurus laticlavius" = "P. laticlavius", "Cephalopholis panamensis" = "C. panamensis", "Epinephelus labriformis" = "E. labriformis", "Cephalopholis colonus" = "C. colonus", "Johnrandallia nigrirostris" = "J. nigrirostris", "Chaetodon humeralis" = "C. humeralis", "Abudefduf troschelii" = "A.troschelii")

ROHR07.rar.gut %>%                                                              #phyloseq object (rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+                        #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  ggtitle("Alpha Diversity of TEP Fish Gut Microbiome (rarefied to 1,000 reads)") +                          #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

####repeat with non-rarefied (but filtered to only include samples with >1,000 reads and ASVs that appear in >2 samples) data

#reorder samples so that fish are grouped by diet
ROHR_07_Gut <- ROHR_07_Gut %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Cephalopholis colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

ROHR_07_Gut %>%                                                              #phyloseq object (non rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+   #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  ggtitle("Alpha Diversity TEP Fish Gut Microbiome (non-rarefied data, ASVs trimmed)") +                     #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

# facet by season & region (although it makes it harder to see the distribution as everything is condensed)
ROHR07.rar.gut %>%                                                              #phyloseq object (rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+   #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  facet_wrap("region_season") +
  ggtitle("Alpha Diversity (Observed) of TEP Fish Gut Microbiome (rarefied to 1,000 reads)") +                                       #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position
```
From these plots, we can see that a-div is generally high, with Observed Diversity (the count of unique OTUs in each sample - a richness measure) generally matching Shannon Diversity (richness + diversity/rel. abund). Observed Diversity is noticeably higher in P. laticlavius (with wider range of observed values)

```{r Skin plotting alpha diversity (another option using observed + shannon & violin plots)}
#reorder samples so that fish are grouped by diet
ROHR07.rar.skin <- ROHR07.rar.skin %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf taurus", "Microspathodon chrysurus", "Chaetodon capistratus", "Chaetodon ocellatus", "Chaetodon striatus", "Abudefduf saxatilis", "Anisotremus virginicus" ))

###Assign each category a color, define sample types and create sample labels

#picked colors by trophic group, with herbivores as different shades of green (territorial = aqua, roving = bright green), carnivores brown, planktivore purple, invertivores blue, and omnivore golden yellow

sample_colors <- c("Abudefduf taurus" = "aquamarine4", "Microspathodon chrysurus" = "aquamarine", "Chaetodon striatus" = "darkgreen", "Anisotremus virginicus" = "chocolate2", "Chaetodon capistratus" = "darkslategray1", "Chaetodon ocellatus" = "deepskyblue2", "Abudefduf saxatilis" = "darkgoldenrod1")

sample_types <- c("Abudefduf taurus", "Microspathodon chrysurus", "Chaetodon capistratus", "Chaetodon ocellatus", "Chaetodon striatus", "Abudefduf saxatilis", "Anisotremus virginicus")

sample_labels <- c("Abudefduf taurus" = "A. taurus", "Microspathodon chrysurus" = "M. chrysurus", "Chaetodon capistratus" = "C. capistratus", "Chaetodon ocellatus" = "C. ocellatus", "Chaetodon striatus" = "C. striatus", "Abudefduf saxatilis" = "A.saxatilis", "Anisotremus virginicus" = "A. virginicus")

ROHR07.rar.skin %>%                                                              #phyloseq object (rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+                        #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  ggtitle("Alpha Diversity of Caribbean Fish Skin Microbiome (rarefied to 1,000 reads)") +                          #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

####repeat with non-rarefied (but filtered to only include samples with >1,000 reads and ASVs that appear in >2 samples) data

#reorder samples so that fish are grouped by diet
ROHR_07_Skin <- ROHR_07_Skin %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf taurus", "Microspathodon chrysurus", "Chaetodon capistratus", "Chaetodon ocellatus", "Chaetodon striatus", "Abudefduf saxatilis", "Anisotremus virginicus" ))

ROHR_07_Skin %>%                                                              #phyloseq object (non rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+   #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  ggtitle("Alpha Diversity Caribbean Fish Skin Microbiome (non-rarefied data, ASVs trimmed)") +                     #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

```

https://mibwurrepo.github.io/Microbial-bioinformatics-introductory-course-Material-2018/alpha-diversities.html#equal-sample-sums

Composite plot with multiple alpha diversity indices, including Inverse Simpson, Gini-Simpson, Shannon, Fisher, and Coverage (although these can be modified to select from the full list contained in the div.df data frame)

- Inverse Simpson Index:
"the effective number of types that is obtained when the weighted arithmetic mean is used to quantify average proportional abundance of types in the dataset of interest" (rdocumentation.org)
In other words, the Inverse Simpson Index is the reciprocal of Simpson's Index (see below), where 1 = low diversity, and highest diversity would be equal to the maximum number of species in the sample
        --> Simpson's Index (D) = quantifies probability that two individuals randomly selected from a sample will belong                to the same species. 0 represents infinite diversity, 1 represents no diversity --> makes little intuitive                 sense, so we often use the Diversity Index (see below) instead...
        --> Simpson's Diversity Index (1 - D) = takes D, subtracts from 1 so that highest diversity = 1 --> at high values               (nearing 1 or 100%), probability that two individuals selected at random will belong to different species is
              high. 
http://www.countrysideinfo.co.uk/simpsons.htm

- Gini-Simpson Index - aka Gibbs-Martin or Blau index
        --> 1 - sum of the (proportion of a given species within a community) squared

- Shannon Diversity Index - aka Shannon-Wiener Index:
        --> Measures diversity of species in a community by taking the (negative of the) sum of the proportion of a given species within a community times the natural log of that same proportion. Higher index = more species present (1 species = 0)

- Fisher Diversity Index - aka Fisher's alpha:
"a diversity index, defined implicitly by the formula S=a*ln(1+n/a) where S is number of taxa, n is number of individuals and a is the Fisher's alpha." https://www.nhm.uio.no/english/research/resources/past/help/diversity.html
       --> parametric diversity index that assumes species abundance follows log distribution
       
- Diversity Coverage 
"The coverage index gives the number of groups needed to have a given proportion of the ecosystem occupied (by default 0.5 ie 50%)." https://microbiome.github.io/tutorials/Alphadiversity.html

```{r alpha diversity plots - multiple indices}
# convert phyloseq object into a long data format.

div.df.gut2 <- div.df.gut[, c("species", "diversity_inverse_simpson", "diversity_gini_simpson", "diversity_shannon", "diversity_fisher", "diversity_coverage")]

# Replace names of diversity metrics/indices

colnames(div.df.gut2) <- c("Species", "Inverse Simpson", "Gini-Simpson", "Shannon", "Fisher", "Coverage")

# check
colnames(div.df.gut2)

### Using Species as id variables
div_df_melt <- reshape2::melt(div.df.gut2)

head(div_df_melt)

# Now use this data frame to plot 
ROHR07_alpha_div <- ggboxplot(div_df_melt, x = "Species", y = "value",
              fill = "Species", 
              palette = "rickandmorty", 
              legend= "right",
              facet.by = "variable", 
              scales = "free")

ROHR07_alpha_div <- ROHR07_alpha_div + rotate_x_text() # we will remove the x axis labels

ROHR07_alpha_div <- ROHR07_alpha_div + rremove("x.text")
ROHR07_alpha_div
```

```{r significance of a-diversity comparisons, echo=FALSE}

lev <- levels(as.factor(div_df_melt$Species)) # get the variables ###with source code, get error that n < m #tried to fix by making Species as factor rather than as character

# make a pairwise list that we want to compare.
L.pairs <- combn(seq_along(lev), 2, simplify = FALSE, FUN = function(i) lev[i])

pval <- list(
  cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
  symbols = c("****", "***", "**", "*", "n.s")
)

ROHR07.alpha.comp <- ROHR07_alpha_div + stat_compare_means(
  comparisons = L.pairs,
  label = "p.signif",
  symnum.args = list(
    cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
    symbols = c("****", "***", "**", "*", "n.s")
  )
)


print(ROHR07.alpha.comp) ###get error message saying: "Warning: cannot compute exact p-value with ties" --> try to fix this (perhaps by reducing dataset and/or specifying a different method to compare the means? --> default is currently "wilcox.test") ###also not very informative given the visual clutter of plotting the significance of pairwise (uncorrected?) comparisons across 10 species --> better as a table and/or clustered into groups by region and/or season perhaps

```


```{r alpha diversity plots - Shannon w/ significance bars attached}
# convert phyloseq object into a long data format.

div.df.gutshannon <- div.df.gut[, c("species", "diversity_shannon")]

# Replace names of diversity metrics/indices

colnames(div.df.gutshannon) <- c("Species", "Shannon")

# check
colnames(div.df.gutshannon)

### Using Species as id variables
div_df_melt_shan <- reshape2::melt(div.df.gutshannon)

head(div_df_melt_shan)

# Now use this data frame to plot 
ROHR07_alpha_div_shannon <- ggboxplot(div_df_melt_shan, x = "Species", y = "value",
              fill = "Species", 
              palette = "rickandmorty", 
              legend= "right",
              facet.by = "variable", 
              scales = "free")

ROHR07_alpha_div_shannon <- ROHR07_alpha_div_shannon + rotate_x_text() # we will remove the x axis labels

ROHR07_alpha_div_shannon <- ROHR07_alpha_div_shannon + rremove("x.text")
ROHR07_alpha_div_shannon

###add significance levels
lev_S <- levels(as.factor(div_df_melt_shan$Species)) # get the variables ###with source code, get error that n < m #tried to fix by making Species as factor rather than as character

# make a pairwise list that we want to compare.
L.pairs_S <- combn(seq_along(lev_S), 2, simplify = FALSE, FUN = function(i) lev_S[i])

pval <- list(
  cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
  symbols = c("****", "***", "**", "*", "n.s")
)

ROHR07.alpha.comp.shan <- ROHR07_alpha_div_shannon + stat_compare_means(
  comparisons = L.pairs_S,
  label = "p.signif",
  symnum.args = list(
    cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
    symbols = c("****", "***", "**", "*", "n.s")
  )
)


print(ROHR07.alpha.comp.shan)

```

Phylogenetic Tree

To calculate phylogenetic diversity, we will need to create a phylogenetic tree with our microbiome data. We can use this site: Workflow for Microbiome Data Analysis: from raw reads to community analyses, by Benjamin J Callahan, Kris Sankaran, Julia A Fukuyama, Paul Joey McMurdie and Susan P Holmes to help with tree creation. (It also includes information on many other steps in the microbiome processing and data visualization pipeline) https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html#construct_phylogenetic_tree

Since the DADA2 method used to assign taxonomy (ASVs) to our 16S reads is reference free, we need to connect the ASVs to their phylogenies de novo. The multiple-alignment step will be done using the DECIPHER R package (3.16); Wright (2016): http://bioconductor.org/packages/release/bioc/html/DECIPHER.html

####Start with gut samples####

```{r Gut multiple alignment using DECIPHER package}
tax_silva_tax_G  = tax_table(ROHR_07_Gut)

seqs <- getSequences(tax_silva_tax_G)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```

Once multiple alignment has been done using DECIPHER, we can use the phangorn package to build a phylogenetic tree.
The code below first builds a neighbor-joining tree, then fits a GTR+G+I maximum likelihood tree ( Generalized time-reversible with Gamma rate variation) using the neighbor-joining tree. Note that the maximum likelihood tree step requires considerable computational power - best to run on cluster (ran on laptop local hard drive for 27 hrs). 

Helpful lecture on different tree types and how they're constructed: https://si.biostat.washington.edu/sites/default/files/modules/2018-SISMID-04.pdf

CRAN documentation for phangorn: https://cran.r-project.org/web/packages/phangorn/vignettes/Trees.html
***this is the most thorough documentation, with steps for distance based methods, parsimony, and maximum likelihood trees

The Molecular Ecologist - "Quick and Dirty Tree Building in R": https://www.molecularecologist.com/2016/02/26/quick-and-dirty-tree-building-in-r/


```{r build NJ tree, then GTR+G+I maximum likelihood tree}

phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) #Neighbor joining tree ### Note, tip order != sequence order
#plot(treeNJ, "unrooted", main="NJ") #plot unrooted Neighbor Joining tree - given the number of tips and the fact that each   tip is currently labeled with a sequence, this just gives us a big, unreadable block of black ink 
treeUPGMA  <- upgma(dm) #UPGMA (unweighted pair group method with arithmetic mean)

#check parsimony score --> the min number of changes needed to describe data for the tree: best tree = lower score
parsimony(treeUPGMA, phangAlign)
#32712 for UPGMA tree

parsimony(treeNJ, phangAlign)
#31827 for NJ tree (better - improve upon this one with maximum likelihood (ML) method below)

#prior to producing a distance matrix, best to test which model best fits data, using a likelihood ratio test
mt <- modelTest(phangAlign, model = c("GTR", "JC", "F81", "K80", "HKY", "SYM")) #test a subset of models to limit run time, GTR + G(4) likely best based on best models from prior (TEP skin microbiome) full set
print(mt)

#looking at the different models and sorting them from lowest (best) AIC to highest, we can see that the models with the lowest AIC scores are:
# Model Name         AIC        BIC
# GTR + G(4)      285164.5    312810.8
# TVM + G(4)      285164.5    312815.1
# SYM + G(4)      285164.5    313291.3

as.pml(mt, "BIC") #best model according to BIC --> AIC & BIC agree for top models

#model: GTR+G(4) 
#loglikelihood: -136140.3 
#unconstrained loglikelihood: -3146.976 
#Discrete gamma model
#Number of rate categories: 4 
#Shape parameter: 0.5292531 

#Rate matrix:
#         a         c         g        t
#a 0.000000 0.8622420 3.0415538 1.520776
#c 0.862242 0.0000000 0.6251062 3.752043
#g 3.041554 0.6251062 0.0000000 1.000000
#t 1.520776 3.7520434 1.0000000 0.000000

#Base frequencies:  
#        a         c         g         t 
#0.2548873 0.1853841 0.2735152 0.2862134 

#save model test results as plain text file (to avoid having to run again)
write.table(mt, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR07_distance_models.txt")

fit = pml(treeNJ, data=phangAlign) #pml object contains the data + tree + many parameters of the model, inc. likelihood
fit 
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
                    rearrangement = "stochastic", control = pml.control(trace = 0)) #rearrangement = "stochastic" helps escape local optima by performing stochastic rearrangements
fitGTR

#model: GTR+G(4)+I 
#loglikelihood: -136459.4 
#unconstrained loglikelihood: -3146.976 
#Proportion of invariant sites: 0.1080756 
#Discrete gamma model
#Number of rate categories: 4 
#Shape parameter: 0.510925 

#Rate matrix:
#          a         c         g        t
#a 0.0000000 0.8295277 3.4377983 1.493780
#c 0.8295277 0.0000000 0.6666211 3.670326
#g 3.4377983 0.6666211 0.0000000 1.000000
#t 1.4937796 3.6703258 1.0000000 0.000000

#Base frequencies:  
#        a         c         g         t 
#0.2526507 0.1884068 0.2520519 0.3068907 

AIC(fitGTR)
#285804.7
BIC(fitGTR)
#313455.3

#bootstrap to test how well edges of tree are supported
bs <- bootstrap.pml(fitGTR, bs=100, optNni=TRUE,
    control = pml.control(trace = 0)) #####have not run this yet - supposed to be computationally demanding - will wait until not running other analyses

cnet <- consensusNet(bs, p=0.2) ######have not run this yet
plot(cnet, show.edge.label=TRUE) ####have not run this yet

#alternative function to do this which hides a few steps (starting from :
# fitGTR <- pml_bb(treeNJ, model="GTR+G(4)+I", control = pml.control(trace = 0))
#fitGTR

```

This phylogenetic tree will be added to a phyloseq object with (1) a sample-by-sequence table, (2) sample metadata, and (3) sequence taxonomies (see code below)

```{r write & read phylogenetic tree (unrooted)}
tree_NJ_tree = phy_tree(fitGTR$tree)

#save tree (on its own) in Newick format
ape::write.tree(tree_NJ_tree, file='~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR07_Gut_phylo_tree.tre')

tree_NJ_tree <- read.tree('~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR07_Gut_phylo_tree.tre')

#merge phylogenetic tree with phyloseq object

ROHR_07_obj_phylo <- merge_phyloseq(ROHR_07_Gut, tree_NJ_tree)
ROHR_07_obj_phylo

#otu_table()   OTU Table:         [ 3218 taxa and 312 samples ]
#sample_data() Sample Data:       [ 312 samples by 45 sample variables ]
#tax_table()   Taxonomy Table:    [ 3218 taxa by 6 taxonomic ranks ]
#phy_tree()    Phylogenetic Tree: [ 3218 tips and 3216 internal nodes ]

saveRDS(ROHR_07_obj_phylo, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Gut_with_phylogenetic_tree.rds")

#read RDS file (if not re-running code from start)

ROHR_07_obj_phylo <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Gut_with_phylogenetic_tree.rds")

```

```{r Root phylogenetic tree}

tree.unrooted <- phy_tree(ROHR_07_obj_phylo)

# Function to root the tree
pick_new_outgroup <- function(tree.unrooted){
  require("magrittr")
  require("data.table")
  require("ape") # ape::Ntip
  # tablify parts of tree that we need.
  treeDT <-
    cbind(
      data.table(tree.unrooted$edge),
      data.table(length = tree.unrooted$edge.length)
    )[1:Ntip(tree.unrooted)] %>%
    cbind(data.table(id = tree.unrooted$tip.label))
  # Take the longest terminal branch as outgroup
  new.outgroup <- treeDT[which.max(length)]$id
  return(new.outgroup)
}

new.outgroup = pick_new_outgroup(tree.unrooted)
new.outgroup

#new outgroup - that is the new root to be added to our tree = 
#TACGTAGGTGGCGAGCGTTGTTCGGATTTATTGGGCGTAAAGGGTCTGTAGGAGGTTTATTAAAATTGGGGTGAAATCCCGGAGCTCAACTCCGGAACTGCCTTGATGACTGATAGACTAGAGTATTGGAGAGGTAAGCGGAATATCAGGTGTAGCGGTGGAATGCGTAGATATCTGATAGAACACCAAAAGCGTAGGCAGCTTACTGGACAATAACTGACTCTGAAAGACGAAAGCGTGGGGAGCAAACAGG

rootedTree = ape::root(tree.unrooted, outgroup=new.outgroup, resolve.root=TRUE)

#plot(rootedTree) #don't plot rooted tree given the number of ASVs - at this stage plotting will not give informative results

#check that tree is, indeed, rooted
is.rooted(rootedTree) #TRUE! Yay, finally!

#make new phyloseq object with ROOTED tree
ROHR_07_obj_phylo_root <- merge_phyloseq(ROHR_07_Gut, rootedTree)
ROHR_07_obj_phylo_root

#save rooted tree
saveRDS(ROHR_07_obj_phylo_root, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Gut_with_phylogenetic_tree_rooted.rds")

#read RDS file (if not re-running code from start)

ROHR_07_obj_phylo_root <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_Gut_with_phylogenetic_tree_rooted.rds")
```

###Repeat w/ Skin####

```{r Skin multiple alignment using DECIPHER package}
tax_silva_tax_S  = tax_table(ROHR_07_Skin)

seqs <- getSequences(tax_silva_tax_S)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignmentS <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```
Once multiple alignment has been done using DECIPHER, we can use the phangorn package to build a phylogenetic tree.
The code below first builds a neighbor-joining tree, then fits a GTR+G+I maximum likelihood tree ( Generalized time-reversible with Gamma rate variation) using the neighbor-joining tree. Note that the maximum likelihood tree step requires considerable computational power - best to run on cluster (ran on laptop local hard drive for 27 hrs). 

Helpful lecture on different tree types and how they're constructed: https://si.biostat.washington.edu/sites/default/files/modules/2018-SISMID-04.pdf

CRAN documentation for phangorn: https://cran.r-project.org/web/packages/phangorn/vignettes/Trees.html
***this is the most thorough documentation, with steps for distance based methods, parsimony, and maximum likelihood trees

The Molecular Ecologist - "Quick and Dirty Tree Building in R": https://www.molecularecologist.com/2016/02/26/quick-and-dirty-tree-building-in-r/

```{r Skin build NJ tree, then GTR+G+I maximum likelihood tree}
phangAlign <- phyDat(as(alignmentS, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) #Neighbor joining tree ### Note, tip order != sequence order
#plot(treeNJ, "unrooted", main="NJ") #plot unrooted Neighbor Joining tree - given the number of tips and the fact that each   tip is currently labeled with a sequence, this just gives us a big, unreadable block of black ink 
treeUPGMA  <- upgma(dm) #UPGMA (unweighted pair group method with arithmetic mean)

#check parsimony score --> the min number of changes needed to describe data for the tree: best tree = lower score
parsimony(treeUPGMA, phangAlign)
#32712 for UPGMA tree

parsimony(treeNJ, phangAlign)
#31827 for NJ tree (better - improve upon this one with maximum likelihood (ML) method below)


fitS = pml(treeNJ, data=phangAlign) #pml object contains the data + tree + many parameters of the model, inc. likelihood
fitS 
fitGTR_S <- update(fitS, k=4, inv=0.2)
fitGTR_S <- optim.pml(fitGTR_S, model="GTR", optInv=TRUE, optGamma=TRUE,
                    rearrangement = "stochastic", control = pml.control(trace = 0)) #rearrangement = "stochastic" helps escape local optima by performing stochastic rearrangements
fitGTR_S

#model: GTR+G(4)+I 
#loglikelihood: -136459.4 
#unconstrained loglikelihood: -3146.976 
#Proportion of invariant sites: 0.1080756 
#Discrete gamma model
#Number of rate categories: 4 
#Shape parameter: 0.510925 

#Rate matrix:
#          a         c         g        t
#a 0.0000000 0.8295277 3.4377983 1.493780
#c 0.8295277 0.0000000 0.6666211 3.670326
#g 3.4377983 0.6666211 0.0000000 1.000000
#t 1.4937796 3.6703258 1.0000000 0.000000

#Base frequencies:  
#        a         c         g         t 
#0.2526507 0.1884068 0.2520519 0.3068907 

AIC(fitGTR_S)
#285804.7
BIC(fitGTR_S)
#313455.3

#bootstrap to test how well edges of tree are supported
bs <- bootstrap.pml(fitGTR_S, bs=100, optNni=TRUE,
    control = pml.control(trace = 0)) #####have not run this yet - supposed to be computationally demanding - will wait until not running other analyses

cnet <- consensusNet(bs, p=0.2) ######have not run this yet
plot(cnet, show.edge.label=TRUE) ####have not run this yet

#alternative function to do this which hides a few steps (starting from :
# fitGTR <- pml_bb(treeNJ, model="GTR+G(4)+I", control = pml.control(trace = 0))
#fitGTR

```

This phylogenetic tree will be added to a phyloseq object with (1) a sample-by-sequence table, (2) sample metadata, and (3) sequence taxonomies (see code below)

```{r Skin - write & read phylogenetic tree (unrooted)}
tree_NJ_tree_S = phy_tree(fitGTR_S$tree)

#save tree (on its own) in Newick format
ape::write.tree(tree_NJ_tree_S, file='~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR07_Skin_phylo_tree.tre')

tree_NJ_tree <- read.tree('~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR07_Skin_phylo_tree.tre')

#merge phylogenetic tree with phyloseq object

ROHR_07_skin_phylo <- merge_phyloseq(ROHR_07_Skin, tree_NJ_tree_S)
ROHR_07_skin_phylo


saveRDS(ROHR_07_skin_phylo, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_skin_with_phylogenetic_tree.rds")

#read RDS file (if not re-running code from start)

ROHR_07_skin_phylo <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_skin_with_phylogenetic_tree.rds")

```

```{r Skin - Root phylogenetic tree}

tree.unrooted.S <- phy_tree(ROHR_07_skin_phylo)

# Function to root the tree
pick_new_outgroup <- function(tree.unrooted.S){
  require("magrittr")
  require("data.table")
  require("ape") # ape::Ntip
  # tablify parts of tree that we need.
  treeDT <-
    cbind(
      data.table(tree.unrooted.S$edge),
      data.table(length = tree.unrooted.S$edge.length)
    )[1:Ntip(tree.unrooted.S)] %>%
    cbind(data.table(id = tree.unrooted.S$tip.label))
  # Take the longest terminal branch as outgroup
  new.outgroup <- treeDT[which.max(length)]$id
  return(new.outgroup)
}

new.outgroup = pick_new_outgroup(tree.unrooted.S)
new.outgroup

#new outgroup - that is the new root to be added to our tree = 
#

rootedTree.S = ape::root(tree.unrooted.S, outgroup=new.outgroup, resolve.root=TRUE)

#plot(rootedTree) #don't plot rooted tree given the number of ASVs - at this stage plotting will not give informative results

#check that tree is, indeed, rooted
is.rooted(rootedTree.S) #TRUE! Yay, finally!

#root:
#TACGGAGGGTGCAAGCGTTGTTCGGAATTATTGGGCGTAAAGCGGGTGTAGGCGGCTTTGCAAGTCAAGTGTGAAATCCCAGGGCCCAACCCTGGAACTGCACTCGATACTGCATCGCTAGAGTCCCGGAGAGGATGGCGGAATTCCAGGTGTAGAGGTGAAATTCGTAGATATCTGGAGGAACACCAGTGGCGAAGGCGGCCATCTGGACGGTGACTGACGCTCAGACCCGAAAGCGTGGGGAGCAAACAGG

#make new phyloseq object with ROOTED tree
ROHR_07_skin_phylo_root <- merge_phyloseq(ROHR_07_Skin, rootedTree.S)
ROHR_07_skin_phylo_root

#save rooted tree
saveRDS(ROHR_07_skin_phylo_root, "~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_skin_with_phylogenetic_tree_rooted.rds")

#read RDS file (if not re-running code from start)

ROHR_07_skin_phylo_root <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_07_trimmed/ROHR_07_skin_with_phylogenetic_tree_rooted.rds")

```

