---
title: "Script n 2 ROHR 06 Swab stats Alpha div"
author: ""
date: "2023-02-01"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

--> Script N. 2 - Fish skin microbiome alpha diversity metrics & phylogenetic tree


Most recent updates: 
December 2, 2024
November 18, 2024 - LLL - splitting Hill # figures by gulf



Project summary

- Fish were collected from 2 sites in the Tropical Eastern Pacific: Las Perlas & Coiba. 
- Sampling during both upwelling & non-upwelling seasons (collections done 2021-22). 
- 10 target species
- Swabs = 9 per species.
- Control swabs (swabs exposed to air in field) collected at each site and season
- Swabs were stored in liquid nitrogen for transport from field to lab, then at -80ÂºC
- DNA was extracted using ZymoBIOMICS 96 well DNA kit - 4 plates total
- DNA was amplified following a modified version of the Earth Microbiome Project 16S protocol using phased primers (1 set per plate)
- Negative PCR controls and extraction controls included in each plate
- Index PCR performed to attach unique barcodes, then all four plates (384 samples) pooled for sequencing on the Illumina MiSeq sequencing platform of the Smithsonian Tropical Research Institute's (STRI) Naos facilities in Panama (courtesy of Marta Vargas)
- Sequenced libraries were demultiplexed at STRI by Marta Vargas using the MiSeq Reporter Software
- These libraries were trimmed with Cutadapt to remove primers before being read into R to start the DADA2 pipeline (following tutorial v 1.16) 
- Silva version 138.1 (mar 10, 2021 update) used for taxonomic assignments
- See "Skin Pre Processing RRR Fish 16S Version 1" document for pipeline and processing up to the creation of the initial phyloseq object

- This document follows the steps outlined in "OPEN & REPRODUCIBLE MICROBIOME DATA ANALYSIS SPRING SCHOOL 2018 v3.0 (Updated 11-Apr-2020)", created by Sudarshan A. Shetty, Leo Lahti, and Gerben DA. Hermes, as well as other sources (linked below)

Shetty Sudarshan A, Lahti Leo, Hermes Gerben DA, & Hauke Smidt. (2020, April 11). Microbial bioinformatics introductory course material 2018 (Version v3.0). Zenodo. http://doi.org/10.5281/zenodo.1436630

```{r load required packages, echo=FALSE}
library(microbiome) # data analysis and visualisation
library(phyloseq) # also the basis of data object. Data analysis and visualisation
library(microbiomeutilities) # some utility tools 
library(RColorBrewer) # nice color options
library(ggpubr) # publication quality figures, based on ggplot2
library(DT) # interactive tables in html and markdown
library(data.table) # alternative to data.frame
library(dplyr) # data handling 

library(phylosmith) # package to create merged meta-data column

library(DECIPHER) # multiple-alignment of ASVs to construct phylogenetic tree
library(phangorn) # used to construct phylogenetic tree

library(picante) # calculating phylogenetic (alpha) diversity

library(microViz) # analysis and visualization of microbiome data (used to add sample_data column about tropic group)
library(forcats) #part of the tidyverse, the forcats library allows reordering and modifying factors in R (needed to cluster fish by trophic group)

library(vegan) # community ecology package: tools for diversity analysis, ordination methods, analysis of dissimilarities
library(hilldiv) # Hill numbers package
library(tidyr) # Help create tidy data (store data well, clean)
library(mirlyn) # "Multiple Iterations of Rarefying for Library Normalization" - used to rarefy data (note that this also has a rarefaction curve function (rarecurve()) which masks the function of the same name in 'vegan')

library(flextable) #create table in word doc directly from R (if have issues with install, try: install.packages("flextable", type = "binary"))
library(officer) #create table in word doc directly from R

library(cowplot) #to join 2 plots together
library(grid) #mess with plot graphics and combined plot axes
```

### Reading in data

Loading our phyloseq object = cleaned TEP fish skin microbiome ASVs, with metadata, tax table, and otu table (no tree yet)
```{r read phyloseq object, echo=FALSE}
ROHR_06_obj_clean = readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean.rds")
```

Before working with the full dataset, given that there are both distinct sampling regions (Coiba & Las Perlas) and sampling seasons (upwelling and non-upwelling), we want to be able to split the dataset into these four groups as needed. To do this, we need a single column that specifies both season and region, rather than 2 columns.
```{r add column to phyloseq metadata file that includes both region and season}

ROHR_06_obj_clean <- merge_treatments(ROHR_06_obj_clean, c("region", "season"))

# Save new metadata
saveRDS(ROHR_06_obj_clean, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean.rds")

#read new metadata
ROHR_06_obj_clean <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean.rds")

```

#### Preliminary glance at our fish skin microbiome dataset

Quick look at cleaned dataset before removing rare ASVs
```{r examining phyloseq object}

#look at phyloseq object features (using microbiome package)
summarize_phyloseq(ROHR_06_obj_clean)

#min reads = 1, max reads = 48663, total reads = 4418389, average reads = 12307.5, median reads = 7967
##note that sparsity is high (0.9876) --> indicates data has many 0s --> common in microbiome data from sequencing-based techniques (will filter these in later steps)

#number of singletons = 29 (0.124% of OTUs are singletons)
```

We can use the coefficient of variation (C.V. = sd(x)/mean(x)) to measure the variability in our ASV abundance across samples (shows the amount of variability in relation to the population mean). The plot below shows CV (calculated for each ASV across samples) versus mean relative abundance. See that there are many ASVs with high CV, low mean; high heterogeneity/variance in ASV abundance - which makes sense considering that we are pooling microbial reads from different species, regions, and season together here. Note that there are many reads still unassigned at the Phylum level (Phylum = NA, in gray).
```{r looking at heterogeneity (variability) in ASV abundance across our dataset}

# the plot_taxa_cv will first convert the counts to relative abundances and then calculate the C.V.

CV.ROHR_06 <- plot_taxa_cv(ROHR_06_obj_clean, plot.type = "scatter")
CV.ROHR_06 + scale_x_log10()

```

Look at distribution of number of sequences kept after ASV picking using DADA2
```{r distribution of reads/sample across fish species}
p_seqdepth.ROHR_06 <- plot_read_distribution(ROHR_06_obj_clean, "species", "density")
print(p_seqdepth.ROHR_06)
```
```{r distribution of ASV counts}
# Make a data table with information on the ASVs
ROHR_06_df_taxa <- data.table(tax_table(ROHR_06_obj_clean),
                          ASVabundance = taxa_sums(ROHR_06_obj_clean),
                          ASV = taxa_names(ROHR_06_obj_clean))

ROHR_06_tax_plot <- ggplot(ROHR_06_df_taxa, aes(ASVabundance)) +
  geom_histogram() + ggtitle("Histogram of ASVs (unique sequence) counts (DADA2)") +
  theme_bw() + scale_x_log10() + ylab("Frequency of ASVs") + xlab("Abundance (raw counts)")

print(ROHR_06_tax_plot)

```

```{r examining number of reads, reads/ASV, singletons, doubletons }

#total number of reads in the dataset
reads_per_asv <- taxa_sums(ROHR_06_obj_clean)
print(sum(reads_per_asv)) #4418389

#total number of reads across all samples 
reads_per_sample <- sample_sums(ROHR_06_obj_clean)
print(sum(reads_per_sample)) #4418389 (sum should be same as above = total number of reads)

#how many ASVs with fewer than 10 reads?
print(length(reads_per_asv[reads_per_asv < 10])) #7451 

#how many reads do these ASVs contain?
print(sum(reads_per_asv[reads_per_asv < 10])) #40695

#how many samples with fewer than 10 reads?
print(length(reads_per_sample[reads_per_sample < 10])) #2

#how many reads do these samples contain?
print(sum(reads_per_sample[reads_per_sample < 10])) #6

#how many samples with fewer than 1000 reads?
print(length(reads_per_sample[reads_per_sample < 1000])) #17 samples with fewer than 1000 reads (out of 359 = ~5% of samples had fewer than 1,000 reads)

#how many reads do these samples contain?
print(sum(reads_per_sample[reads_per_sample < 1000])) #7461 

#how many singletons?
length(which(taxa_sums(ROHR_06_obj_clean) <= 1)) #29 

#how many doubletons?
length(which(taxa_sums(ROHR_06_obj_clean) == 2)) #653 doubletons

```
From the code above, we can see that our full swab dataset contains quite a few rare ASVs . In addition, 17 of our samples have fewer than 1,000 reads. We can further visualize this by plotting out the distribution of read counts in our samples.

```{r visualize number of reads per sample}
#calculate total number of reads per sample and sort from smallest # to largest #
sort(sample_sums(ROHR_06_obj_clean))

#visualize distribution of read counts
hist(sample_sums(ROHR_06_obj_clean), main="Histogram: Read Counts", xlab="Total Reads", 
     border="blue", col="green", las=1, breaks=50)
```
Looking at the histogram above, we can see that the majority of our samples have between 0 and ~12,000 reads, with fewer samples having higher numbers of reads (up to 48,663). To remove potentially spurious ASVs, we will want to filter these reads. We will begin by removing samples that have fewer than 1,000 reads, then further filter ASVs that are not present in at least 2 samples (these are relatively low-stringency filters, given that we expect the skin microbiome to harbor rare taxa and want to avoid filtering these out of the dataset if possible). 

```{r removing samples w/ <1000 reads}
# Remove samples with fewer than 1000 reads from phyloseq object
ROHR_06_obj_clean.1000 <- subset_samples(ROHR_06_obj_clean, sample_sums(ROHR_06_obj_clean) > 1000)

#revove ASVs not present in any sample (occurs after having removed samples)
ROHR_06_obj_clean.1000 <- prune_taxa(taxa_sums(ROHR_06_obj_clean.1000) > 0, ROHR_06_obj_clean.1000)
ROHR_06_obj_clean.1000 # down to 23391 taxa in 341 samples

#checking current number of reads
reads_per_asv1 <- taxa_sums(ROHR_06_obj_clean.1000)
print(sum(reads_per_asv1)) #4409928 reads left (removed 8461 reads)

#check that this worked & visualize distribution of read counts
hist(sample_sums(ROHR_06_obj_clean.1000), main="Histogram: Read Counts", xlab="Total Reads", 
     border="blue", col="green", las=1, breaks=50)

```

``` {r filter ASVs found in <2 samples}

####### Filter ASVs that are not present in at least 2 samples
condition <- function(x) { sum(x > 0) >= 2 }
taxaToKeep <- filter_taxa(ROHR_06_obj_clean.1000, condition)

ROHR_06_obj_clean_updated <- prune_taxa(taxaToKeep, ROHR_06_obj_clean.1000)
ROHR_06_obj_clean_updated #this brings us down to 7293 taxa in 341 samples

#checking current number of reads
reads_per_asv2 <- taxa_sums(ROHR_06_obj_clean_updated)
print(sum(reads_per_asv2)) #3991107 reads left (removed 418821 reads)

reads_per_sample2 <- sample_sums(ROHR_06_obj_clean_updated)
print(length(reads_per_sample2[reads_per_sample2 < 3000]))

#Save phyloseq object with filtered taxa
saveRDS(ROHR_06_obj_clean_updated, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean_updated.rds")

#Read phyloseq
ROHR_06_obj_clean_updated <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean_updated.rds")
```

```{r check number of samples per species region and season}
# Check how many samples of each fish species were collected in each gulf & season

## Extract the sample metadata
sample_data <- as.data.frame(sample_data(ROHR_06_obj_clean_updated))

# Create a contingency table of species by region and season
species_region_season_counts <- table(sample_data$species, sample_data$region, sample_data$season)

# Convert the table to a data frame for easier manipulation
species_region_season_df <- as.data.frame(species_region_season_counts)

# Rename columns for clarity
colnames(species_region_season_df) <- c("Species", "Region", "Season", "Count")

# Pivot data to have `Season` as columns and add row totals for each Species and Region combination
species_region_season_wide <- species_region_season_df %>%
  pivot_wider(names_from = Season, values_from = Count, values_fill = 0) %>%
  mutate(`n by Region` = rowSums(select(., -Species, -Region)))

# Calculate the summed total for each species across all regions
species_totals <- species_region_season_wide %>%
  group_by(Species) %>%
  summarize(`Total by Species` = sum(`n by Region`))

# Add the species totals to the main table
species_region_season_wide <- species_region_season_wide %>%
  left_join(species_totals, by = "Species")

# Mark "Total by Species" as `NA` except in the last region row for each species
species_region_season_wide <- species_region_season_wide %>%
  group_by(Species) %>%
  mutate(`Total by Species` = ifelse(row_number() == n(), `Total by Species`, NA)) %>%
  ungroup()

# Calculate column totals without duplicating the "n by Region" column in the bottom row
column_totals <- colSums(select(species_region_season_wide, -Species, -Region), na.rm = TRUE)
column_totals <- c(Species = "Total", Region = "", as.list(column_totals))
column_totals["n by Region"] <- NA  # Remove the duplicate in the "n by Region" column for the bottom row

# Bind column totals as the last row
species_region_season_table2 <- rbind(species_region_season_wide, column_totals)

# Arrange rows to group by Species, and display the final table
species_region_season_table2 <- species_region_season_table2 %>%
  arrange(Species, Region)

print(species_region_season_table2)

# Save table as csv file

# Specify correct file path
file_path <- "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/species_region_season_table2.csv"

# Save the table as a CSV file
write.csv(species_region_season_table2, file = file_path, row.names = FALSE)

# Message to confirm saving & correct location
cat("Table saved as:", file_path)
```
While removing rare ASVs is necessary, as many of these can be due to sequencing errors, we are also removing many ASVs potentially corresponding to true rare taxa found in the fish skin microbiome (this is the filtering step that removes the largest number of sequences by far, with a drop from 4409928 reads to 3991107 - 418821 reads removed). Greater sampling of each fish species may allow us to further identify these rare taxa. We've chosen to remove ASVs found in less than 2 samples as we are interested in determining the microbial communities associated with these different fishes' skin, rather than those that may randomly be found on this surface due to contact with the surrounding seawater. Downstream, we'll also use metrics that tend to weigh more abundant microbes, as there are most likely the dominant members of the fish skin microbiome which shape the environment and may play the most important functional roles for the host.

```{r filter ASVs unassigned at phylum level}

##Removing ASVs unassigned (NA) at the phylum level
ROHR_06_obj_clean_updated_P <- subset_taxa(ROHR_06_obj_clean_updated, Phylum != "NA")
ROHR_06_obj_clean_updated_P #down to 6551 taxa in 341 samples (742 taxa unassigned at phylum level removed) --> potentially add these back in as needed, depending on analyses to be run

#checking new number of reads
reads_per_asv3 <- taxa_sums(ROHR_06_obj_clean_updated_P)
print(sum(reads_per_asv3)) #3069516 reads left (921591 reads removed)

#Save phyloseq object with filtered taxa
saveRDS(ROHR_06_obj_clean_updated_P, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean_updated_phyla.rds")

#read pyloseq object (if not starting from top)
ROHR_06_obj_clean_updated_P = readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean_updated_phyla.rds")
```

######################################################
### ALPHA DIVERSITY METRICS
- Measures within individual taxa richness & evenness
- Common metrics/indices = Shannon, Inverse Simpson, Simpson, Gini, Observed and Chao1 (do not take into account phylogeny of taxa)
- Faith's PD (phylogenetic diversity) --> uses phylogenetic distance

****"It is important to note that, alpha diversity indices are sensitive to noise that is inherent to application of polymerase chain reaction and the sequencing errors" - Shetty et al (2020)

*** Need to consider sequencing depth - that is, how many taxa did we actually capture - for each sample, and, as needed, normalize samples to equal depth before running alpha diversity indices

Alpha diversity lecture: https://evolution.unibas.ch/walser/bacteria_community_analysis/2015-02-10_MBM_tutorial_combined.pdf

```{r look at number of reads}
summary(sample_sums(ROHR_06_obj_clean_updated_P)) #see that there is a large difference in number of reads (407 to 39934) --> although we filtered out samples with fewer than 1,000 reads earlier, the next steps removed a considerable number of ASVs, thus reducing the number of reads/sample 
```

Plot rarefaction curve for the ASV data for the full dataset. This allows us to see if/how microbial richness has been captured by the sequencing effort. All samples appear to reach a plateau, indicating sufficient sampling depth to capture most diversity, although some samples with >400 species (unique ASVs) and fewer reads are likely missing some rare taxa. Most samples begin to reach a plateau at around 2,500 reads, which we'll keep in mind when we rarefy our dataset to even depth for comparisons across samples and diversity metrics. Out of curiosity, I plotted 3 sets of curves, each corresponding to the filtering steps we did above (after removing samples with <1000 reads; after removing ASVs in <2 samples; after removing ASVs not assigned to phylum level), which allows us to visualize how these filtering steps impact the # reads & ASVs per sample (changing the total sample size and slightly reducing the number of species seen in a given sample, but not the slope of the accumulation curves).
```{r rarefaction curves for full dataset after removing samples w/ <1000 reads}
otu_tab_ROHR06c <- t(abundances(ROHR_06_obj_clean.1000))
rarcurve_ROHR06c <- vegan::rarecurve(otu_tab_ROHR06c, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab_ROHR06c), 
                                   col = "blue", cex = 0.6))
```

```{r rarefaction curves for trimmed dataset after removing ASVs in <2 samples}
otu_tab_ROHR06b <- t(abundances(ROHR_06_obj_clean_updated))
rarcurve_ROHR06b <- vegan::rarecurve(otu_tab_ROHR06b, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab_ROHR06b), 
                                   col = "blue", cex = 0.6))
```

```{r rarefaction curves for full (trimmed) dataset - unassigned phyla removed, echo=FALSE}
otu_tab_ROHR06 <- t(abundances(ROHR_06_obj_clean_updated_P))
rarcurve_ROHR06 <- vegan::rarecurve(otu_tab_ROHR06, 
                      step = 50, label = FALSE, 
                      sample = min(rowSums(otu_tab_ROHR06), 
                                   col = "blue", cex = 0.6))
```
For the a-diversity analyses, we will begin by rarefying (normalizing) our data to at least 2,500 reads (this will exclude  more samples and  ASVs), to try and capture as many samples as possible while standardizing the library size and capturing the microbial species present (see Weiss et al. 2017). There is great debate about the practice of rarefying samples to even depth - here we have chosen to follow the advice given by recent publications including Schloss et al. (2023 pre-print), Weiss et al. (2017), and the package developed by Cameron et al. (2021) to repeatedly rarefy (that is, draw our target number of sequences from the same sample multiple times) our samples to even sequencing depth. We chose 2,500 reads as our cutoff despite the fact that it will remove ~22% of our samples (combined with the earlier filtering step, this means removing 79 samples out of 359), due to the rarefaction curves (shown above) showing that below ~2,500, many species are not captured for a given sample. Unlike some studies that place the cutoff much higher (e.g., 10,000 reads), the nature of our dataset meant that we had many samples with lower read counts and high heterogeneity in reads/ASVs per sample. This cutoff can be modified to exclude fewer/more samples with low sequencing depth.

mirlyn package for rarefying libraries through multiple iterations (Cameron et al. 2021)
https://github.com/escamero/mirlyn

```{r normalizing data - repeated rarefaction to 2500 reads}
#first need to transform data format from phyloseq which has taxa = columns to MCMCglmm package (ued by mirl) taxa = rows format
# Create a sample phyloseq object --> used phyloseq object with rooted tree created later in the pipeline
physeqROHR06_2 <- ROHR_06_obj_phylo_root

# Access the abundance table and transpose it
abundance_transposed <- t(otu_table(physeqROHR06_2))

# Create a new phyloseq object with the transposed abundance table
physeqROHR06_transposed2 <- phyloseq(otu_table(abundance_transposed), sample_data(physeqROHR06_2), tax_table(physeqROHR06_2))

#here we rarefy the full dataset down to 2500 reads per sample 100 times (creating a new object with a list of all 100 of these subsampled datasets), NOT trimming any OTUs that are no longer in the rarefied dataset (this line created an issue). OTUs are samples w/o replacement (avoiding possible double-counting of an OTU in a given sample) 
#note that doing this repeated sampling takes time (on laptop took ~5 hrs to run 1000 iterations and maxed out memory), re-ran with 10 iterations (for now - computer is able to deal with ~100 reps, but this takes ~1 hr to run) and would need to switch to cluster to run with higher computational power.
ROHR06.rar.rep.2 <- mirl(physeqROHR06_transposed2, libsize = 2500, rep = 10, set.seed = 777, replace = FALSE) #62 samples removed at this step
```

```{r a-div with mirlyn package}

# Generates dataframe of alpha-diversity metric from mirl_object (from Cameron et al. 2021) w/ uncertainty & variation due to rarefaction characterized
# default diversity index = Shannon diversity index, but all diversity indices in vegan are supported

#Shannon diversity
alphadiv_df <- alphadivDF(ROHR06.rar.rep.2)

# Generates visualization from alphadiv_df - Shannon diversity index
alphawichVis(alphadiv_df, xvar = "species")


#Inverse simpson
alphadiv_df_invsim <- alphadivDF(ROHR06.rar.rep.2, diversity = "invsimpson")

# Generates visualization from alphadiv_df - 
alphawichVis(alphadiv_df_invsim, xvar = "species")

```

issue with betamatPCA_object (maybe from initial data transformation done at the beginning to run rarefaction?) 11/13/23
```{r b-div PCA with mirlyn package - NOT RUN}

####HAVE NOT MANAGED TO GET THIS CODE TO RUN

# Generates PCA from mirl_object for specified dissimilarity metric (dsim).
betamatPCA_object <- betamatPCA(ROHR06.rar.rep.2, dsim = "bray")

# Generates visualization of PCA.
betamatPCAvis(betamatPCA_object, groups = c("Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Cephalopholis colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis"), reps = 10, colours = c("aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue","deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2"))

```

```{r normalizing data - single sampling w/o replacement}
#check current reads/sample
reads_per_sample <- sample_sums(ROHR_06_obj_clean_updated_P)

#how many samples with fewer than 2500 reads?
print(length(reads_per_sample[reads_per_sample < 2500])) #62 samples (& 690 ASVs) will be excluded under this criterion

set.seed(9242)  # This will help in reproducing the filtering and normalisation. 

ROHR06.rar <- rarefy_even_depth(ROHR_06_obj_clean_updated_P, sample.size = 2500) 

```

```{r save rarefied data as new phyloseq object}
# Save rarefied data
saveRDS(ROHR06.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_2500.rds")

#read rarefied data
ROHR06.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_2500.rds")

#check how much data post-rarefaction
print(ROHR06.rar) # 6482 taxa, 279 samples
```

```{r check number of samples per species region and season after rarefying}
# Check how many samples of each fish species were collected in each gulf & season

## Extract the sample metadata
sample_data <- as.data.frame(sample_data(ROHR06.rar))

# Create a contingency table of species by region and season
species_region_season_counts <- table(sample_data$species, sample_data$region, sample_data$season)

# Convert the table to a data frame for easier manipulation
species_region_season_df <- as.data.frame(species_region_season_counts)

# Rename columns for clarity
colnames(species_region_season_df) <- c("Species", "Region", "Season", "Count")

# Pivot data to have `Season` as columns and add row totals for each Species and Region combination
species_region_season_wide <- species_region_season_df %>%
  pivot_wider(names_from = Season, values_from = Count, values_fill = 0) %>%
  mutate(`n by Region` = rowSums(select(., -Species, -Region)))

# Calculate the summed total for each species across all regions
species_totals <- species_region_season_wide %>%
  group_by(Species) %>%
  summarize(`Total by Species` = sum(`n by Region`))

# Add the species totals to the main table
species_region_season_wide <- species_region_season_wide %>%
  left_join(species_totals, by = "Species")

# Mark "Total by Species" as `NA` except in the last region row for each species
species_region_season_wide <- species_region_season_wide %>%
  group_by(Species) %>%
  mutate(`Total by Species` = ifelse(row_number() == n(), `Total by Species`, NA)) %>%
  ungroup()

# Calculate column totals without duplicating the "n by Region" column in the bottom row
column_totals <- colSums(select(species_region_season_wide, -Species, -Region), na.rm = TRUE)
column_totals <- c(Species = "Total", Region = "", as.list(column_totals))
column_totals["n by Region"] <- NA  # Remove the duplicate in the "n by Region" column for the bottom row

# Bind column totals as the last row
species_region_season_table3 <- rbind(species_region_season_wide, column_totals)

# Arrange rows to group by Species, and display the final table
species_region_season_table3 <- species_region_season_table3 %>%
  arrange(Species, Region)

print(species_region_season_table3)

# Save table as csv file

# Specify correct file path
file_path <- "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/species_region_season_table3_rar.csv"

# Save the table as a CSV file
write.csv(species_region_season_table3, file = file_path, row.names = FALSE)

# Message to confirm saving & correct location
cat("Table saved as:", file_path)
```
### Alpha diversity - non-phylogenetic diversities

- Alpha diversity metrics/indices give us information about the variation of microbes within a single sample

The "alpha" function within the microbiome package gives "global ecosystem state variables", including richness, evenness, and diversity. 
```{r calculate a-div - observed richness, other forms of richness, diversity, evenness, dominance, rarity...}
ROHR06.div <- alpha(ROHR06.rar, index = "all")

datatable(ROHR06.div)

write.csv(ROHR06.div, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics.csv")

#if not starting from beginning: read csv file:
ROHR06.div <- read.csv("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics.csv")
```

```{r save this table to word doc}

# Create a flextable object from your data
ft <- flextable(ROHR06.div)

# Save the flextable to a Word document
doc <- read_docx()
doc <- body_add_flextable(doc, ft)
print(doc, target = "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics_rarefied2500.docx")

```

```{r fix sample IDs to not start with an integer}
#make sure to only run this once, otherwise you'll affix multiple names
sample_names(ROHR_06_obj_clean_updated_P) <- paste("RRR", gsub("-", "_", sample_names(ROHR_06_obj_clean_updated_P)), sep = "_")

```

```{r calculate a-div not-rarefied, observed richness, other forms of richness, diversity, evenness, dominance, rarity...}
ROHR06.div.whole <- alpha(ROHR_06_obj_clean_updated_P, index = "all")

datatable(ROHR06.div.whole)

write.csv(ROHR06.div.whole, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics_non_rarefied.csv")

#if not starting from beginning: read csv file:
ROHR06.div.whole <- read.csv("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics_non_rarefied.csv")
```

```{r save this table to word doc}

# Create a flextable object from your data
ft <- flextable(ROHR06.div.whole)

# Rename columns (replace with actual column names as needed)
colnames(ROHR06.div.whole) <- c("Sample ID", "Observed Richness", "Chao1", "Inverse Simpson Div", "Gini Simpson Div","Shannon Div", "Fisher Div", "Coverage Div", "Camargo Even", "Pielou Even", "Simpson Even", "Evar Even", "Bulla Even", "Dom DBP", "Dom DMN", "Dom Absolute", "Dom Relative", "Dom Simpson", "Dom Core Abund", "Dom Gini", "Rar log Modulo Skewness", "Rar Low Abund", "Rar Rare Abund")

# Define a function to split data by columns (keeping "Sample ID")
split_columns <- function(data, columns_per_chunk) {
  col_indices <- seq(2, ncol(data))  # Skip the "Sample ID" column (index 1)
  chunks <- split(col_indices, ceiling(seq_along(col_indices) / columns_per_chunk))
  
  # Add "Sample ID" to each chunk
  lapply(chunks, function(cols) data[, c(1, cols)])
}

# Split data into chunks of columns (excluding "Sample ID" as the first column)
column_chunks <- split_columns(ROHR06.div.whole, columns_per_chunk = 11)

# Create a new Word document
doc <- read_docx()

# Loop through column chunks and add each as a separate table with "Sample ID" in bold
for (i in seq_along(column_chunks)) {
  chunk <- column_chunks[[i]]
  
  # Create a flextable for each column chunk with "Sample ID" in bold
  ft_chunk <- flextable(chunk) %>%
    theme_vanilla() %>%
    align(align = "center", part = "all") %>%
    set_table_properties(layout = "autofit", width = 1) %>%
    bold(j = "Sample ID", bold = TRUE) %>%
    bold(i = 1, bold = TRUE, part = "header") %>%
    fontsize(size = 6, part = "all") %>%
    bg(i = 1, bg = "#D3D3D3", part = "header") 

  # Add the table chunk to the document
  doc <- body_add_flextable(doc, value = ft_chunk)
  doc <- body_add_par(doc, "") # Add space between tables if needed
}

# Save the document
print(doc, target = "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics_non_rarefied.docx")

```


```{r plot a-div data (one option) - shannon diversity boxplots}
# get the metadata out as separate object
ROHR06.meta <- meta(ROHR06.rar)

# Add the rownames as a new column for easy integration later.
ROHR06.meta$sam_name <- rownames(ROHR06.meta)

# Add the rownames to diversity table
ROHR06.div$sam_name <- rownames(ROHR06.div)

# merge these two data frames into one
div.df <- merge(ROHR06.div, ROHR06.meta, by = "sam_name")

# check the tables
colnames(div.df)

#reorder samples so that fish are grouped by diet
div.df <- div.df %>%
  mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Paranthias colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

# Now use this data frame to plot 
ROHR06.shannon <- ggboxplot(div.df, 
               x = "species", 
               y = "diversity_shannon",
              fill = "species", 
              palette = "rickandmorty",
              title = "Shannon Diversity of TEP Fish Skin Microbiome",
              xlab = "Host species",
              ylab = "Diversity (Shannon)")

ROHR06.shannon <- ROHR06.shannon + rotate_x_text()

print(ROHR06.shannon)

# Add significance comparisons (are there significant differences in Shannon diversity between fish species?)
# For better visualization, need to change the way significance comparisons are displayed across bars.

colnames(div.df)
div_df_melt <- reshape2::melt(div.df)

lev <- levels(div_df_melt$species) # get the variables

# make a pairwise list that we want to compare.
L.pairs <- combn(seq_along(lev), 2, simplify = FALSE, FUN = function(i) lev[i])

pval <- list(
  cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
  symbols = c("****", "***", "**", "*", "n.s")
)


ROHR06.alpha.comp.shannon <- ROHR06.shannon + stat_compare_means(
  comparisons = L.pairs,
  label = "p.signif",
  symnum.args = list(
    cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 0.1, 1),
    symbols = c("****", "***", "**", "*", "n.s")
  )
)

print(ROHR06.alpha.comp.shannon)

# facet by region & season
ROHR06.shannon.RS <- ggboxplot(div.df, 
               x = "species", 
               y = "diversity_shannon",
              fill = "species", 
              palette = "rickandmorty",
              title = "Shannon Diversity of TEP Fish Skin Microbiome",
              xlab = "Host species",
              ylab = "Diversity (Shannon)",
              facet.by = "region_season")

ROHR06.shannon.RS <- ROHR06.shannon.RS + rotate_x_text()

print(ROHR06.shannon.RS)

```
```{r code for exporting the last a-diversity plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Shannon Diversity Region Season 21-22", path = '~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

Plotting a-diversity (in this case using Shannon Diversity) by host species, shows that diversity is high across all hosts (close to upper range, as Shannon Index goes from 0-5), with some variation in amount of within-sample diversity depending on the species (e.g., C. humeralis appears to generally have lower alpha diversity than most other fish, regardless of sampling season or region). Faceting by season & region shows that, although some species (e.g. A. xanthopterus in Coiba and C. humeralis at both sites) appear to host differentially diverse microbiomes by season, most remain relatively consistent across seasons and regions, for the most part having scores ~4. These high scores make sense given that Shannon Diversity measures both the number of species (in this case, microbial taxa) and inequality between species abundances, both of which are high in microbial datasets (and can be further visualized with the stacked barplots of relative abundances in the b-diversity section below).

--> explanation + formulas & calculations of Shannon-Wiener & Simpson's Indices: https://www.webpages.uidaho.edu/veg_measure/Modules/Lessons/Module%209(Composition&Diversity)/Detailed_Calculations.htm

We can gain a bit more insight on the distribution of the alpha diversity metrics by using violin plots, instead of simple box plots, following code outlined in: https://rpubs.com/lgschaerer/515637

***note that in the "plot_richness" documentation, they specifically advise against working with trimmed, normalized count data to obtain meaningful results, but the data I'm currently working with HAS been trimmed and normalized --> "You must use untrimmed, non-normalized count data for meaningful results, as many of these estimates are highly dependent on the number of singletons. You can always trim the data later on if needed, just not before using this function." 

Currently have 3 versions of the plot, one with the filtered and rarefied data (ROHR06.rar), one with the non-rarefied and filtered data (samples with fewer than 2,500 reads + ASVs found in only 2 samples + ASVs unassigned at phylum level removed - ROHR_06_obj_clean_updated_P), and a final one with partially filtered data (only samples with fewer than 1,000 reads removed - ROHR_06_obj_clean). Pick the one that is most appropriate to accurately display alpha diversity. 
```{r plotting alpha diversity (another option using observed + shannon & violin plots)}
#reorder samples so that fish are grouped by diet
ROHR06.rar <- ROHR06.rar %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Paranthias colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

###Assign each category a color, define sample types and create sample labels

#picked colors by trophic group, with herbivores as different shades of green (territorial = aqua, roving = bright green), carnivores brown, planktivore purple, invertivores blue, and omnivore golden yellow

sample_colors <- c("Abudefduf concolor" = "aquamarine4", "Microspathodon dorsalis" = "aquamarine2", "Acanthurus xanthopterus" = "chartreuse3", "Prionurus laticlavius" = "darkgreen", "Cephalopholis panamensis" = "brown4", "Epinephelus labriformis" = "chocolate2", "Paranthias colonus" = "darkmagenta", "Johnrandallia nigrirostris" = "darkslategray", "Chaetodon humeralis" = "deepskyblue2", "Abudefduf troschelii" = "darkgoldenrod1")

sample_types <- c("Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Cephalopholis panamensis", "Epinephelus labriformis", "Paranthias colonus", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Abudefduf troschelii")

sample_labels <- c("Abudefduf concolor" = "A. concolor", "Microspathodon dorsalis" = "M. dorsalis", "Acanthurus xanthopterus" = "A. xanthopterus", "Prionurus laticlavius" = "P. laticlavius", "Cephalopholis panamensis" = "C. panamensis", "Epinephelus labriformis" = "E. labriformis", "Paranthias colonus" = "P. colonus", "Johnrandallia nigrirostris" = "J. nigrirostris", "Chaetodon humeralis" = "C. humeralis", "Abudefduf troschelii" = "A.troschelii")

ROHR06.rar_adiv_obs_shan <- ROHR06.rar %>%                                                              #phyloseq object (rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+                        #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  ggtitle("Alpha Diversity of TEP Fish Skin Microbiome (rarefied to 2,500 reads)") +                          #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

ROHR06.rar_adiv_obs_shan

####repeat with non-rarefied (but filtered to only include samples with >1000 reads and ASVs that appear in >2 samples) data

#reorder samples so that fish are grouped by diet
ROHR_06_obj_clean_updated_P <- ROHR_06_obj_clean_updated_P %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Paranthias colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

ROHR_06_obj_clean_updated_P %>%                                                              #phyloseq object (non rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+   #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  ggtitle("Alpha Diversity TEP Fish Skin Microbiome (non-rarefied data, ASVs trimmed)") +                     #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

####repeat with non-rarefied (but filtered to only include samples with >1,000 reads) data

#reorder samples so that fish are grouped by diet
ROHR_06_obj_clean.1000 <- ROHR_06_obj_clean.1000 %>%
  ps_mutate(species = fct_relevel(species, 
           "Abudefduf concolor", "Microspathodon dorsalis", "Acanthurus xanthopterus", "Prionurus laticlavius", "Johnrandallia nigrirostris", "Chaetodon humeralis", "Paranthias colonus", "Abudefduf troschelii", "Cephalopholis panamensis", "Epinephelus labriformis" ))

ROHR_06_obj_clean.1000 %>%                                                              #phyloseq object (non rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed", "Shannon")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+   #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+  
  ggtitle("Alpha Diversity TEP Fish Skin Microbiome (non-rarefied data)") +                                 #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position

# facet by season & region (although it makes it harder to see the distribution as everything is condensed)
ROHR06.rar %>%                                                              #phyloseq object (rarefied)
  plot_richness(
    x = "species",                                                #compare diversity of datatype
    measures = c("Observed")) +                           #choose diversity measures
  geom_violin(aes(fill = species), show.legend = FALSE)+          #make violin plot, set fill aes to species
  geom_boxplot(width=0.1) +                                          #add boxplot, set width
  theme_classic()+                                                   #change theme to classic
  xlab(NULL)+                                                        #no label on x-axis
  theme(axis.text.y.left = element_text(size = 20),                  #adjust y-axis text
        axis.text.x = element_text(size = 20, hjust = 0.5, vjust = 1, angle = 0),           #adjust x-axis label position
        axis.title.y = element_text(size = 20))+                     #adjust y-axis title
  theme(strip.text = element_text(face = "bold", size = 25))+        #adjust headings
  scale_fill_manual(values = sample_colors)+   #set fill colors
  scale_x_discrete(                                                  #change x-axis labels
    breaks = sample_types, 
    labels = sample_labels)+ 
  facet_wrap("region_season") +
  ggtitle("Alpha Diversity (Observed) of TEP Fish Skin Microbiome (rarefied to 1,000 reads)") +                                       #add title
  rotate_x_text()+
  theme(plot.title=element_text(size = 25, face = "bold", hjust = 0.5)) #change title size, face and position
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Alpha Diversity Obs & Shan", plot = ROHR06.rar_adiv_obs_shan, path = '~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

From these plots, we can see that a-div is generally high, with Observed Diversity (the count of unique OTUs in each sample - a richness measure) generally matching Shannon Diversity (richness + diversity/rel. abund). Observed Diversity is noticeably lower in C. humeralis, J. nigrirostris, and (to a lesser extent) A. troschelii (which has a long tail to the upper end). C. humeralis and J. nigrirostris are both sessile invertebrate feeders --> could lower a-div be linked to this feeding strategy/diet? 


https://mibwurrepo.github.io/Microbial-bioinformatics-introductory-course-Material-2018/alpha-diversities.html#equal-sample-sums

Composite plot with multiple alpha diversity indices, including Inverse Simpson, Gini-Simpson, Shannon, Fisher, and Coverage (although these can be modified to select from the full list contained in the div.df data frame)

- Inverse Simpson Index:
"the effective number of types that is obtained when the weighted arithmetic mean is used to quantify average proportional abundance of types in the dataset of interest" (rdocumentation.org)
In other words, the Inverse Simpson Index is the reciprocal of Simpson's Index (see below), where 1 = low diversity, and highest diversity would be equal to the maximum number of species in the sample
        --> Simpson's Index (D) = quantifies probability that two individuals randomly selected from a sample will belong                to the same species. 0 represents infinite diversity, 1 represents no diversity --> makes little intuitive                 sense, so we often use the Diversity Index (see below) instead...
        --> Simpson's Diversity Index (1 - D) = takes D, subtracts from 1 so that highest diversity = 1 --> at high values               (nearing 1 or 100%), probability that two individuals selected at random will belong to different species is
              high. 
http://www.countrysideinfo.co.uk/simpsons.htm

- Gini-Simpson Index - aka Gibbs-Martin or Blau index
        --> 1 - sum of the (proportion of a given species within a community) squared

- Shannon Diversity Index - aka Shannon-Wiener Index:
        --> Measures diversity of species in a community by taking the (negative of the) sum of the proportion of a given species within a community times the natural log of that same proportion. Higher index = more species present (1 species = 0)

- Fisher Diversity Index - aka Fisher's alpha:
"a diversity index, defined implicitly by the formula S=a*ln(1+n/a) where S is number of taxa, n is number of individuals and a is the Fisher's alpha." https://www.nhm.uio.no/english/research/resources/past/help/diversity.html
       --> parametric diversity index that assumes species abundance follows log distribution
       
- Diversity Coverage 
"The coverage index gives the number of groups needed to have a given proportion of the ecosystem occupied (by default 0.5 ie 50%)." https://microbiome.github.io/tutorials/Alphadiversity.html

```{r alpha diversity plots - multiple indices}
# convert phyloseq object into a long data format.

div.df2 <- div.df[, c("species", "diversity_inverse_simpson", "diversity_gini_simpson", "diversity_shannon", "diversity_fisher", "diversity_coverage")]

# Replace names of diversity metrics/indices

colnames(div.df2) <- c("Species", "Inverse Simpson", "Gini-Simpson", "Shannon", "Fisher", "Coverage")

# check
colnames(div.df2)


### Using Species as id variables
div_df_melt <- reshape2::melt(div.df2)

head(div_df_melt)

# Now use this data frame to plot 
ROHR06_alpha_div <- ggboxplot(div_df_melt, x = "Species", y = "value",
              fill = "Species", 
              palette = "rickandmorty", 
              legend= "right",
              facet.by = "variable", 
              scales = "free")

ROHR06_alpha_div <- ROHR06_alpha_div + rotate_x_text() # we will remove the x axis labels

ROHR06_alpha_div <- ROHR06_alpha_div + rremove("x.text")
ROHR06_alpha_div
```
```{r code for exporting multi-index a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Alpha Diversity Plots 21-22", plot = ROHR06_alpha_div, path = '~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

### Adding a Phylogenetic Tree

To calculate phylogenetic diversity, we will need to create a phylogenetic tree with our microbiome data. We referenced: Workflow for Microbiome Data Analysis: from raw reads to community analyses, by Benjamin J Callahan, Kris Sankaran, Julia A Fukuyama, Paul Joey McMurdie and Susan P Holmes to help with tree creation. (It also includes information on many other steps in the microbiome processing and data visualization pipeline) https://bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/Microbiome/MicrobiomeWorkflowII.html#construct_phylogenetic_tree

Since the DADA2 method used to assign taxonomy (ASVs) to our 16S reads is reference free, we need to connect the ASVs to their phylogenies de novo. The multiple-alignment step will be done using the DECIPHER R package (2.26.0 = current version loaded); Wright (2016): http://bioconductor.org/packages/release/bioc/html/DECIPHER.html

```{r multiple alignment using DECIPHER package}
#if get error with getSequences function ("Error in getSequences() : could not find function "getSequences") run:
library(dada2); packageVersion("dada2") #to ensure dada2 loaded properly
tax_silva_tax  = tax_table(ROHR_06_obj_clean_updated_P)

seqs <- getSequences(tax_silva_tax)
names(seqs) <- seqs # This propagates to the tip labels of the tree
alignment <- AlignSeqs(DNAStringSet(seqs), anchor=NA,verbose=FALSE)
```

Once multiple alignment has been done using DECIPHER (2.26.0), we can use the phangorn package (2.11.1) to build a phylogenetic tree.
The code below first builds a neighbor-joining tree, then fits a GTR+G+I maximum likelihood tree ( Generalized time-reversible with Gamma rate variation) using the neighbor-joining tree. Note that the maximum likelihood tree step requires considerable computational power - best to run on cluster (ran on laptop local hard drive for 27 hrs). 

Helpful lecture on different tree types and how they're constructed: https://si.biostat.washington.edu/sites/default/files/modules/2018-SISMID-04.pdf

CRAN documentation for phangorn: https://cran.r-project.org/web/packages/phangorn/vignettes/Trees.html
***this is the most thorough documentation, with steps for distance based methods, parsimony, and maximum likelihood trees

The Molecular Ecologist - "Quick and Dirty Tree Building in R": https://www.molecularecologist.com/2016/02/26/quick-and-dirty-tree-building-in-r/


```{r build NJ tree then GTR+G+I maximum likelihood tree}
#this is old code for previous run-through of skin swab data (did not re-run as although exact reads have been modified slightly by Cutadapt edits, best model should remain +/- the same)

#prior to producing a distance matrix, it is best to test which model best fits data, using a likelihood ratio test

#mt <- modelTest(phangAlign) #this runs the full gammut of models (92) and takes a while to run
#print(mt)


#looking at the different models and sorting them from lowest (best) AIC to highest, we can see that the models with the lowest AIC scores are:
# Model Name         AIC 
# GTR + G(4)      474026.9
# TVM + G(4)      474037.8
# GTR + G(4) + I  474073.5 ###current model
# TVM + G(4) + I  474153.3
# TIM3 + G(4) + I 474378.4

#as.pml(mt, "BIC") #best model according to BIC --> AIC & BIC agree for top models

#model: GTR+G(4) 
#loglikelihood: -224643.4 
#unconstrained loglikelihood: -3173.164 
#Discrete gamma model
#Number of rate categories: 4 
#Shape parameter: 0.5150499 

#Rate matrix:
#          a         c         g        t
#a 0.0000000 0.8578735 3.1551964 1.440954
#c 0.8578735 0.0000000 0.5895977 3.625795
#g 3.1551964 0.5895977 0.0000000 1.000000
#t 1.4409536 3.6257951 1.0000000 0.000000

#Base frequencies:  
#        a         c         g         t 
#0.2475282 0.1917162 0.2644824 0.2962732 

######note that the current model used to calculate the phylogenetic tree below is GTR+G(4)+I, which is among the top 5 models when sorted by AIC score (could perhaps further optimize by using GTR+G(4) instead, but difference b/w these models is small)

#view GTR+G(4)+I model used below --> values are slightly different from those of fitGTR below

#as.pml(mt, "GTR+G(4)+I")

#model: GTR+G(4)+I 
#loglikelihood: -224665.7 
#unconstrained loglikelihood: -3173.164 
#Proportion of invariant sites: 0.1502231 
#Discrete gamma model
#Number of rate categories: 4 
#Shape parameter: 0.5105912 

#Rate matrix:
#          a         c         g        t
#a 0.0000000 0.9144501 3.1121425 1.424444
#c 0.9144501 0.0000000 0.6417293 3.914681
#g 3.1121425 0.6417293 0.0000000 1.000000
#t 1.4244442 3.9146807 1.0000000 0.000000

#Base frequencies:  
#        a         c         g         t 
#0.2528884 0.1810248 0.2676787 0.2984081 

###alternatively, can select a few (rather than the full suite of) models to test
#mt <- modelTest(phangAlign, model=c("JC", "F81", "K80", "HKY", "SYM", "GTR"), 
#                control = pml.control(trace = 0))


#save model test results as plain text file (to avoid having to run again)
#write.table(mt, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR06_distance_models.txt")

#resumed running code here 02/02/23

phangAlign <- phyDat(as(alignment, "matrix"), type="DNA")
dm <- dist.ml(phangAlign)
treeNJ <- NJ(dm) #Neighbor joining tree ### Note, tip order != sequence order
#plot(treeNJ, "unrooted", main="NJ") #plot unrooted Neighbor Joining tree - given the number of tips and the fact that each tip is currently labeled with a sequence, this just gives us a big, unreadable block of black ink 
treeUPGMA  <- upgma(dm) #UPGMA (unweighted pair group method with arithmetic mean)

#check parsimony score --> the min number of changes needed to describe data for the tree: best tree = lower score
parsimony(treeUPGMA, phangAlign)
#54319 for UPGMA tree

parsimony(treeNJ, phangAlign)
#52799 for NJ tree (better - improve upon this one with maximum likelihood (ML) method below)

fit = pml(treeNJ, data=phangAlign) #pml object contains the data + tree + many parameters of the model, inc. likelihood
fit 
fitGTR <- update(fit, k=4, inv=0.2)
fitGTR <- optim.pml(fitGTR, model="GTR", optInv=TRUE, optGamma=TRUE,
                    rearrangement = "stochastic", control = pml.control(trace = 0)) #rearrangement = "stochastic" helps escape local optima by performing stochastic rearrangements
fitGTR

#model: GTR+G(4)+I 
#loglikelihood: -232863.2 
#unconstrained loglikelihood: -3578.576 
#Proportion of invariant sites: 0.3147269 
#Discrete gamma model
#Number of rate categories: 4 
#Shape parameter: 0.5015827 

#Rate matrix:
#          a         c         g        t
#a 0.0000000 0.8781283 3.2162936 1.306211
#c 0.8781283 0.0000000 0.6944254 4.096171
#g 3.2162936 0.6944254 0.0000000 1.000000
#t 1.3062110 4.0961714 1.0000000 0.000000

#Base frequencies:  
#        a         c         g         t 
#0.2690807 0.1683362 0.2475712 0.3150119 

AIC(fitGTR)
#491944.4
BIC(fitGTR)
#550693.4

#bootstrap to test how well edges of tree are supported
#bs <- bootstrap.pml(fitGTR, bs=100, optNni=TRUE,
#    control = pml.control(trace = 0)) #####have not run this yet - supposed to be computationally demanding - will wait until not running other analyses --> started running on 2/6/23 in early morning, has run (when Mac not sleeping) for past 36+ hrs - will need to shift to cluster: stopped running code chunk on 2/8 to restart computer and move forward with b-div code

#cnet <- consensusNet(bs, p=0.2) ######have not run this yet
#plot(cnet, show.edge.label=TRUE) ####have not run this yet

#alternative function to do this which hides a few steps (starting from :
# fitGTR <- pml_bb(treeNJ, model="GTR+G(4)+I", control = pml.control(trace = 0))
#fitGTR

```

This phylogenetic tree will be added to a phyloseq object with (1) a sample-by-sequence table, (2) sample metadata, and (3) sequence taxonomies (see code below)

```{r write & read phylogenetic tree (unrooted)}
tree_NJ_tree = phy_tree(fitGTR$tree)

#save tree (on its own) in Newick format
ape::write.tree(tree_NJ_tree, file='~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR06_phylo_tree.tre')

tree_NJ_tree <- read.tree('~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR06_phylo_tree.tre')

#merge phylogenetic tree with phyloseq object

ROHR_06_obj_phylo <- merge_phyloseq(ROHR_06_obj_clean_updated_P, tree_NJ_tree)
ROHR_06_obj_phylo

#phyloseq-class experiment-level object --> this was the old phyloseq object with the old cutadapt error
#otu_table()   OTU Table:         [ 6182 taxa and 340 samples ]
#sample_data() Sample Data:       [ 340 samples by 43 sample variables ]
#tax_table()   Taxonomy Table:    [ 6182 taxa by 6 taxonomic ranks ]
#phy_tree()    Phylogenetic Tree: [ 6182 tips and 6180 internal nodes ]

#phyloseq-class experiment-level object --> this is the new one when Cutadapt was re-run
#otu_table()   OTU Table:         [ 6551 taxa and 341 samples ]
#sample_data() Sample Data:       [ 341 samples by 43 sample variables ]
#tax_table()   Taxonomy Table:    [ 6551 taxa by 6 taxonomic ranks ]
#phy_tree()    Phylogenetic Tree: [ 6551 tips and 6549 internal nodes ]

saveRDS(ROHR_06_obj_phylo, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_with_phylogenetic_tree.rds")

#read RDS file (if not re-running code from start)

ROHR_06_obj_phylo <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_with_phylogenetic_tree.rds")

```

```{r Root phylogenetic tree}

tree.unrooted <- phy_tree(ROHR_06_obj_phylo)

# Function to root the tree
pick_new_outgroup <- function(tree.unrooted){
  require("magrittr")
  require("data.table")
  require("ape") # ape::Ntip
  # tablify parts of tree that we need.
  treeDT <-
    cbind(
      data.table(tree.unrooted$edge),
      data.table(length = tree.unrooted$edge.length)
    )[1:Ntip(tree.unrooted)] %>%
    cbind(data.table(id = tree.unrooted$tip.label))
  # Take the longest terminal branch as outgroup
  new.outgroup <- treeDT[which.max(length)]$id
  return(new.outgroup)
}

new.outgroup = pick_new_outgroup(tree.unrooted)
new.outgroup

#new outgroup - that is the new root to be added to our tree = TACGGAGGGTGCAAGCGTTATCCGGAATCACTGGGTTTAAAGGGTGCGTAGGCGGAATTATAAGTCAGAGGTGAAATCCCGTCGCTAAACGACGGAACTGCCTTTGATACTGTTATTCTTGAATTGTGTTGAGGTTAGCGGAATGTGGCATGTAGCGGTGAAATGCATAGATATGCCATAGAACACCGATTGCGAAGGCAGCTAACTGGACATTTATTGACGCTGAGGCACGAAAGCGTGGGGAGCGAACAGG


rootedTree = ape::root(tree.unrooted, outgroup=new.outgroup, resolve.root=TRUE)

#plot(rootedTree) #don't plot rooted tree given the number of ASVs - at this stage plotting will not give informative results

#check that tree is, indeed, rooted
is.rooted(rootedTree) #TRUE! Yay, finally!

#make new phyloseq object with ROOTED tree
ROHR_06_obj_phylo_root <- merge_phyloseq(ROHR_06_obj_clean_updated_P, rootedTree)
ROHR_06_obj_phylo_root

saveRDS(ROHR_06_obj_phylo_root, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_with_rooted_phylogenetic_tree.rds")

#phyloseq-class experiment-level object
#otu_table()   OTU Table:         [ 6551 taxa and 341 samples ]
#sample_data() Sample Data:       [ 341 samples by 43 sample variables ]
#tax_table()   Taxonomy Table:    [ 6551 taxa by 6 taxonomic ranks ]
#phy_tree()    Phylogenetic Tree: [ 6551 tips and 6550 internal nodes ]

```

```{r read in rooted tree phyloseq object}
ROHR_06_obj_phylo_root <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_with_rooted_phylogenetic_tree.rds")
```

#Rarefy phyloseq with rooted tree
```{r normalizing data w/ tree - single sampling w/o replacement}
#check current reads/sample
reads_per_sample <- sample_sums(ROHR_06_obj_phylo_root)

#how many samples with fewer than 2500 reads?
print(length(reads_per_sample[reads_per_sample < 2500])) #62 samples (& 580 ASVs) will be excluded under this criterion

set.seed(9242)  # This will help in reproducing the filtering and normalisation. 

ROHR06.rar <- rarefy_even_depth(ROHR_06_obj_phylo_root, sample.size = 2500) 

```
```{r save updated rarefied data as new phyloseq object}
# Save updated rarefied data
saveRDS(ROHR06.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_2500_tree.rds")

#read rarefied data
ROHR06.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_2500_tree.rds")

#check how much data post-rarefaction
print(ROHR06.rar) # 6493 taxa, 279 samples
```

```{r fix sample IDs to not start with an integer}
#make sure to only run this once, otherwise you'll affix multiple names
#sample_names(ROHR06.rar) <- paste("RRR", gsub("-", "_", sample_names(ROHR06.rar)), sep = "_")

#sample_names(ROHR_06_obj_phylo_root) <- paste("RRR", gsub("-", "_", sample_names(ROHR_06_obj_phylo_root)), sep = "_")

```

#Hill Numbers - series of 3 differently-weighted alpha diversity metrics

For a good recap of Hill diversity, read over Roswell, Dushoff, and Winfree (2021), "A conceptual guide to measuring species diversity" - https://onlinelibrary.wiley.com/doi/10.1111/oik.07202

We're following the code written in Clever et al. (2022) and available on the Bocas Biome webpage: https://bocasbiome.github.io/wf4.html

#Hill numbers, without rarefying reads (only converting to relative abundance)

```{r add new metadata column to code in diet, echo = FALSE}

# define a list of species and their corresponding diet categories
species_diet <- list("Abudefduf concolor"="territorial herbivore",
                     "Abudefduf troschellii"="omnivore",
                     "Acanthurus xanthopterus"="roving herbivore",
                     "Cephalopholis colonus"="planktivore",
                     "Chaetodon humeralis"="invertivore",
                     "Cephalopholis panamensis"="carnivore",
                     "Epinephelus labriformis"="carnivore",
                     "Johnrandallia nigrirostris"="invertivore",
                     "Microspathodon dorsalis"="territorial herbivore",
                     "Prionurus laticlavius"="roving herbivore")

# define a function to assign diet categories based on input values
assign_diet <- function(value) {
  ifelse(value == "Abudefduf concolor", "territorial herbivore",
         ifelse(value == "Abudefduf troschelii", "omnivore",
                ifelse(value == "Acanthurus xanthopterus", "roving herbivore",
                       ifelse(value == "Cephalopholis colonus", "planktivore",
         ifelse(value == "Chaetodon humeralis", "invertivore",
                ifelse(value == "Cephalopholis panamensis", "carnivore",
                       ifelse(value == "Epinephelus labriformis", "carnivore",
         ifelse(value == "Johnrandallia nigrirostris", "invertivore",
                ifelse(value == "Microspathodon dorsalis", "territorial herbivore",
                        ifelse(value == "Prionurus laticlavius", "roving herbivore"))))))))))
}

# create a vector of diet categories based on the "diet_info" column
diet_info <- ROHR_06_obj_phylo_root@sam_data$species
diet_categories <- sapply(diet_info, assign_diet)

# add the new "diet" metadata column to the phyloseq object
ROHR_06_obj_phylo_root@sam_data$diet <- diet_categories

# view the updated metadata table
ROHR_06_obj_phylo_root@sam_data

#rarefied dataset
# create a vector of diet categories based on the "diet_info" column
diet_info <- ROHR06.rar@sam_data$species
diet_categories <- sapply(diet_info, assign_diet)

# add the new "diet" metadata column to the phyloseq object
ROHR06.rar@sam_data$diet <- diet_categories

# view the updated metadata table
ROHR06.rar@sam_data
```

```{r transform ASV table and check tree tips}
asv.whole <- t(otu_table(ROHR_06_obj_phylo_root))
tree.whole <- phy_tree(ROHR_06_obj_phylo_root)
identical(sort(rownames(asv.whole)), sort(tree.whole$tip.label))
```

```{r normalize reads to relative abundance}
asv.whole.norm <- microbiome::transform(asv.whole, transform = "compositional")
```

"The three forms of Hill diversity most commonly used by ecologists are species richness, and modifications of the traditional Shannon and Simpson indices. The key insight of Hill (1973) was that these three measures are special cases of the same general equation (Box 4). These three forms of Hill diversity â which we will refer to as species richness, HillâShannon diversity and HillâSimpson diversity â differ only in how they scale rarity (Box 5). Richness uses an arithmetic rarity scale, which gives high leverage to, and therefore remains very sensitive to, rare species; HillâSimpson diversity uses a reciprocal scale, which shifts leverage towards, and is thus dominated by common species; HillâShannon uses a logarithmic scale, and falls between the two." (Roswell et al, 2021)

Species Richness
```{r species richness - Hill}
hillq0 <- estimate_richness(ROHR_06_obj_phylo_root, measures = "Observed")
```

Hill-Shannon (Shannon Exponential)
```{r Hill-Shannon}
shannon.hill <- exp(vegan::diversity(t(asv.whole.norm), index = "shannon"))
shannon.whole.df <- as.data.frame(shannon.hill)
```

Hill-Simpson (Simpson Index)
```{r Hill-Simpson}
1/(1-(vegan::diversity(t(asv.whole.norm), index = "simpson")))
simpson.hill<-1/(1-(vegan::diversity(t(asv.whole.norm), index = "simpson")))
simpson.whole.df <- as.data.frame(simpson.hill)
```

Combine all elements into a single data frame

```{r merge data into phyloseq}
newDF.whole <- data.frame(hillq0, shannon.hill, simpson.hill)

newDF.whole <- data.frame(hillq0, shannon.hill, simpson.hill,
                          sample_data(ROHR_06_obj_phylo_root))

ROHR_06_obj_phylo_root.phylo.hill <- merge_phyloseq(otu_table(ROHR_06_obj_phylo_root),
                                sample_data(newDF.whole),
                                tax_table(ROHR_06_obj_phylo_root),
                                phy_tree(ROHR_06_obj_phylo_root))
ROHR_06_obj_phylo_root.phylo.hill

#save new phyloseq object with Hill numbers integrated
saveRDS(ROHR_06_obj_phylo_root.phylo.hill, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_phylo_root_obj_phylo_tree_Hill.rds")
```

```{r read phyloseq with hill numbers}
ROHR_06_obj_phylo_root.phylo.hill <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_phylo_root_obj_phylo_tree_Hill.rds")
```

#Hill numbers, reads rarefied to 2500 sequences/sample

```{r transform ASV table and check tree tips}
asv.whole <- t(otu_table(ROHR06.rar))
tree.whole <- phy_tree(ROHR06.rar)
identical(sort(rownames(asv.whole)), sort(tree.whole$tip.label))
```

```{r normalize reads to relative abundance}
asv.whole.norm <- microbiome::transform(asv.whole, transform = "compositional")
```

Species Richness
```{r species richness - Hill}
hillq0 <- estimate_richness(ROHR06.rar, measures = "Observed")
```

Hill-Shannon (Shannon Exponential)
```{r Hill-Shannon}
shannon.hill <- exp(vegan::diversity(t(asv.whole.norm), index = "shannon"))
shannon.whole.df <- as.data.frame(shannon.hill)
```

Hill-Simpson (Simpson Index)
```{r Hill-Simpson}
1/(1-(vegan::diversity(t(asv.whole.norm), index = "simpson")))
simpson.hill<-1/(1-(vegan::diversity(t(asv.whole.norm), index = "simpson")))
simpson.whole.df <- as.data.frame(simpson.hill)
```

Combine all elements into a single data frame

```{r merge data into phyloseq}
newDF.whole.r <- data.frame(hillq0, shannon.hill, simpson.hill)

newDF.whole.r <- data.frame(hillq0, shannon.hill, simpson.hill,
                          sample_data(ROHR06.rar))

ROHR_06_obj_phylo_root.phylo.hill.r <- merge_phyloseq(otu_table(ROHR06.rar),
                                sample_data(newDF.whole.r),
                                tax_table(ROHR06.rar),
                                phy_tree(ROHR06.rar))
ROHR_06_obj_phylo_root.phylo.hill.r
sample_data(ROHR_06_obj_phylo_root.phylo.hill.r)
#save new phyloseq object with Hill numbers integrated
saveRDS(ROHR_06_obj_phylo_root.phylo.hill.r, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_phylo_root_obj_phylo_tree_Hill_rar.rds")

#read if not starting from top
ROHR_06_obj_phylo_root.phylo.hill.r <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_phylo_root_obj_phylo_tree_Hill_rar.rds")
```

```{r calculate other a-div on rarefied dataset, observed richness, other forms of richness, diversity, evenness, dominance, rarity...}

setwd("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft")

ROHR_06_obj_phylo_root.div.r <- alpha(ROHR_06_obj_phylo_root.phylo.hill.r, index = "all")

datatable(ROHR_06_obj_phylo_root.div.r)

write.csv(ROHR_06_obj_phylo_root.div.r, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR_06_obj_phylo_root_alpha_div_metrics_rarefied2500.csv")

#if not starting from beginning: read csv file:
ROHR_06_obj_phylo_root.div.r <- read.csv("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR_06_obj_phylo_root_alpha_div_metrics_rarefied2500.csv")
```

```{r save this table to word doc}

# Create a flextable object from your data
ft <- flextable(ROHR06.div.r)

# Rename columns (replace with actual column names as needed)
colnames(ROHR06.div.r) <- c("Sample ID", "Observed Richness", "Chao1", "Inverse Simpson Div", "Gini Simpson Div","Shannon Div", "Fisher Div", "Coverage Div", "Camargo Even", "Pielou Even", "Simpson Even", "Evar Even", "Bulla Even", "Dom DBP", "Dom DMN", "Dom Absolute", "Dom Relative", "Dom Simpson", "Dom Core Abund", "Dom Gini", "Rar log Modulo Skewness", "Rar Low Abund", "Rar Rare Abund")

# Define a function to split data by columns (keeping "Sample ID")
split_columns <- function(data, columns_per_chunk) {
  col_indices <- seq(2, ncol(data))  # Skip the "Sample ID" column (index 1)
  chunks <- split(col_indices, ceiling(seq_along(col_indices) / columns_per_chunk))
  
  # Add "Sample ID" to each chunk
  lapply(chunks, function(cols) data[, c(1, cols)])
}

# Split data into chunks of columns (excluding "Sample ID" as the first column)
column_chunks <- split_columns(ROHR06.div.r, columns_per_chunk = 11)

# Create a new Word document
doc <- read_docx()

# Loop through column chunks and add each as a separate table with "Sample ID" in bold
for (i in seq_along(column_chunks)) {
  chunk <- column_chunks[[i]]
  
  # Create a flextable for each column chunk with "Sample ID" in bold
  ft_chunk <- flextable(chunk) %>%
    theme_vanilla() %>%
    align(align = "center", part = "all") %>%
    set_table_properties(layout = "autofit", width = 1) %>%
    bold(j = "Sample ID", bold = TRUE) %>%
    bold(i = 1, bold = TRUE, part = "header") %>%
    fontsize(size = 6, part = "all") %>%
    bg(i = 1, bg = "#D3D3D3", part = "header") 

  # Add the table chunk to the document
  doc <- body_add_flextable(doc, value = ft_chunk)
  doc <- body_add_par(doc, "") # Add space between tables if needed
}

# Save the document
print(doc, target = "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/ROHR06_skin_alpha_div_metrics_rarefied2500.docx")

```

In Clever et al. (2022) workflow, they are looking at inner and outer zones of a reef, and comparing inner, outer, and outer disturbed zones. In our case, we are looking at two different gulfs/regions (with multiple sites w/in a gulf) which experience different degrees of seasonality (Las Perlas = seasonal upwelling; Coiba = no seasonal upwelling), across 2 sampling seasons. It is thus of interest to see if:
(a) fish skin microbiome alpha diversity changes across gulfs (Coiba vs. Las Perlas)
(b) **within a gulf across seasons (we would expect seasonal effects to be seen in Las Perlas, with marked changes beteen upwelling & non-upwelling sampling, given the drastic, rapid environmental changes that take place during upwelling; on the other hand, stochastic changes in the microbiome - which is known to be highly dynamic, could result in between-timepoint differences being observed in Coiba populations as well).
(c) between species of fish (we will do species-by-species analyses in a separate script)

To test (a) and (b), it makes sense to divide the dataset into region-season groups (e.g., Las Perlas upwelling), yielding 4 distinct groups we can compare. 

## Unrarefied Hill numbers - region, season, and species + plots

```{r subset dataset into region and season}
#Coiba
Coiba_up <- subset_samples(ROHR_06_obj_phylo_root.phylo.hill, region_season == "Coiba upwelling")
Coiba_up <- prune_taxa(taxa_sums(Coiba_up) > 0, Coiba_up) #4495 taxa and 86 samples
Coiba_up
Coiba_N <- subset_samples(ROHR_06_obj_phylo_root.phylo.hill, region_season == "Coiba non-upwelling")
Coiba_N <- prune_taxa(taxa_sums(Coiba_N) > 0, Coiba_N) #4625 taxa and 88 samples
Coiba_N
#Las Perlas
LP_up <- subset_samples(ROHR_06_obj_phylo_root.phylo.hill, region_season == "Las Perlas upwelling")
LP_up <- prune_taxa(taxa_sums(LP_up) > 0, LP_up) #4593 taxa and 83 samples
LP_up
LP_N <- subset_samples(ROHR_06_obj_phylo_root.phylo.hill, region_season == "Las Perlas non-upwelling")
LP_N <- prune_taxa(taxa_sums(LP_N) > 0, LP_N) #4496 taxa and 84 samples
LP_N

#From the summary of the phyloseq objects, we can see that the number of taxa across all samples within a given season & region is relatively consistent (ranging from 4495 taxa to 4625 taxa across 83 to 88 samples)
```
```{r transpose ASV tables to ASVs as rows}
asv.Coiba_up <- t(otu_table(Coiba_up))
asv.Coiba_N <- t(otu_table(Coiba_N))
asv.LP_up <- t(otu_table(LP_up))
asv.LP_N <- t(otu_table(LP_N))
```

```{r transform counts to relative abundance}
asv.Coiba_up.norm <- microbiome::transform(asv.Coiba_up, transform = "compositional")
asv.Coiba_N.norm <- microbiome::transform(asv.Coiba_N, transform = "compositional")
asv.LP_up.norm <- microbiome::transform(asv.LP_up,
                                             transform = "compositional")
asv.LP_N.norm <- microbiome::transform(asv.LP_N,
                                             transform = "compositional")
```

"Now we can test alpha diversity per zone based on Hill numbers at different diversity levels (q = 0 (observed), q = 1 (shannon exponential), q = 2 (multiplicative simpson)).

Note: for Hill numbers, alpha diversity per zone cannot be obtained as a mean across samples. The calculation is different as performed by function alpha_div()." - Clever et al. (2022)

What do these "different diversity levels" do for us?
- since no single metric fully captures diversity, using a trio of metrics that complement each other can help us best understand the complex microbial diversity in our samples. Observed: species richness - "Species richness emphasizes (provides higher leverage to) rare species" (Roswell, 2022), Shannon exponential: "HillâShannon diversity emphasizes neither rare nor common species.", multiplicative simpson: "HillâSimpson diversity emphasizes (provides higher leverage to) the common species."

```{r test a-div per season & region}
#Coiba upwelling
alpha_div(countable = asv.Coiba_up.norm, qvalue = 0)
alpha_div(countable = asv.Coiba_up.norm, qvalue = 1)
alpha_div(countable = asv.Coiba_up.norm, qvalue = 2)

#Coiba non-upwelling
alpha_div(countable = asv.Coiba_N.norm, qvalue = 0)
alpha_div(countable = asv.Coiba_N.norm, qvalue = 1)
alpha_div(countable = asv.Coiba_N.norm, qvalue = 2)

#Las Perlas upwelling
alpha_div(countable = asv.LP_up.norm, qvalue = 0)
alpha_div(countable = asv.LP_up.norm, qvalue = 1)
alpha_div(countable = asv.LP_up.norm, qvalue = 2)

#Las Perlas non-upwelling
alpha_div(countable = asv.LP_N.norm, qvalue = 0)
alpha_div(countable = asv.LP_N.norm, qvalue = 1)
alpha_div(countable = asv.LP_N.norm, qvalue = 2)
```

Test significance of differences between a-div

```{r run Kruskal-Wallis test and post-hoc Dunn}

#full dataset
temp_table <- data.frame(sample_data(ROHR_06_obj_phylo_root))
temp_table[, c(7:10, 13:14, 24:42)] <- list(NULL) #adjust these numbers to define which metadata columns to EXCLUDE
temp_table <- temp_table %>% tibble::rownames_to_column("Sample")
temp_table <- temp_table %>% dplyr::rename(Group = region_season)
temp_table$Group <- as.character(temp_table$Group)
temp_table$species <- as.character(temp_table$species)

#comparisons between seasons & regions
hill_hierarchy <- temp_table[, c(1, 19)] #adjust these numbers for the comparisons between groups you want to make - in this case RRR number (renamed to "Sample") & Group (region_season)

#comparisons between species
hill_hierarchy2 <- temp_table[, c(1, 8)] #RRR number (Sample) & species

#comparisons between sites (within a region)
hill_hierarchy3 <- temp_table[, c(1, 5)] #RRR number & site

#comparisons between trophic groups
hill_hierarchy4 <- temp_table[, c(1, 20)] #RRR number (Sample) & diet
```

```{r significance test season region}
divtestresult.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy, posthoc = TRUE)
divtestresult.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy, posthoc = TRUE)
divtestresult.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy, posthoc = TRUE)

divtestresult.q0 #no significant differences between seasons and regions in observed a-div --> BUT, while not significantly different, Las Perlas samples during upwelling are more different from Coiba samples (p = 0.25 (N) and 0.17 (up)) and Las Perlas non-upwelling samples (p = 0.12), while Coiba samples and Las Perlas non-upwelling samples are all virtually identical under this diversity metric (p = 0.97 to 1). (based on p.adj values)

#normality.pvalue = 7.448866e-16
#homogeneity.pvalue = 0.02430449
#groups = 4
#"Kruskal-Wallis Test"
# chi.squared          df     p.value 
#  5.6283808   3.0000000   0.1311585 

divtestresult.q1 #NSD in shannon exponential across all regions & seasons

#normality.pvalue = 1.33244e-15
#homogeneity.pvalue = 0.04392937
#groups = 4
#"Kruskal-Wallis Test"
# chi.squared          df     p.value 
#  0.1165579   3.0000000   0.9897789 

divtestresult.q2 #NSD in multiplicative simpson across all regions & seasons

#normality.pvalue = 1.745303e-16
#homogeneity.pvalue = 0.09204465
#groups = 4
#"Kruskal-Wallis Test"
# chi.squared          df     p.value 
#  0.5879204   3.0000000   0.8991921 

```

```{r significance test between species}
divtestresult2.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy2, posthoc = TRUE)
divtestresult2.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy2, posthoc = TRUE)
divtestresult2.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy2, posthoc = TRUE)

divtestresult2.q0
divtestresult2.q1
divtestresult2.q2

#q0 - observed richness
#normality.pvalue = 7.448866e-16
#homogeneity.pvalue = 6.257145e-12
#groups = 10
#method = "Kruskal-Wallis Test"
#test
#  chi.squared           df      p.value 
# 8.534729e+01 9.000000e+00 1.390635e-14 

#q1 - shannon exponential (weighs ASVs by frequency)
#normality.pvalue = 1.33244e-15
#homogeneity.pvalue = 1.591371e-05
#groups = 10
#method = "Kruskal-Wallis Test"
#test
#  chi.squared           df      p.value 
# 6.239570e+01 9.000000e+00 4.619374e-10 

#q2 - simpson
#normality.pvalue = 1.745303e-16
#homogeneity.pvalue = 0.01022253
#groups = 10
#method = "Kruskal-Wallis Test"
#test
#  chi.squared           df      p.value 
# 4.385421e+01 9.000000e+00 1.501471e-06 

```
Looking at observed richness (q0) between species, we can see that the individual richness values vary quite a bit both within individuals of a same species, and across fish species. Non-parametric Kruskal-Wallis tests show significant differences between many species (p < 0.05; A. tro:C. pana, A. con:C. hum, A. xan:C. hum, C. col:C. hum, C. pana:C. hum, ...), not-quite-significant differences between others.

Looking at Shannon exponential (q1) between species, we also see variation in values across samples and significant differences between certain species, although fewer than with observed richness - in both cases, the largest differences are between invertivores (C. humeralis & J. nigrirostris) and carnivores (C. panamensis, E. labriformis).

Looking at Simpson multiplicative inverse (q2) between species, we also see variation in values across samples and significant differences between certain species, again, mostly relative to the invertivorous butterflyfishes (C. humeralis & J. nigrirostris). 

**Note that for all three of these metrics, the data is neither normal nor homogeneous 

```{r significance test site}
divtestresult3.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy3, posthoc = TRUE)
divtestresult3.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy3, posthoc = TRUE)
divtestresult3.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy3, posthoc = TRUE)
divtestresult3.q0
divtestresult3.q1
divtestresult3.q2
```
No significant differences in alpha diversity between sites. (not normal, but (barely) homogeneous (p = 0.076))

```{r significance test trophic group}
divtestresult4.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy4, posthoc = TRUE)
divtestresult4.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy4, posthoc = TRUE)
divtestresult4.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy4, posthoc = TRUE)
divtestresult4.q0
divtestresult4.q1
divtestresult4.q2
```

```{r break species data into season & region - new column}
ROHR_06_obj_phylo_root.phylo.hill <- merge_treatments(ROHR_06_obj_phylo_root.phylo.hill, c("species", "region", "season"))

# Save new metadata
saveRDS(ROHR_06_obj_phylo_root.phylo.hill, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_unrarefied_Hill_merged.rds")
```

```{r adjust sample data on hill phyloseq}

data.hill.whole_n <- data.frame(sample_data(ROHR_06_obj_phylo_root.phylo.hill)) %>%
  pivot_longer(cols = 1:3, names_to = "Index",  values_to = "Diversity")
data.hill.whole_n$Fraction_delailed <- NULL
data.hill.whole_n <- data.hill.whole_n[, c(1:4, 6, 11:12, 43:47)] ###Adjust #s to match metadata columns to keep
data.hill.whole_n <- dplyr::mutate_if(data.hill.whole_n, is.factor, as.character)
data.hill.whole_n$Index <- stringr::str_replace(
  data.hill.whole_n$Index, "Observed", "Observed (q=0)")
data.hill.whole_n$Index <- stringr::str_replace(
  data.hill.whole_n$Index, "shannon.hill", "Shannon exponential (q=1)")
data.hill.whole_n$Index <- stringr::str_replace(
  data.hill.whole_n$Index, "simpson.hill", "Simpson multi. inverse (q=2)")
data.hill.whole_n <- data.hill.whole_n[order(data.hill.whole_n$Index),]

saveRDS(data.hill.whole_n, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_hill_datatable.rds")
```

```{r read data with hill num}
data.hill.whole <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_hill_datatable.rds")
```

```{r color codes - don't run}

#from ROHR_06_Swab_Figures:
# set colors for seasons - "#660066" = non-upwelling, "#336633" = upwelling BUT have colors assigned for different sites w/in each gulf for temperature figure
c2 <- c("#660066", "#336633") #will need 2 more colors to show Coiba & Las Perlas separately
#temp figure:
site_cols <- c( "#FFAD65", "#FF6E00", "#662700", "#99FFFF", "#009999", "#006666",
                "#00CCCC", "#00FFFF", "#993D00", "#FF8E33", "#CC5500", "#FFCA99")

sites <- c("Afuera", "Damas", "Uvas", "Contadora", "Saboga", "MogoMogo", #water sampling
           "Pacheca", "Chapera", "Machete", "Cocos", "Frijol", "Granito" ) #fish only

##colors for individual fish species
col_AconcS <- "aquamarine4"
col_AtroS <- "darkgoldenrod1"
col_AxanS <- "chartreuse3"
col_CcolS <- "darkmagenta"
col_ChumS <- "deepskyblue2"
col_CpanS <- "chocolate4" 
col_ElabS <- "chocolate2"
col_JnigS <- "darkblue" 
col_MdorS <- "turquoise"
col_PlatS <- "darkgreen"
col_WaterS <- "#BE93D4"
```

```{r summary figure a-div Hill numbers}
level_order <- c('Coiba upwelling', 'Coiba non-upwelling', 'Las Perlas upwelling', 'Las Perlas non-upwelling')
level_order2 <- c('Observed', 'shannon.hill', 'simpson.hill')
level_order3 <- c('Abudefduf concolor', 'Microspathodon dorsalis', 'Acanthurus xanthopterus', 
                  'Prionurus laticlavius', 'Johnrandallia nigrirostris', 'Chaetodon humeralis',
                  'Cephalopholis colonus', 'Abudefduf troschelii', 'Cephalopholis panamensis', 
                  'Epinephelus labriformis')
level_order3a <- c('Acon', 'Mdor', 'Axan', 'Plat', 'Jnig', 'Chum', 'Ccol', 'Atro', 'Cpan', 'Elab')
level_order4 <- c('Coiba', 'Las Perlas')

#season & region palette
disp.pal2 <- c("#662700", "#FF6E00", "#006666", "#009999")
#individual fish palette
disp.pal3 <- c( "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2")
disp.pal4 <- c( "darkslategray", "darkslategray4", "chartreuse4", "#1F3E20", "midnightblue",
                "deepskyblue4", "mediumpurple4", "goldenrod4", "#4E342F", "darkorange3")
#MAY NEED 3RD PALETTE FOR FISH + SEASON

```

```{r define levels - gulf & season}
data.hill.whole$region_season <- factor(data.hill.whole$region_season,
                               levels = c('Coiba upwelling', 'Coiba non-upwelling', 'Las Perlas upwelling', 
                                          'Las Perlas non-upwelling'))
```

```{r define levels - gulf only}
data.hill.whole$region <- factor(data.hill.whole$region,
                               levels = c('Coiba', 'Las Perlas'))
```

```{r define levels - species}
data.hill.whole$species <- factor(data.hill.whole$species,
                               levels = c('Abudefduf concolor', 'Microspathodon dorsalis', 
                                          'Acanthurus xanthopterus', 'Prionurus laticlavius', 
                                          'Johnrandallia nigrirostris', 'Chaetodon humeralis',
                                          'Cephalopholis colonus', 'Abudefduf troschelii', 
                                          'Cephalopholis panamensis', 'Epinephelus labriformis'))
```

```{r define levels - species_region_season}

data.hill.whole$species_region_season <- factor(data.hill.whole$species_region_season,
                               levels = c('Abudefduf concolor Coiba upwelling', 'Abudefduf concolor Coiba non-upwelling', 'Abudefduf concolor Las Perlas upwelling', 'Abudefduf concolor Las Perlas non-upwelling',
                                          'Microspathodon dorsalis Coiba upwelling', 'Microspathodon dorsalis Coiba non-upwelling', 'Microspathodon dorsalis Las Perlas upwelling', 'Microspathodon dorsalis Las Perlas non-upwelling', 
                                          'Acanthurus xanthopterus Coiba upwelling', 'Acanthurus xanthopterus Coiba non-upwelling', 'Acanthurus xanthopterus Las Perlas upwelling', 'Acanthurus xanthopterus Las Perlas non-upwelling',
                                          'Prionurus laticlavius Coiba upwelling', 'Prionurus laticlavius Coiba non-upwelling', 'Prionurus laticlavius Las Perlas upwelling', 'Prionurus laticlavius Las Perlas non-upwelling', 
                                          'Johnrandallia nigrirostris Coiba upwelling', 'Johnrandallia nigrirostris Coiba non-upwelling', 'Johnrandallia nigrirostris Las Perlas upwelling', 'Johnrandallia nigrirostris Las Perlas non-upwelling', 
                                          'Chaetodon humeralis Coiba upwelling',  'Chaetodon humeralis Coiba non-upwelling',  'Chaetodon humeralis Las Perlas upwelling',  'Chaetodon humeralis Las Perlas non-upwelling',
                                          'Cephalopholis colonus Coiba upwelling', 'Cephalopholis colonus Coiba non-upwelling', 'Cephalopholis colonus Las Perlas upwelling', 'Cephalopholis colonus Las Perlas non-upwelling', 
                                          'Abudefduf troschelii Coiba upwelling',  'Abudefduf troschelii Coiba non-upwelling',  'Abudefduf troschelii Las Perlas upwelling',  'Abudefduf troschelii Las Perlas non-upwelling', 
                                          'Cephalopholis panamensis Coiba upwelling', 'Cephalopholis panamensis Coiba non-upwelling', 'Cephalopholis panamensis Las Perlas upwelling', 'Cephalopholis panamensis Las Perlas non-upwelling', 
                                          'Epinephelus labriformis Coiba upwelling', 'Epinephelus labriformis Coiba non-upwelling', 'Epinephelus labriformis Las Perlas upwelling', 'Epinephelus labriformis Las Perlas non-upwelling'))
```

```{r new list for species region season}
level_order5 <- c('Abudefduf concolor Coiba upwelling', 'Abudefduf concolor Coiba non-upwelling', 'Abudefduf concolor Las Perlas upwelling', 'Abudefduf concolor Las Perlas non-upwelling',
                                          'Microspathodon dorsalis Coiba upwelling', 'Microspathodon dorsalis Coiba non-upwelling', 'Microspathodon dorsalis Las Perlas upwelling', 'Microspathodon dorsalis Las Perlas non-upwelling', 
                                          'Acanthurus xanthopterus Coiba upwelling', 'Acanthurus xanthopterus Coiba non-upwelling', 'Acanthurus xanthopterus Las Perlas upwelling', 'Acanthurus xanthopterus Las Perlas non-upwelling',
                                          'Prionurus laticlavius Coiba upwelling', 'Prionurus laticlavius Coiba non-upwelling', 'Prionurus laticlavius Las Perlas upwelling', 'Prionurus laticlavius Las Perlas non-upwelling', 
                                          'Johnrandallia nigrirostris Coiba upwelling', 'Johnrandallia nigrirostris Coiba non-upwelling', 'Johnrandallia nigrirostris Las Perlas upwelling', 'Johnrandallia nigrirostris Las Perlas non-upwelling', 
                                          'Chaetodon humeralis Coiba upwelling',  'Chaetodon humeralis Coiba non-upwelling',  'Chaetodon humeralis Las Perlas upwelling',  'Chaetodon humeralis Las Perlas non-upwelling',
                                          'Cephalopholis colonus Coiba upwelling', 'Cephalopholis colonus Coiba non-upwelling', 'Cephalopholis colonus Las Perlas upwelling', 'Cephalopholis colonus Las Perlas non-upwelling', 
                                          'Abudefduf troschelii Coiba upwelling',  'Abudefduf troschelii Coiba non-upwelling',  'Abudefduf troschelii Las Perlas upwelling',  'Abudefduf troschelii Las Perlas non-upwelling', 
                                          'Cephalopholis panamensis Coiba upwelling', 'Cephalopholis panamensis Coiba non-upwelling', 'Cephalopholis panamensis Las Perlas upwelling', 'Cephalopholis panamensis Las Perlas non-upwelling', 
                                          'Epinephelus labriformis Coiba upwelling', 'Epinephelus labriformis Coiba non-upwelling', 'Epinephelus labriformis Las Perlas upwelling', 'Epinephelus labriformis Las Perlas non-upwelling')

disp.pal5 <- c( "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745")

disp.pal6 <- c( "aquamarine4", "aquamarine4","aquamarine4","aquamarine4","turquoise","turquoise","turquoise","turquoise","chartreuse3","chartreuse3","chartreuse3","chartreuse3", "darkgreen","darkgreen", "darkgreen", "darkgreen", "darkblue","darkblue","darkblue","darkblue", "deepskyblue2","deepskyblue2", "deepskyblue2", "deepskyblue2",   "darkmagenta","darkmagenta", "darkmagenta", "darkmagenta", "darkgoldenrod1", "darkgoldenrod1","darkgoldenrod1", "darkgoldenrod1", "chocolate4","chocolate4","chocolate4","chocolate4", "chocolate2","chocolate2", "chocolate2", "chocolate2")

disp.pal7 <- c("#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999")

#labels = c('Acon C-U', 'Acon C-N', 'Acon LP-U', 'Acon LP-N', 'Mdor C-U','Mdor C-N','Mdor LP-U','Mdor LP-N', 'Axan C-U', 'Axan C-N','Axan LP-U','Axan LP-N','Plat C-U', 'Plat C-N','Plat LP-U','Plat LP-N','Jnig C-U', 'Jnig C-N','Jnig LP-U','Jnig LP-N','Chum C-U', 'Chum C-N','Chum LP-U','Chum LP-N', 'Ccol C-U', 'Ccol C-N','Ccol LP-U','Ccol LP-N','Atro C-U', 'Atro C-N', 'Atro LP-U', 'Atro LP-N', 'Cpan C-U', 'Cpan C-N', 'Cpan LP-U', 'Cpan LP-N', 'Elab C-U', 'Elab C-N', 'Elab LP-U', 'Elab LP-N')

disp.pal8 <- c( "aquamarine4", "aquamarine2","aquamarine4", "aquamarine2","darkgreen", "#9CCC65","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "darkblue","#5B6CB0", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate",
                "aquamarine4", "aquamarine2","aquamarine4", "aquamarine2","darkgreen", "#9CCC65","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "darkblue","#5B6CB0", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate")

disp.pal9 <- c("#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999")

disp.pal10 <- c( "aquamarine4", "aquamarine2","turquoise","turquoise1","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","chocolate4","chocolate", "chocolate2","#F9A745", "#662700", "#FF6E00",
                "aquamarine4", "aquamarine2","turquoise","turquoise1","chartreuse3","chartreuse","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate2", "#F9A745", "#662700", "#FF6E00")

disp.pal11 <- c( "aquamarine4", "aquamarine2","turquoise","turquoise1","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","chocolate4","chocolate", "#006666", "#009999",
                "aquamarine4", "aquamarine2","turquoise","turquoise1","chartreuse3","chartreuse","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","#006666", "#009999")
```

```{r define graph theme}
font_theme = theme(
        axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        axis.title.y = element_text(size = 16),
        axis.text.y = element_text(size = 10))
```

```{r first a-div figure - season region}

fig01a <-
        ggplot(data = data.hill.whole, aes(x = factor(region, level = level_order4),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=region_season, colour=region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal2)+
        scale_fill_manual(values=disp.pal2)+
        ggtitle("Alpha Diversity Hill Numbers", subtitle="Whole microbiome")+
        theme(plot.title=element_text(size=18)) +
        theme(plot.subtitle=element_text(size=16)) +
        font_theme+
        stat_summary(aes(fill=region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999"),
                     show.legend = FALSE) +
        scale_x_discrete(name="Region") + scale_y_continuous(name="Hill diversity") +
        theme(plot.title = element_text(family=NA, face="bold", size=16)) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              linewidth =1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12),
              legend.title = element_text(size = 14))
fig01a
```
```{r first a-div figure - season region}

fig01a.1 <-
        ggplot(data = data.hill.whole, aes(x = factor(region, level = level_order4),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=region_season), geom ='errorbar') + #alpha=0.3, if want to increase transparency
        geom_boxplot(aes(fill=region_season, colour=region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal2) +
        scale_fill_manual(values=disp.pal2) +
  labs(fill = "Region and Season", colour = "Region and Season") +
        font_theme+
        stat_summary(aes(fill=region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999"),
                     show.legend = FALSE) +
        scale_x_discrete(name="Region") + scale_y_continuous(name="Hill diversity") +
        theme(plot.title = element_text(family=NA, face="bold", size=16)) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              linewidth =1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12),
              legend.title = element_text(size = 14))
fig01a.1
```

```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Alpha Diversity Unrarefied Obs Shan Simp.pdf", plot = fig01a.1, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Alpha Diversity Unrarefied Obs Shan Simp.png", plot = fig01a.1, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

From this figure, we can see that there is high heterogeneity in a-diversity across samples within each gulf and sampling season. Observed numbers range from near zero to almost one thousand (Las Perlas, upwelling), with medians around 150-200 and means consistently higher (~ 200-250; likely due to several high-diversity outliers in each group bringing the average up). Similar patterns are seen for Shannon and Simpson (albeit scaled differently). Although the large variation makes it hard to distinguish significant patterns, the range in a-diversity observed in Las Perlas during upwelling appears to generally be greater than the other sites and seasons, with a higher mean and median for Observed and Shannon diversity (while these differences are not significant, as can be seen in the Kruskall-Wallis tests run above, they suggest trends that may become significant with a larger sample size and/or when filtering out more samples with low total read counts that are skewing the a-diversity values towards zero).

```{r check sequences per sample in dataset}
reads_per_sample2 <- sample_sums(ROHR_06_obj_phylo_root.phylo.hill)

#how many samples with fewer than 2500 reads?
print(length(reads_per_sample2[reads_per_sample2 < 2500])) #62 samples with fewer than 2500 reads

#visualize distribution of read counts
hist(sample_sums(ROHR_06_obj_phylo_root.phylo.hill), main="Histogram: Read Counts", xlab="Total Reads", 
     border="blue", col="green", las=1, breaks=50)

# Remove samples with fewer than 2500 reads from phyloseq object
ROHR_06.phylo.hill.2500 <- subset_samples(ROHR_06.phylo.hill, sample_sums(ROHR_06.phylo.hill) > 2500)

#revove ASVs not present in any sample (occurs after having removed samples)
ROHR_06.phylo.hill.2500 <- prune_taxa(taxa_sums(ROHR_06.phylo.hill.2500) > 0, ROHR_06.phylo.hill.2500)
ROHR_06.phylo.hill.2500 # down to 6547 taxa in 279 samples

#visualize distribution of read counts
hist(sample_sums(ROHR_06.phylo.hill.2500), main="Histogram: Read Counts", xlab="Total Reads", 
     border="blue", col="green", las=1, breaks=50)
```
```{r adjust sample data on hill phyloseq}

data.hill.whole_n2 <- data.frame(sample_data(ROHR_06.phylo.hill.2500)) %>%
  pivot_longer(cols = 1:3, names_to = "Index",  values_to = "Diversity")
data.hill.whole_n2$Fraction_delailed <- NULL
data.hill.whole_n2 <- data.hill.whole_n2[, c(1:4, 6, 11:12, 43:45)] ###Adjust #s to match metadata columns to keep
data.hill.whole_n2 <- dplyr::mutate_if(data.hill.whole_n2, is.factor, as.character)
data.hill.whole_n2$Index <- stringr::str_replace(
  data.hill.whole_n2$Index, "Observed", "Observed (q=0)")
data.hill.whole_n2$Index <- stringr::str_replace(
  data.hill.whole_n2$Index, "shannon.hill", "Shannon exponential (q=1)")
data.hill.whole_n2$Index <- stringr::str_replace(
  data.hill.whole_n2$Index, "simpson.hill", "Simpson multi. inverse (q=2)")
data.hill.whole_n2 <- data.hill.whole_n2[order(data.hill.whole_n2$Index),]

saveRDS(data.hill.whole_n2, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_hill_datatable_2.rds")
```

```{r a-div figure - season region w/ trimmed data}

fig01b <-
        ggplot(data = data.hill.whole_n2, aes(x = factor(region, level = level_order4),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=region_season, colour=region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal2)+
        scale_fill_manual(values=disp.pal2)+
        ggtitle("Alpha Diversity Hill Numbers", subtitle="Whole microbiome - trimmed")+
        theme(plot.title=element_text(size=18)) +
        theme(plot.subtitle=element_text(size=16)) +
        font_theme+
        stat_summary(aes(fill=region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999"),
                     show.legend = FALSE) +
        scale_x_discrete(name="Region") + scale_y_continuous(name="Hill diversity") +
        theme(plot.title = element_text(family=NA, face="bold", size=16)) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12),
              legend.title = element_text(size = 14))
fig01b
```

```{r second a-div figure - species by species}
#fix labels at bottom (3-letter fish codes?)
#add seasonal comparison bars = 2 bars for each species
fig02 <-
        ggplot(data = data.hill.whole, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species, colour=species), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal4, name = "Species")+
        scale_fill_manual(values=disp.pal3, name = "Species")+
        theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
  labs(fill = "Species", colour = "Species") +
        font_theme +
        stat_summary(aes(fill=species), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c( "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2",
                "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2",
                "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2"),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A.con', 'M.dor', 'A.xan', 
                                                              'P.lat', 'J.nig', 'C.hum', 'C.col', 'A.tro', 
                                                              'C.pan', 'E.lab')) + 
        scale_y_continuous(name="Hill diversity") +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14))
fig02
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill Numbers Species Unrarefied.pdf", plot = fig02, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Hill Numbers Species Unrarefied.png", plot = fig02, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

```{r third a-div figure - species by species region season}
#fix labels at bottom (3-letter fish codes?)
#add seasonal comparison bars = 2 bars for each species
fig02.1 <- ggplot(data = data.hill.whole, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_region_season, colour=species_region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal6, name = "Species")+
        scale_fill_manual(values=disp.pal5, name = "Species")+
   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745",
                               "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745",
                               "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(name="Hill diversity") +
  rotate_x_text(60) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig02.1
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill Numbers Species Region Season Unrarefied.pdf", plot = fig02.1, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Hill Numbers Species Region Season Unrarefied.png", plot = fig02.1, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

To make this figure less busy, we'll only show one Hill # (e.g., q0) - to do this, we'll need to subset the pivot we did with the data to create "data.hill.whole"

```{r subset pivot table to only q0 (Observed)}
data.hill.observed <- data.hill.whole %>%
  filter(Index == "Observed (q=0)")
```

```{r fourth a-div figure - species by species region season - q0 only}
#fix labels at bottom (3-letter fish codes?)
#add seasonal comparison bars = 2 bars for each species
fig02.2 <- ggplot(data = data.hill.observed, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_region_season, colour=species_region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal9, name = "Species")+ #6
        scale_fill_manual(values=disp.pal5, name = "Species")+ #5
   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(name="Hill diversity") +
  rotate_x_text(60) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig02.2
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill q0 Species Region Season Unrarefied V2.pdf", plot = fig02.2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Hill q0 Species Region Season Unrarefied V2.png", plot = fig02.1, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

###In progress 12/3/24 - stacked panels 2 gulfs

Further subset by gulf

```{r create season & species column}

# Merge 'species' and 'season' into a new column
data.hill.observed$species_season <- paste(data.hill.observed$species, data.hill.observed$season, sep = " ")

# View the updated data frame
print(data.hill.observed)
```

```{r subset pivot table to only Gulf of ChiriquÃ­}
data.hill.observed.chiriqui <- data.hill.observed %>%
  filter(region == "Coiba")
```

```{r subset pivot table to only Gulf of ChiriquÃ­}
data.hill.observed.perlas <- data.hill.observed %>%
  filter(region == "Las Perlas")
```

```{r define levels - species_season - Coiba}
data.hill.observed.chiriqui$species_season <- factor(data.hill.observed.chiriqui$species_season,
                               levels = c('Abudefduf concolor non-upwelling', 'Abudefduf concolor upwelling', 
                                          'Microspathodon dorsalis non-upwelling', 'Microspathodon dorsalis upwelling', 
                                         'Acanthurus xanthopterus non-upwelling',  'Acanthurus xanthopterus upwelling', 
                                          'Prionurus laticlavius non-upwelling', 'Prionurus laticlavius upwelling', 
                                          'Johnrandallia nigrirostris non-upwelling','Johnrandallia nigrirostris upwelling', 
                                          'Chaetodon humeralis non-upwelling', 'Chaetodon humeralis upwelling', 
                                          'Cephalopholis colonus non-upwelling','Cephalopholis colonus upwelling', 
                                          'Abudefduf troschelii non-upwelling','Abudefduf troschelii upwelling',  
                                           'Cephalopholis panamensis non-upwelling', 'Cephalopholis panamensis upwelling',
                                          'Epinephelus labriformis non-upwelling', 'Epinephelus labriformis upwelling' ))

data.hill.observed.perlas$species_season <- factor(data.hill.observed.perlas$species_season,
                               levels = c('Abudefduf concolor non-upwelling', 'Abudefduf concolor upwelling', 
                                          'Microspathodon dorsalis non-upwelling', 'Microspathodon dorsalis upwelling', 
                                          'Acanthurus xanthopterus non-upwelling', 'Acanthurus xanthopterus upwelling', 
                                          'Prionurus laticlavius non-upwelling', 'Prionurus laticlavius upwelling', 
                                          'Johnrandallia nigrirostris non-upwelling','Johnrandallia nigrirostris upwelling', 
                                           'Chaetodon humeralis non-upwelling','Chaetodon humeralis upwelling', 
                                          'Cephalopholis colonus non-upwelling','Cephalopholis colonus upwelling', 
                                          'Abudefduf troschelii non-upwelling','Abudefduf troschelii upwelling',
                                          'Cephalopholis panamensis non-upwelling', 'Cephalopholis panamensis upwelling', 
                                          'Epinephelus labriformis non-upwelling','Epinephelus labriformis upwelling'))
```

```{r new list for species season}
level_order8 <- c('Abudefduf concolor non-upwelling', 'Abudefduf concolor upwelling', 
                                          'Microspathodon dorsalis non-upwelling', 'Microspathodon dorsalis upwelling', 
                                          'Acanthurus xanthopterus non-upwelling', 'Acanthurus xanthopterus upwelling', 
                                          'Prionurus laticlavius non-upwelling', 'Prionurus laticlavius upwelling', 
                                          'Johnrandallia nigrirostris non-upwelling', 'Johnrandallia nigrirostris upwelling', 
                                          'Chaetodon humeralis non-upwelling',  'Chaetodon humeralis upwelling',  
                                          'Cephalopholis colonus non-upwelling', 'Cephalopholis colonus upwelling', 
                                          'Abudefduf troschelii non-upwelling', 'Abudefduf troschelii  upwelling',  
                                          'Cephalopholis panamensis non-upwelling', 'Cephalopholis panamensis upwelling', 
                                          'Epinephelus labriformis  non-upwelling', 'Epinephelus labriformis  upwelling')

```

```{r fourth a-div figure - species by species Gulf of ChiriquÃ­ - q0 only}
#fix labels at bottom (3-letter fish codes?)
#add seasonal comparison bars = 2 bars for each species
fig02.3 <- ggplot(data = data.hill.observed.chiriqui, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_season, colour=species_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal8, name = "Species Season")+
        scale_fill_manual(values=disp.pal8, name = "Species Season")+
   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4", "aquamarine2","darkgreen", "#9CCC65","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "darkblue","#5B6CB0", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(name="Hill diversity (Observed q0)") +
  rotate_x_text(60) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig02.3
```
Here, we have non-upwelling (end of wet season 2021) first, in darker shades, followed by upwelling (end of dry season 2022), in lighter shades, for each species. From this plot, we can see that observed richness tends to be relatively constant between seasons in a given host species in Coiba, with the exception of A. xanthopterus, which seems to have a big drop between the wet (non-upwelling) and dry (upwelling) seasons. It's unclear what may be behind this - there are clearly some higher values that may be skewing it higher.

```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill q0 Coiba Species Unrarefied.pdf", plot = fig02.3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Hill q0 Coiba Species Unrarefied.png", plot = fig02.3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```


```{r fourth a-div figure - species by species season - q0 only - Las Perlas}
#fix labels at bottom (3-letter fish codes?)
#add seasonal comparison bars = 2 bars for each species
fig02.4 <- ggplot(data = data.hill.observed.perlas, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_season, colour=species_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal8, name = "Species")+
        scale_fill_manual(values=disp.pal8, name = "Species")+
   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4", "aquamarine2","darkgreen", "#9CCC65","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "darkblue","#5B6CB0", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(name="Hill diversity (Observed q0)") +
  rotate_x_text(60) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig02.4
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill q0 Species Season Perlas Unrarefied.pdf", plot = fig02.4, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Hill q0 Species Season Perlas Unrarefied.png", plot = fig02.4, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

In comparison to Coiba, for Las Perlas we can see that the carnivores seem to have quite a bit higher a-diversity (mean of ~250 to ~500, compared to ~250-350), especially E. labriformis. Furthermore, within species and across seasons, we see much more change, with a small, yet consistent increase in observed richness across all 4 herbivores and the planktivore. On the other hand, the invertivores, omnivore, and carnivores show no consistent pattern, either barely changing (J. nigrirostris) or decreasing slightly (A. troschelii, C. humeralis, C. panamensis) or increasing (only for E. labriformis)

Now that we have these two figures, we can stack them, to facilitate comparisons (we'll also likely want to adjust the axes slightly so that they are comparable across both graphs)

Following the order established previously, we'll have the Gulf of Chiriqui (Coiba) on top and the Gulf of Panama (Las Perlas) on the bottom, which means we can remove the x-axis label from the Coiba figure

We can also probably reduce the slant of the x-axis text since we now have more space for each label

We'll also try out adding a color-coded box around each panel, corresponding to the pre-established color for each gulf


```{r update Coiba plot}
fig02.5 <- ggplot(data = data.hill.observed.chiriqui, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_season, colour=species_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal8, name = "Species Season")+
        scale_fill_manual(values=disp.pal8, name = "Species Season")+
   theme(axis.text.x = element_text(angle = 0, hjust = 0.5, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4", "aquamarine2","darkgreen", "#9CCC65","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "darkblue","#5B6CB0", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name=NULL, labels = c()) + #'A. con', 'M. dor', 'A. xan', 'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 'C. pan', 'E. lab'
                           scale_y_continuous(limits = c(0, 970), name="Hill diversity (Observed q0)") +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color = "#CC5500", size = 1.5), # orange box
              #panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig02.5
```
###In progress 12/3/24
```{r update Las Perlas plot}
fig02.6 <- ggplot(data = data.hill.observed.perlas, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_season, colour=species_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal8, name = "Species")+
        scale_fill_manual(values=disp.pal8, name = "Species")+
   theme(axis.text.x = element_text(angle = 0, hjust = 0.5, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4", "aquamarine2","darkgreen", "#9CCC65","darkgreen", "#9CCC65",  "darkblue","#5B6CB0", "darkblue","#5B6CB0", "darkmagenta","magenta3", "darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(limits = c(0, 970), name="Hill diversity (Observed q0)") +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color =  "#009999", size = 1.5), # turquoise box around plot
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig02.6
  
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill q0 Species Season Coiba Unrarefied FOR COWPLOT.pdf", plot = fig02.5, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)

ggsave(filename = "TEP Skin Microbiome Hill q0 Species Season Perlas Unrarefied FOR COWPLOT.pdf", plot = fig02.6, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Hill q0 Species Season Coiba Unrarefied FOR COWPLOT.png", plot = fig02.5, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)

ggsave(filename = "TEP Skin Microbiome Hill q0 Species Season Perlas Unrarefied FOR COWPLOT.png", plot = fig02.6, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

```{r cowplot Coiba and Las Perlas together}
# library(cowplot) #if not already loaded

# Remove y-axis title from both plots
fig02.5_adj <- fig02.5 +
  theme(axis.title.y = element_blank())
fig02.6_adj <- fig02.6 +
  theme(axis.title.y = element_blank())

# Add a shared y-axis title
shared_y_axis <- ggdraw() +
  draw_label("Hill diversity (Observed q0)", angle = 90, vjust = 1.5, size = 18)

# Combine the adjusted plots with aligned sizes
combined_plot <- plot_grid(
  shared_y_axis,                           # Shared y-axis title
  plot_grid(fig02.5_adj, fig02.6_adj, 
            ncol = 1,                      # Arrange plots in one column
            align = "v",                   # Align vertically for uniform sizes
            axis = "lr"),                  # Align left and right axes
  rel_widths = c(0.05, 1),                 # Space for y-axis title
  nrow = 1
)

# Display the combined plot
print(combined_plot)
```
```{r optimize cowplot Coiba and Las Perlas together}

# Remove y-axis title from both plots
fig02.5_adj <- fig02.5 +
  theme(axis.title.y = element_blank())
fig02.6_adj <- fig02.6 +
  theme(axis.title.y = element_blank())

# Remove x-axis title
fig02.6_adj <- fig02.6_adj +
  theme(axis.title.x = element_blank())

# Add a shared y-axis title
shared_y_axis <- ggdraw() +
  draw_label("Hill diversity (Observed q0)", angle = 90, vjust = 1.5, size = 18)

# Add shared x-axis title
shared_x_axis <- ggdraw() +
  draw_label("Fish Host Species", hjust = 0.5, size = 18)

# Combine the adjusted plots with aligned sizes
combined_plot2 <- plot_grid(
  shared_y_axis,  # Shared y-axis title
  plot_grid(fig02.5_adj, fig02.6_adj, 
            ncol = 1,  # Arrange plots in one column
            align = "v",  # Align vertically for uniform sizes
            axis = "lr"),  # Align left and right axes
  rel_widths = c(0.05, 1),  # Space for y-axis title
  nrow = 1  # Align y-axis and plots in one row
)

# Add the x-axis title at the bottom, centered
final_plot <- plot_grid(
  combined_plot2,   # The combined plot without x-axis title
  shared_x_axis,    # The shared x-axis title below
  nrow = 2,         # Stack the plots vertically
  rel_heights = c(1, 0.05)  # Space for the x-axis title at the bottom
)

# Display the final plot
print(final_plot)
```
```{r add label for each gulf to plot}
# Remove y-axis title from both plots
fig02.5_adj <- fig02.5 +
  theme(axis.title.y = element_blank()) +
  annotation_custom(
    grob = textGrob("Gulf of ChiriquÃ­", x = 0.05, y = 0.95, just = c("left", "top"), gp = gpar(fontsize = 14, fontface = "italic"))
  )

fig02.6_adj <- fig02.6 +
  theme(axis.title.y = element_blank()) +
  annotation_custom(
    grob = textGrob("Gulf of Panama", x = 0.05, y = 0.95, just = c("left", "top"), gp = gpar(fontsize = 14, fontface = "italic"))
  )

# Remove x-axis title
fig02.6_adj <- fig02.6_adj + 
  theme(axis.title.x = element_blank())

# Add a shared y-axis title
shared_y_axis <- ggdraw() + 
  draw_label("Hill diversity (Observed q0)", angle = 90, vjust = 1.5, size = 18)

# Add shared x-axis title, but we'll position it at the bottom and center it later
shared_x_axis <- ggdraw() + 
  draw_label("Fish Host Species", hjust = 0.5, size = 18)

# Combine the adjusted plots with aligned sizes
combined_plot3 <- plot_grid(
  shared_y_axis,  # Shared y-axis title
  plot_grid(fig02.5_adj, fig02.6_adj, 
            ncol = 1,  # Arrange plots in one column
            align = "v",  # Align vertically for uniform sizes
            axis = "lr"),  # Align left and right axes
  rel_widths = c(0.05, 1),  # Space for y-axis title
  nrow = 1  # Align y-axis and plots in one row
)

# Add the x-axis title at the bottom, centered
final_plot2 <- plot_grid(
  combined_plot3,   # The combined plot without x-axis title
  shared_x_axis,    # The shared x-axis title below
  nrow = 2,         # Stack the plots vertically
  rel_heights = c(1, 0.05)  # Space for the x-axis title at the bottom
)

# Display the final plot
print(final_plot2)

```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "Stacked TEP Skin Microbiome Hill q0 Species Season Unrarefied.pdf", plot = final_plot2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "Stacked TEP Skin Microbiome Hill q0 Species Season Unrarefied.png", plot = final_plot2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

#Add water to plot as well

```{r read in water plots}
fig03.1 = readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/water_barchart_coiba.rds")

fig03.2 = readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft/water_barchart_lasperlas.rds")
```

```{r integrate water plots for comparison - Coiba}
# Taking fig03.1 from the water a-div script

# Combine fig02.5 and fig03.1 with fig03.1 taking up 1/8 of the width
combined_plot_top <- plot_grid(
  fig02.5_adj,              # Boxplot (fig02.5)
  fig03.1,              # Bar plot (fig03.1)
  ncol = 2,             # Arrange in 2 columns
  rel_widths = c(8, 1),  # fig02.5 takes up 8/9 of the width, fig03.1 takes up 1/9
  align = "h",          # Align horizontally
  axis = "tb"           # Align the axes (top/bottom)
)

# Display the combined plot
print(combined_plot_top)

```
```{r integrate water plots for comparison - Las Perlas}
# Taking fig03.2 from the water a-div script

# Combine fig02.5 and fig03.1 with fig03.1 taking up 1/8 of the width
combined_plot_bottom <- plot_grid(
  fig02.6_adj,              # Boxplot (fig02.5)
  fig03.2,              # Bar plot (fig03.1)
  ncol = 2,             # Arrange in 2 columns
  rel_widths = c(8, 1),  # fig02.5 takes up 8/9 of the width, fig03.1 takes up 1/9
  align = "h",          # Align horizontally
  axis = "tb"           # Align the axes (top/bottom)
)

# Display the combined plot
print(combined_plot_bottom)

```
```{r combine both gulfs}
# Combine the adjusted plots with aligned sizes
combined_plot4 <- plot_grid(
  shared_y_axis,  # Shared y-axis title
  plot_grid(combined_plot_top, combined_plot_bottom, 
            ncol = 1,  # Arrange plots in one column
            align = "v",  # Align vertically for uniform sizes
            axis = "lr"),  # Align left and right axes
  rel_widths = c(0.05, 1),  # Space for y-axis title
  nrow = 1  # Align y-axis and plots in one row
)

# Add the x-axis title at the bottom, centered
final_plot4 <- plot_grid(
  combined_plot4,   # The combined plot without x-axis title
  shared_x_axis,    # The shared x-axis title below
  nrow = 2,         # Stack the plots vertically
  rel_heights = c(1, 0.05)  # Space for the x-axis title at the bottom
)

# Display the final plot
print(final_plot4)
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "Combined TEP Skin Microbiome Hill q0 Species Season Unrarefied WITH WATER.pdf", plot = final_plot4, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "Combined TEP Skin Microbiome Hill q0 Species Season Unrarefied WITH WATER.png", plot = final_plot4, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

We'll add fish and water icons to identify species on this plot

```{r read in fish icons}
## fish icons - color-coded by species w/ black outlines
icon_Aconc2 <- readPNG("AconcolorC_outline.png") #using same genus-level silhouettes for now as placeholders
icon_Atro2 <- readPNG("AtroscheliiC_outline.png")
icon_Axan2 <- readPNG("AxanthopterusC_outline.png")
icon_Chum2 <- readPNG("ChumeralisC_outline.png")
icon_Cpan2 <- readPNG("CpanamensisC_outline.png")
icon_Elab2 <- readPNG("ElabriformisC_outline.png")
icon_Jnig2 <- readPNG("JnigrirostrisC_outline.png")
icon_Mdor2 <- readPNG("MdorsalisC_outline.png")
icon_Ccol2 <- readPNG("CcolonusC_outline.png")
icon_Plat2 <- readPNG("PlaticlaviusC_outline.png")
icon_Water2 <- readPNG("Water1C_outline.png")
```
```{r add icons to plot above}
#add space to bottom of plot for icons
# Add the x-axis title at the bottom, centered
final_plot5 <- plot_grid(
  combined_plot4,   # The combined plot without x-axis title
  shared_x_axis,    # The shared x-axis title below
  nrow = 2,         # Stack the plots vertically
  rel_heights = c(1, 0.1)  # Space for the x-axis title at the bottom
)

# Display the final plot
print(final_plot5)

#add icons to graph
final_plot5_icons <- ggdraw() +
  draw_plot(final_plot5) +
  draw_image(image = icon_Aconc2, x = -0.37, y =  -0.42, scale = 0.050) +# A con
  draw_image(image = icon_Mdor2, x = -0.29, y = -0.42, scale = 0.050) +# M dor
  draw_image(image = icon_Axan2, x =  -0.21, y =  -0.42, scale = 0.060) +# A xan
  draw_image(image = icon_Plat2, x = -0.13, y = -0.42, scale = 0.060) +# P lat
  draw_image(image = icon_Jnig2, x = -0.05, y = -0.42, scale = 0.040) +# J nig
  draw_image(image = icon_Chum2, x = 0.03, y = -0.42, scale = 0.040) +# C hum
  draw_image(image = icon_Cpan2, x =  0.11, y = -0.42, scale = 0.060) +# C pan
  draw_image(image = icon_Elab2, x = 0.19, y =  -0.42, scale = 0.060) +# E lab
  draw_image(image = icon_Ccol2, x = 0.27, y = -0.42, scale = 0.060) +# C col
  draw_image(image = icon_Atro2, x = 0.35, y = -0.42, scale = 0.050) +# A tro
  draw_image(image = icon_Water2, x = 0.45, y = -0.42, scale = 0.060) # Water

final_plot5_icons
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "Combined TEP Skin Microbiome Hill q0 Species Season Unrarefied WITH WATER and icons.pdf", plot = final_plot5_icons, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "Combined TEP Skin Microbiome Hill q0 Species Season Unrarefied WITH WATER and icons.png", plot = final_plot5_icons, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

## Rarefied Hill numbers - region, season, and species
```{r subset dataset into region and season}
#Coiba
Coiba_up <- subset_samples(ROHR_06.phylo.hill.r, region_season == "Coiba upwelling")
Coiba_up <- prune_taxa(taxa_sums(Coiba_up) > 0, Coiba_up) #4095 taxa and 74 samples
Coiba_up
Coiba_N <- subset_samples(ROHR_06.phylo.hill.r, region_season == "Coiba non-upwelling")
Coiba_N <- prune_taxa(taxa_sums(Coiba_N) > 0, Coiba_N) #4022 taxa and 65 samples
Coiba_N
#Las Perlas
LP_up <- subset_samples(ROHR_06.phylo.hill.r, region_season == "Las Perlas upwelling")
LP_up <- prune_taxa(taxa_sums(LP_up) > 0, LP_up) #4031 taxa and 74 samples
LP_up
LP_N <- subset_samples(ROHR_06.phylo.hill.r, region_season == "Las Perlas non-upwelling")
LP_N <- prune_taxa(taxa_sums(LP_N) > 0, LP_N) #4020 taxa and 66 samples
LP_N

#From the summary of the phyloseq objects, we can see that the number of taxa across all samples within a given season & region is relatively consistent (ranging from 4020 to 4095 taxa across 65 to 74 samples)
```
```{r transpose ASV tables to ASVs as rows}
asv.Coiba_up <- t(otu_table(Coiba_up))
asv.Coiba_N <- t(otu_table(Coiba_N))
asv.LP_up <- t(otu_table(LP_up))
asv.LP_N <- t(otu_table(LP_N))
```

```{r transform counts to relative abundance}
asv.Coiba_up.norm <- microbiome::transform(asv.Coiba_up, transform = "compositional")
asv.Coiba_N.norm <- microbiome::transform(asv.Coiba_N, transform = "compositional")
asv.LP_up.norm <- microbiome::transform(asv.LP_up,
                                             transform = "compositional")
asv.LP_N.norm <- microbiome::transform(asv.LP_N,
                                             transform = "compositional")
```

"Now we can test alpha diversity per zone based on Hill numbers at different diversity levels (q = 0 (observed), q = 1 (shannon exponential), q = 2 (multiplicative simpson)).

Note: for Hill numbers, alpha diversity per zone cannot be obtained as a mean across samples. The calculation is different as performed by function alpha_div()." - Clever et al. (2022)

What do these "different diversity levels" do for us?
- since no single metric fully captures diversity, using a trio of metrics that complement each other can help us best understand the complex microbial diversity in our samples. Observed: species richness - "Species richness emphasizes (provides higher leverage to) rare species" (Roswell, 2022), Shannon exponential: "HillâShannon diversity emphasizes neither rare nor common species.", multiplicative simpson: "HillâSimpson diversity emphasizes (provides higher leverage to) the common species."

```{r test a-div per season & region}
#Coiba upwelling
alpha_div(countable = asv.Coiba_up.norm, qvalue = 0)
alpha_div(countable = asv.Coiba_up.norm, qvalue = 1)
alpha_div(countable = asv.Coiba_up.norm, qvalue = 2)

#Coiba non-upwelling
alpha_div(countable = asv.Coiba_N.norm, qvalue = 0)
alpha_div(countable = asv.Coiba_N.norm, qvalue = 1)
alpha_div(countable = asv.Coiba_N.norm, qvalue = 2)

#Las Perlas upwelling
alpha_div(countable = asv.LP_up.norm, qvalue = 0)
alpha_div(countable = asv.LP_up.norm, qvalue = 1)
alpha_div(countable = asv.LP_up.norm, qvalue = 2)

#Las Perlas non-upwelling
alpha_div(countable = asv.LP_N.norm, qvalue = 0)
alpha_div(countable = asv.LP_N.norm, qvalue = 1)
alpha_div(countable = asv.LP_N.norm, qvalue = 2)
```

Test significance of differences between a-div

```{r run Kruskal-Wallis test and post-hoc Dunn}

#full dataset
temp_table <- data.frame(sample_data(ROHR_06_obj_phylo_root))
temp_table[, c(7:10, 13:14, 24:42)] <- list(NULL) #adjust these numbers to define which metadata columns to EXCLUDE
temp_table <- temp_table %>% tibble::rownames_to_column("Sample")
temp_table <- temp_table %>% dplyr::rename(Group = region_season)
temp_table$Group <- as.character(temp_table$Group)
temp_table$species <- as.character(temp_table$species)

#comparisons between seasons & regions
hill_hierarchy <- temp_table[, c(1, 19)] #adjust these numbers for the comparisons between groups you want to make - in this case RRR number (renamed to "Sample") & Group (region_season)

#comparisons between species
hill_hierarchy2 <- temp_table[, c(1, 8)] #RRR number (Sample) & species

#comparisons between sites (within a region)
hill_hierarchy3 <- temp_table[, c(1, 5)] #RRR number & site

```

```{r significance test season region}
#if get error, make sure "RRR_" has been affixed in front of sample IDs in "RRR" column
divtestresult.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy, posthoc = TRUE)
divtestresult.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy, posthoc = TRUE)
divtestresult.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy, posthoc = TRUE)

divtestresult.q0 #no significant differences between seasons and regions in observed a-div --> BUT, while not significantly different, Las Perlas samples during upwelling are more different in observed richness from Coiba "upwelling" samples (p = 0.36) than any other pair tested. (based on p.adj values)

#normality.pvalue = 0.00000007182716
#homogeneity.pvalue = 0.6770346 #data are non normal but ARE homogeneous (post-rarefaction)
#groups = 4
#"Kruskal-Wallis Test"
# chi.squared          df     p.value 
#  3.6056587   3.0000000   0.3073149  #higher than w/ non-rarefied data 

divtestresult.q1 #NSD in shannon exponential across all regions & seasons

#normality.pvalue = 2.602725e-11
#homogeneity.pvalue = 0.2682853
#groups = 4
#"Kruskal-Wallis Test"
# chi.squared          df     p.value 
#  0.2799380   3.0000000   0.9637564  #p-val slightly lower post-rarefaction
 
divtestresult.q2 #NSD in multiplicative simpson across all regions & seasons

#normality.pvalue = 1.304612e-13
#homogeneity.pvalue = 0.2723675
#groups = 4
#"Kruskal-Wallis Test"
# chi.squared          df     p.value 
#  0.2791919   3.0000000   0.9638932   #higher than w/ non-rarefied data

```

```{r significance test between species}
divtestresult2.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy2, posthoc = TRUE)
divtestresult2.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy2, posthoc = TRUE)
divtestresult2.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy2, posthoc = TRUE)

divtestresult2.q0
divtestresult2.q1
divtestresult2.q2

#q0 - observed richness
#normality.pvalue = 0.00000007182716 #data not normal nor homogeneous...
#homogeneity.pvalue = 0.008006304
#groups = 10
#method = "Kruskal-Wallis Test"
#test
#  chi.squared           df      p.value 
# 6.863098e+01 9.000000e+00 2.823387e-11 

#q1 - shannon exponential (weighs ASVs by frequency)
#normality.pvalue = 2.602725e-11
#homogeneity.pvalue = 0.04509324
#groups = 10
#method = "Kruskal-Wallis Test"
#test
#  chi.squared           df               p.value 
# 59.111952611510787  9.000000000000000  0.000000001987434  

#q2 - simpson
#normality.pvalue = 1.304612e-13
#homogeneity.pvalue = 0.187195
#groups = 10
#method = "Kruskal-Wallis Test"
#test
#  chi.squared           df           p.value 
# 48.9740662184748  9.0000000000000  0.0000001678501 

```
Looking at observed richness (q0) between species, we can see that the individual richness values vary quite a bit both within individuals of a same species, and across fish species. Non-parametric Kruskal-Wallis tests show significant differences between many species (A. tro:C. pana p-adj = 0.020, A. xan:C pana = 0.016, A. conc:C. hum = <0.001, A.tro:C. hum = 0.011, A.xan:C.hum = 0.004, C.col:C.hum = <0.001, C.pan:C.hum = <0.001, A.con:E.lab = 0.013, A.tro:E.lab = 0.002, A.xan:E.lab = 0.001, C.col:E.lab = 0.025, ....), not-quite-significant differences between others.

Looking at Shannon exponential (q1) between species, we also see variation in values across samples and significant differences between certain species, although fewer than with observed richness - in both cases, the largest differences are between invertivores (C. humeralis & J. nigrirostris) and carnivores (C. panamensis, E. labriformis).

Looking at Simpson multiplicative inverse (q2) between species, we also see variation in values across samples and significant differences between certain species, again, mostly relative to the invertivorous butterflyfishes (C. humeralis & J. nigrirostris). 

```{r significant test site}
divtestresult3.q0 <- div_test(asv.whole.norm, qvalue = 0,
                             hierarchy = hill_hierarchy3, posthoc = TRUE)
divtestresult3.q1 <- div_test(asv.whole.norm, qvalue = 1,
                             hierarchy = hill_hierarchy3, posthoc = TRUE)
divtestresult3.q2 <- div_test(asv.whole.norm, qvalue = 2,
                             hierarchy = hill_hierarchy3, posthoc = TRUE)
divtestresult3.q0
divtestresult3.q1
divtestresult3.q2
```
No significant differences in alpha diversity between sites.

```{r break species data into season & region - new column}
ROHR_06.phylo.hill.r <- merge_treatments(ROHR_06.phylo.hill.r, c("species", "region", "season"))

# Save new metadata
saveRDS(ROHR_06.phylo.hill.r, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_Hill_merged.rds")

#read new metadata
ROHR_06.phylo.hill.r <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_Hill_merged.rds")
```

```{r adjust sample data on hill phyloseq}

data.hill.whole_n <- data.frame(sample_data(ROHR_06.phylo.hill.r)) %>%
  pivot_longer(cols = 1:3, names_to = "Index",  values_to = "Diversity")
data.hill.whole_n$Fraction_delailed <- NULL
data.hill.whole_n <- data.hill.whole_n[, c(1:4, 6, 11:12, 14, 43:47)] ###Adjust #s to match metadata columns to keep
data.hill.whole_n <- dplyr::mutate_if(data.hill.whole_n, is.factor, as.character)
data.hill.whole_n$Index <- stringr::str_replace(
  data.hill.whole_n$Index, "Observed", "Observed (q=0)")
data.hill.whole_n$Index <- stringr::str_replace(
  data.hill.whole_n$Index, "shannon.hill", "Shannon exponential (q=1)")
data.hill.whole_n$Index <- stringr::str_replace(
  data.hill.whole_n$Index, "simpson.hill", "Simpson multi. inverse (q=2)")
data.hill.whole_n <- data.hill.whole_n[order(data.hill.whole_n$Index),]

saveRDS(data.hill.whole_n, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_hill_datatable_rar.rds")
```

```{r read data with hill num}
data.hill.whole <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_hill_datatable_rar.rds")
```

```{r color codes - don't run}

#from ROHR_06_Swab_Figures:
# set colors for seasons - "#660066" = non-upwelling, "#336633" = upwelling BUT have colors assigned for different sites w/in each gulf for temperature figure
c2 <- c("#660066", "#336633") #will need 2 more colors to show Coiba & Las Perlas separately
#temp figure:
site_cols <- c( "#FFAD65", "#FF6E00", "#662700", "#99FFFF", "#009999", "#006666",
                "#00CCCC", "#00FFFF", "#993D00", "#FF8E33", "#CC5500", "#FFCA99")

sites <- c("Afuera", "Damas", "Uvas", "Contadora", "Saboga", "MogoMogo", #water sampling
           "Pacheca", "Chapera", "Machete", "Cocos", "Frijol", "Granito" ) #fish only

##colors for individual fish species
col_AconcS <- "aquamarine4"
col_AtroS <- "darkgoldenrod1"
col_AxanS <- "chartreuse3"
col_CcolS <- "darkmagenta"
col_ChumS <- "deepskyblue2"
col_CpanS <- "chocolate4" 
col_ElabS <- "chocolate2"
col_JnigS <- "darkblue" 
col_MdorS <- "turquoise"
col_PlatS <- "darkgreen"
col_WaterS <- "#BE93D4"
```

```{r summary figure a-div Hill numbers}
level_order <- c('Coiba upwelling', 'Coiba non-upwelling', 'Las Perlas upwelling', 'Las Perlas non-upwelling')
level_order2 <- c('Observed', 'shannon.hill', 'simpson.hill')
level_order3 <- c('Abudefduf concolor', 'Microspathodon dorsalis', 'Acanthurus xanthopterus', 
                  'Prionurus laticlavius', 'Johnrandallia nigrirostris', 'Chaetodon humeralis',
                  'Cephalopholis colonus', 'Abudefduf troschelii', 'Cephalopholis panamensis', 
                  'Epinephelus labriformis')
level_order3a <- c('Acon', 'Mdor', 'Axan', 'Plat', 'Jnig', 'Chum', 'Ccol', 'Atro', 'Cpan', 'Elab')
level_order4 <- c('Coiba', 'Las Perlas')

#season & region palette
disp.pal2 <- c("#662700", "#FF6E00", "#006666", "#009999")
#individual fish palette
disp.pal3 <- c( "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2")
disp.pal4 <- c( "darkslategray", "darkslategray4", "chartreuse4", "#1F3E20", "midnightblue",
                "deepskyblue4", "mediumpurple4", "goldenrod4", "#4E342F", "darkorange3")
#MAY NEED 3RD PALETTE FOR FISH + SEASON

```

```{r define levels - gulf & season}
data.hill.whole$region_season <- factor(data.hill.whole$region_season,
                               levels = c('Coiba upwelling', 'Coiba non-upwelling', 'Las Perlas upwelling', 
                                          'Las Perlas non-upwelling'))
```

```{r define levels - gulf only}
data.hill.whole$region <- factor(data.hill.whole$region,
                               levels = c('Coiba', 'Las Perlas'))
```

```{r define levels - species}
data.hill.whole$species <- factor(data.hill.whole$species,
                               levels = c('Abudefduf concolor', 'Microspathodon dorsalis', 
                                          'Acanthurus xanthopterus', 'Prionurus laticlavius', 
                                          'Johnrandallia nigrirostris', 'Chaetodon humeralis',
                                          'Cephalopholis colonus', 'Abudefduf troschelii', 
                                          'Cephalopholis panamensis', 'Epinephelus labriformis'))
```

```{r define graph theme}
font_theme = theme(
        axis.title.x = element_text(size = 16),
        axis.text.x = element_text(size = 12),
        axis.title.y = element_text(size = 16),
        axis.text.y = element_text(size = 10))
```

```{r first a-div figure - season region}

fig04 <-
        ggplot(data = data.hill.whole, aes(x = factor(region, level = level_order4),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=region_season, colour=region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal2)+
        scale_fill_manual(values=disp.pal2)+
        theme(plot.title=element_text(size=18)) +
        theme(plot.subtitle=element_text(size=16)) +
        font_theme+
        stat_summary(aes(fill=region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999",
                               "#662700", "#FF6E00", "#006666", "#009999"),
                     show.legend = FALSE) +
        scale_x_discrete(name="Region") + scale_y_continuous(name="Hill diversity") +
        theme(plot.title = element_text(family=NA, face="bold", size=16)) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12),
              legend.title = element_text(size = 14)) +
  labs(fill = "Region and Season", colour = "Region and Season")

fig04
```
```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Alpha Diversity RAREFIED Obs Shan Simp.pdf", plot = fig04, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome Alpha Diversity RAREFIED Obs Shan Simp.png", plot = fig04, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

From this figure, we can see that there is high heterogeneity in a-diversity across samples within each gulf and sampling season. Observed numbers range from near zero to almost one thousand (Las Perlas, upwelling), with medians around 200 and means consistently higher (~ 200-3-0; likely due to several high-diversity outliers in each group bringing the average up). Similar patterns are seen for Shannon and Simpson (albeit scaled differently). Although the large variation makes it hard to distinguish significant patterns, the range in a-diversity observed in Las Perlas during upwelling appears to generally be greater than the other sites and seasons, with a higher mean and median for Observed and Shannon diversity (while these differences are not significant, as can be seen in the Kruskall-Wallis tests run above, they suggest trends that may become significant with a larger sample size and/or when filtering out more samples with low total read counts that are skewing the a-diversity values towards zero). 

```{r second a-div figure - species by species}
#fix labels at bottom (3-letter fish codes?)
#add seasonal comparison bars = 2 bars for each species
fig05 <-
        ggplot(data = data.hill.whole, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species, colour=species), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal4, name = "Species")+
        scale_fill_manual(values=disp.pal3, name = "Species")+
        font_theme +
        stat_summary(aes(fill=species), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c( "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2",
                "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2",
                "aquamarine4", "turquoise", "chartreuse3", "darkgreen", "darkblue",
                "deepskyblue2", "darkmagenta", "darkgoldenrod1", "chocolate4", "chocolate2"),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A.con', 'M.dor', 'A.xan', 
                                                              'P.lat', 'J.nig', 'C.hum', 'C.col', 'A.tro', 
                                                              'C.pan', 'E.lab')) + 
        scale_y_continuous(name="Hill diversity") +
        theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14))
fig05
```

```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill Numbers RAREFIED species.pdf", plot = fig05, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome  Hill Numbers RAREFIED species.png", plot = fig05, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

####Jump here for levels
```{r define levels - species_region_season}

data.hill.whole$species_region_season <- factor(data.hill.whole$species_region_season,
                               levels = c('Abudefduf concolor Coiba upwelling', 'Abudefduf concolor Coiba non-upwelling', 'Abudefduf concolor Las Perlas upwelling', 'Abudefduf concolor Las Perlas non-upwelling',
                                          'Microspathodon dorsalis Coiba upwelling', 'Microspathodon dorsalis Coiba non-upwelling', 'Microspathodon dorsalis Las Perlas upwelling', 'Microspathodon dorsalis Las Perlas non-upwelling', 
                                          'Acanthurus xanthopterus Coiba upwelling', 'Acanthurus xanthopterus Coiba non-upwelling', 'Acanthurus xanthopterus Las Perlas upwelling', 'Acanthurus xanthopterus Las Perlas non-upwelling',
                                          'Prionurus laticlavius Coiba upwelling', 'Prionurus laticlavius Coiba non-upwelling', 'Prionurus laticlavius Las Perlas upwelling', 'Prionurus laticlavius Las Perlas non-upwelling', 
                                          'Johnrandallia nigrirostris Coiba upwelling', 'Johnrandallia nigrirostris Coiba non-upwelling', 'Johnrandallia nigrirostris Las Perlas upwelling', 'Johnrandallia nigrirostris Las Perlas non-upwelling', 
                                          'Chaetodon humeralis Coiba upwelling',  'Chaetodon humeralis Coiba non-upwelling',  'Chaetodon humeralis Las Perlas upwelling',  'Chaetodon humeralis Las Perlas non-upwelling',
                                          'Cephalopholis colonus Coiba upwelling', 'Cephalopholis colonus Coiba non-upwelling', 'Cephalopholis colonus Las Perlas upwelling', 'Cephalopholis colonus Las Perlas non-upwelling', 
                                          'Abudefduf troschelii Coiba upwelling',  'Abudefduf troschelii Coiba non-upwelling',  'Abudefduf troschelii Las Perlas upwelling',  'Abudefduf troschelii Las Perlas non-upwelling', 
                                          'Cephalopholis panamensis Coiba upwelling', 'Cephalopholis panamensis Coiba non-upwelling', 'Cephalopholis panamensis Las Perlas upwelling', 'Cephalopholis panamensis Las Perlas non-upwelling', 
                                          'Epinephelus labriformis Coiba upwelling', 'Epinephelus labriformis Coiba non-upwelling', 'Epinephelus labriformis Las Perlas upwelling', 'Epinephelus labriformis Las Perlas non-upwelling'))
```

```{r new list for species region season}
level_order5 <- c('Abudefduf concolor Coiba upwelling', 'Abudefduf concolor Coiba non-upwelling', 'Abudefduf concolor Las Perlas upwelling', 'Abudefduf concolor Las Perlas non-upwelling',
                                          'Microspathodon dorsalis Coiba upwelling', 'Microspathodon dorsalis Coiba non-upwelling', 'Microspathodon dorsalis Las Perlas upwelling', 'Microspathodon dorsalis Las Perlas non-upwelling', 
                                          'Acanthurus xanthopterus Coiba upwelling', 'Acanthurus xanthopterus Coiba non-upwelling', 'Acanthurus xanthopterus Las Perlas upwelling', 'Acanthurus xanthopterus Las Perlas non-upwelling',
                                          'Prionurus laticlavius Coiba upwelling', 'Prionurus laticlavius Coiba non-upwelling', 'Prionurus laticlavius Las Perlas upwelling', 'Prionurus laticlavius Las Perlas non-upwelling', 
                                          'Johnrandallia nigrirostris Coiba upwelling', 'Johnrandallia nigrirostris Coiba non-upwelling', 'Johnrandallia nigrirostris Las Perlas upwelling', 'Johnrandallia nigrirostris Las Perlas non-upwelling', 
                                          'Chaetodon humeralis Coiba upwelling',  'Chaetodon humeralis Coiba non-upwelling',  'Chaetodon humeralis Las Perlas upwelling',  'Chaetodon humeralis Las Perlas non-upwelling',
                                          'Cephalopholis colonus Coiba upwelling', 'Cephalopholis colonus Coiba non-upwelling', 'Cephalopholis colonus Las Perlas upwelling', 'Cephalopholis colonus Las Perlas non-upwelling', 
                                          'Abudefduf troschelii Coiba upwelling',  'Abudefduf troschelii Coiba non-upwelling',  'Abudefduf troschelii Las Perlas upwelling',  'Abudefduf troschelii Las Perlas non-upwelling', 
                                          'Cephalopholis panamensis Coiba upwelling', 'Cephalopholis panamensis Coiba non-upwelling', 'Cephalopholis panamensis Las Perlas upwelling', 'Cephalopholis panamensis Las Perlas non-upwelling', 
                                          'Epinephelus labriformis Coiba upwelling', 'Epinephelus labriformis Coiba non-upwelling', 'Epinephelus labriformis Las Perlas upwelling', 'Epinephelus labriformis Las Perlas non-upwelling')

disp.pal5 <- c( "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745")

disp.pal6 <- c( "aquamarine4", "aquamarine4","aquamarine4","aquamarine4","turquoise","turquoise","turquoise","turquoise","chartreuse3","chartreuse3","chartreuse3","chartreuse3", "darkgreen","darkgreen", "darkgreen", "darkgreen", "darkblue","darkblue","darkblue","darkblue", "deepskyblue2","deepskyblue2", "deepskyblue2", "deepskyblue2",   "darkmagenta","darkmagenta", "darkmagenta", "darkmagenta", "darkgoldenrod1", "darkgoldenrod1","darkgoldenrod1", "darkgoldenrod1", "chocolate4","chocolate4","chocolate4","chocolate4", "chocolate2","chocolate2", "chocolate2", "chocolate2")

disp.pal7 <- c("#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999", "#662700", "#FF6E00", "#006666", "#009999")

#labels = c('Acon C-U', 'Acon C-N', 'Acon LP-U', 'Acon LP-N', 'Mdor C-U','Mdor C-N','Mdor LP-U','Mdor LP-N', 'Axan C-U', 'Axan C-N','Axan LP-U','Axan LP-N','Plat C-U', 'Plat C-N','Plat LP-U','Plat LP-N','Jnig C-U', 'Jnig C-N','Jnig LP-U','Jnig LP-N','Chum C-U', 'Chum C-N','Chum LP-U','Chum LP-N', 'Ccol C-U', 'Ccol C-N','Ccol LP-U','Ccol LP-N','Atro C-U', 'Atro C-N', 'Atro LP-U', 'Atro LP-N', 'Cpan C-U', 'Cpan C-N', 'Cpan LP-U', 'Cpan LP-N', 'Elab C-U', 'Elab C-N', 'Elab LP-U', 'Elab LP-N')
```

```{r third a-div species - season - region}

fig06 <-
        ggplot(data = data.hill.whole, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_region_season, colour=species_region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal6, name = "Species")+
        scale_fill_manual(values=disp.pal5, name = "Species")+
   theme(axis.text.x = element_text(angle = 45, hjust = 1, face = "italic")) +
        font_theme+
        stat_summary(aes(fill=species_region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745",
                               "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745",
                               "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(name="Hill diversity") +
  rotate_x_text(60) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12, face = "italic"),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig06
```

```{r code for exporting a-div plot above as a pdf}
ggsave(filename = "TEP Skin Microbiome Hill Numbers RAREFIED Species Region Season.pdf", plot = fig06, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='pdf', dpi=700)
```

```{r code for exporting a-div plot above as a png}
ggsave(filename = "TEP Skin Microbiome  Hill Numbers RAREFIED Species Region Season.png", plot = fig06, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 12, height = 8, device='png', dpi=700)
```

####In Progress 11/18/24 #### RAREFIED FIGS

```{r a-div species - season - region v 2}
# fix these colors!!!! (idea was to have some part of bar color indicate season/region while the other indicates fish species)

fig07 <-
        ggplot(data = data.hill.whole, aes(x = factor(species, level = level_order3),
                                           y=Diversity)) +
        stat_boxplot(aes(colour=species_region_season), geom ='errorbar') + #alpha=0.3
        geom_boxplot(aes(fill=species_region_season, colour=species_region_season), alpha=0.6,
                     outlier.alpha = 0.7, outlier.shape = 16) +
        scale_colour_manual(values=disp.pal7, name = "Species")+
        scale_fill_manual(values=disp.pal5, name = "Species")+
        font_theme+
        stat_summary(aes(fill=species_region_season), fun="mean", geom="point", size=2.5, shape=18,
                     position=position_dodge(width=0.75),
                     color=c(  "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745",
                               "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745",
                               "aquamarine4", "aquamarine2","aquamarine4","aquamarine2","turquoise","turquoise1","turquoise","turquoise1","chartreuse3","chartreuse","chartreuse3","chartreuse", "darkgreen", "#9CCC65", "darkgreen", "#9CCC65", "darkblue","#5B6CB0","darkblue","#5B6CB0", "deepskyblue2","deepskyblue", "deepskyblue2", "deepskyblue", "darkmagenta","magenta3", "darkmagenta", "magenta3", "darkgoldenrod1", "#FDDE0F","darkgoldenrod1", "#FDDE0F", "chocolate4","chocolate","chocolate4","chocolate", "chocolate2","#F9A745", "chocolate2", "#F9A745"
                               ),
                     show.legend = FALSE) +
        scale_x_discrete(name="Fish Host Species", labels = c('A. con', 'M. dor', 'A. xan', 
                                                              'P. lat', 'J. nig', 'C. hum', 'C. col', 'A. tro', 
                                                              'C. pan', 'E. lab')) + 
                           scale_y_continuous(name="Hill diversity") +
  rotate_x_text(60) +
        theme(plot.title = element_text(family=NA, face="bold", size=16)) +
        facet_wrap(~factor(Index, levels=unique(Index)),scales="free",
                   nrow=2, ncol = 3) +
        theme(strip.background = element_rect(color="black", fill="white",
                                              size=1, linetype="blank"),
              strip.text.x = element_text(size = 14),
              panel.border = element_rect(fill = NA, color="black"),
              panel.grid.major = element_blank(),
              panel.grid.minor = element_blank(),
              panel.background = element_rect(fill = "white",colour = NA),
              legend.key = element_rect(fill = "transparent", color = NA),
              legend.text = element_text(size = 12),
              legend.title = element_text(size = 14),
              legend.position = "none") #remove legend from plot
fig07
```
