---
title: "ROHR_06_Swab_Figures"
author: ""
date: "2023-02-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Most recent edit: 4/7/24 - messing with temp graph & colored axes of PCoA plots

RRR TEP Fish Skin Microbiome Project Summary

- Fish were collected from 2 sites in the Tropical Eastern Pacific: Las Perlas & Coiba. 
- Sampling during both upwelling & non-upwelling seasons (collections done 2021-22). 
- 10 target species
- Swabs = 9 per species.
- Control swabs (swabs exposed to air in field) collected at each site and season
- Swabs were stored in liquid nitrogen for transport from field to lab, then at -80ºC
- DNA was extracted using ZymoBIOMICS 96 well DNA kit - 4 plates total
- DNA was amplified following a modified version of the Earth Microbiome Project 16S protocol using phased primers (1 set per plate)
- Negative PCR controls and extraction controls included in each plate
- Index PCR performed to attach unique barcodes, then all four plates (384 samples) pooled for sequencing on the Illumina MiSeq sequencing platform of the Smithsonian Tropical Research Institute's (STRI) Naos facilities in Panama (courtesy of Marta Vargas)
- Sequenced libraries were demultiplexed at STRI by Marta Vargas using the MiSeq Reporter Software
- These libraries were trimmed with Cutadapt to remove primers before being read into R to start the DADA2 pipeline (following tutorial v 1.16) 
- Silva version 138.1 (mar 10, 2021 update) used for taxonomic assignments
- See "Skin Pre Processing RRR Fish 16S Sequences Summer 2022" document for pipeline and processing up to the creation of the initial phyloseq object
- See "ROHR_06_Swab_stats_alpha" and "ROHR_06_Swab_stats_beta" for the full set of alpha and beta diversity analyses done for these data


In this document, I have drafted various versions of figures for publication, including ones to be placed in the main text as well as ones that will either be in the supplemental materials or cut altogether. The main figures include (1) ones that provide context and visual descriptions of the study design - e.g., map of study sites, list of study species showing their phylogenetic relationships, etc., (2) individual and multi-panel figures providing summary graphics illustrating various alpha and beta diversity metrics - either with full (all 10 species, 2 seasons, 2 regions) or split datasets, (3) stacked barplots to visualize microbial community composition across host species, sampling gulfs, and seasons. Different versions of the same plot are placed in separate code chunks with a version # (e.g., V1) indicated in the chunk header. Once choices are finalized, the selected version of the plot will be labeled with "FINAL" and copied to a cleaned document for publication alongside the main paper. 


####Section 1 - Maps, Study Design, and Overall Methods
```{r load required packages for maps, echo=FALSE}
# insert necessary packages for data visualizations (not full list of packages used for initial statistics and data processing)

library(maps)

#packages for map creation (from "Geocomputation with R")
library(sf) #"Support for simple features, a standardized way to encode spatial vector data" - from CRAN
library(terra) #Spatial data analysis - w/ vector & raster data
library(dplyr) #working with data frame like objects
library(spData) #"Diverse spatial datasets for demonstrating, benchmarking and teaching spatial data analysis"
library(spDataLarge) #Large spatial datasets
library(tmap)    # for static and interactive maps
library(leaflet) # for interactive maps
library(ggplot2) # tidyverse data visualization package

#library(rmapshaper) # simplify map to shorten rendering time - if using ms_simplify
library(grid) #combine map & inset
library(tmaptools) # working with map shapefiles
library(cowplot) # placing multiple ggplot maps together

library(gridExtra) # creating multiple plots side-by-side - or use ggarrange (from ggpubr)
library(ggspatial) # add scale bar (ggplot2)
library(ggpubr) # plots of publication
```

Online version of book "Geocomputation with R" by Robin Lovelace, Jakub Nowosad, and James Muenchow (first accessed 2-14-23): https://r.geocompx.org/
--> Chapter 9 of book = Making maps with R using the tmap package for map creation

Alternative is to use ggplot2 and its mapping capabilities, as outlined in Chapter 6 of the "ggplot2: Elegant Graphics for Data Analysis" online resource https://ggplot2-book.org/maps.html

Trial basic map of Panama using ggplot2 and the maps package
```{r Panama map basic, echo=FALSE}
PA_map_1 <- map_data(map = "world2", region = "panama") %>% 
  select(lon = long, lat, group, id = subregion)
head(PA_map_1)

ggplot(PA_map_1, aes(lon, lat, group = group)) +
  geom_polygon(fill = "white", colour = "grey50") + 
  coord_quickmap()
```
Using modified code from R with White Dwarf blog post "Map any region of the world with R" to create a map that places Panama within the broader geographic context
```{r Panama map basic with surroundings, echo= FALSE}

## make a df with only the country (Panama) to overlap, using the maps package data
map_data_es <- map_data('world2')[map_data('world2')$region == "Panama",]
## The map (maps + ggplot2 )
ggplot() +
    ## First layer: worldwide map (will crop to only show chosen region with xlim/ylim with coord_fixed layer below)
    geom_polygon(data = map_data("world2"),
                 aes(x=long, y=lat, group = group),
                 color = '#9c9c9c', fill = '#f3f3f3') +
    ## Second layer: country map
    geom_polygon(data = map_data_es,
                 aes(x=long, y=lat, group = group),
                 color = 'burlywood4', fill = 'blanchedalmond') +
    coord_map() +
    coord_fixed(ratio = 1,
                xlim = c(277, 283), #sets limits for x-axis (latitude coordinates) and y-axis (longitude)
                ylim = c(7, 10)) +
    ggtitle("Panama map trial") +
  xlab("Longitude (º)") +
  ylab("Latitude (º)") +
    theme(panel.background =element_rect(fill = 'paleturquoise3'))
```
```{r Panama (Bocas) map test with sampling dots (not our project), echo = FALSE}
## make a df with only the country (Panama) to overlap, using the maps package data
map_data_es <- map_data('world2')[map_data('world2')$region == "Panama",]

##convert negative longitude coords to + values (e.g., 360-82.188706, to match base map)
360 - 82.188706
360 - 82.187738
360 - 82.184815 
360 - 82.183134
360 - 82.144648
360 - 82.14345
360 -82.162696 
360 -82.162147
360 -82.224027 
360 -82.223606
360 -82.247359
360 -82.247359

#list out lat/lon coordinates for sampling sites (Heather's data points in Bocas)
mangrove_islands <- tibble::tribble( 
  ~Island_side,   ~lat,     ~lon,
  "Daphnia Wind", 9.263343, 277.8113,
  "Daphnia Lee",  9.26318, 277.8123, 
  "Cougar Wind",  9.260556, 277.8152, 
  "Cougar Lee",   9.260659, 277.8169, 
  "Tapir Wind",   9.229135, 277.8554, 
  "Tapir Lee",    9.228313, 277.8565, 
  "Sid Wind",     9.210902, 277.8373, 
  "Sid Lee",      9.209542, 277.8379,
  "Orca Wind",    9.249792, 277.776, 
  "Orca Lee",     9.248696, 277.7764,
  "Gragger Wind", 9.277985, 277.7526, 
  "Gragger Lee",  9.27822,  277.7526, ###this data point for longitude needs to be corrected, put same as wind as placeholder for now
)

## map (maps + ggplot2 )
ggplot() +
    ## First layer: worldwide map (will crop to only show chosen region with xlim/ylim with coord_fixed layer below)
    geom_polygon(data = map_data("world2"),
                 aes(x=long, y=lat, group = group),
                 color = '#9c9c9c', fill = '#f3f3f3') +
    ## Second layer: country map
    geom_polygon(data = map_data_es,
                 aes(x=long, y=lat, group = group),
                 color = 'burlywood4', fill = 'blanchedalmond') +
   geom_point(data = mangrove_islands, mapping = aes(x = lon, y = lat), colour = "red") + 
    coord_map() +
    coord_fixed(ratio = 1,
                xlim = c(277.5, 278), #sets limits for x-axis (latitude coordinates) and y-axis (longitude)
                ylim = c(9, 9.5)) +
    ggtitle("Bocas sampling trial map") +
  xlab("Longitude (º)") +
  ylab("Latitude (º)") +
    theme(panel.background =element_rect(fill = 'paleturquoise3'))

```
With the basic map provided by the "maps" package, we don't have enough resolution to see the islands of interest and get clear coastlines. Thus, we'll need a higher resolution map, such as the 'Panama Corregimientos Boundaries 2022" open source one by Milton Solano available on STRI's GIS Data Portal (using version last updated December 6, 2022) - https://stridata-si.opendata.arcgis.com/datasets/SI::panama-corregimientos-boundaries-2022/about

Alternatively, there are also maps on STRI's GIS Data Portal for:
- the geology of Panama - Milton Solano, 2022 (https://stridata-si.opendata.arcgis.com/datasets/SI::geology-of-the-republic-of-panama-1/explore?location=8.418656%2C-80.103084%2C7.97)

- protected areas of Panama (both marine & terrestrial) - Milton Solano (2006, but data updated 2022 - includes Cordillera de Coiba, for instance) - https://stridata-si.opendata.arcgis.com/datasets/SI::panamas-protected-national-parks-layer/explore?location=9.013336%2C-81.702228%2C6.81

- Panama Watersheds (2022), Normal Fault Lines (2022), Interpreted Fault Lines, District Boundaries, Rivers of Panama, 1:50K topographic sheets mosaic (2021), Province boundaries, Vegetation cover time series


--> helpful page on reading in shapefiles into R: https://r-graph-gallery.com/168-load-a-shape-file-into-r.html

```{r read in Panama shapefile, echo = FALSE}
setwd("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/Panama_Corregimientos_Boundaries_2022") #temporarily set working directory to map source file location

Panama_map <- st_read("Corregimientos_2022.shp", quiet = TRUE, stringsAsFactors = FALSE)
Panama_map
```
The Panama map shapefile contains 680 observations (one per corregimiento) of 13 variables (12 fields): variables = OBJECTID, Corregimie, Distrito, Provincia, COD_PROV, COD_DIST, COD_CORR, ID_CORR< Area_SQKM, GlobalID, SHAPE_Leng, SHAPE_Area, geometry.

In the data summary, they note that Panama's number of Corregimientos is rapidly increasing as the country develops and Indigenous communities' territories gain additional official recognition. As such, the map boundaries are already dated: while 680 Corregimientos are included in the 2022 version, there were already 702 in 2020. While having the most up-to-date land boundaries is not of great importance for our current marine mapping purposes, it will be worth regularly revising, especially as new marine protected areas get developed as well. 

```{r plot basic shapefile of Panama to visualize, echo = FALSE}
ggplot(Panama_map) + 
    geom_sf() + 
    coord_sf() #plots outlines of each corregimiento
```
Using ggplot2 for a better map: https://ggplot2-book.org/maps.html

```{r use ggplot2 to make a nicer map of Panama}
## map (sf + ggplot2 )
ggplot() +
    ## First layer: Panama map
    geom_sf(data = Panama_map,
                 color = 'azure4', fill = 'lightcyan3') +
  coord_sf() +
    ggtitle("Map of Panama") +
  xlab("Longitude (º)") +
  ylab("Latitude (º)")
```

Help with creating multiple plots in one: http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/81-ggplot2-easy-way-to-mix-multiple-graphs-on-the-same-page/

```{r create inset map with Coiba and Las Perlas zoomed-in sites}
pa_map_ggplot <- 
  Panama_map %>% 
  ggplot() +
  geom_sf() +
  # Prevent ggplot from slightly expanding the map limits beyond the bounding box of the spatial objects
  coord_sf(expand = FALSE) +
  theme_void() +
  theme(
    legend.justification = c(0, 1),
    legend.position = c(0, .95)
    ) +
  theme(
    text = element_text(family = "Futura-Medium"),
    legend.title = element_text(family = "Futura-Bold", size = 10),
    legend.text = element_text(family = "Futura-Medium", size = 10)
    )  

pa_map_ggplot <- 
  pa_map_ggplot +
  geom_rect(
    xmin = 395000, #coiba box. 50000 m x-axis
    ymin = 796000, #77000 m y-axis
    xmax = 445000,
    ymax = 873000,
    fill = NA, 
    colour = "navy",
    linewidth = 0.6,
    linetype = 3 #dotted line
  ) +
   geom_rect(
    xmin = 695000, #las perlas box. 50000 m x-axis
    ymin = 893500, #77000 m y-axis
    xmax = 745000,
    ymax = 970500,
    fill = NA, 
    colour = "navy",
    linewidth = 0.6
  )
 
 pa_map_ggplot



#Coiba inset
coiba_map <- pa_map_ggplot +
  coord_sf(
    xlim = c(395000, 445000), #50000 m
    ylim = c(796000, 873000), #77000 m
    expand = FALSE
    )
coiba_map

#Las Perlas inset
lasperlas_map <- pa_map_ggplot +
  coord_sf(
    xlim = c(695000, 745000), #50000 m
    ylim = c(893500, 970500), #77000 m
    expand = FALSE
    )
lasperlas_map

#attempt 1 at combining
pa_map_sites <- grid.arrange( 
             coiba_map, lasperlas_map, 
             ncol = 2) # both site maps side-by-side 

grid.arrange(pa_map_ggplot, 
             pa_map_sites, 
             ncol = 1, nrow = 2) # full map up top, both site maps side-by-side 

#attempt 2 at combining
ggarrange(pa_map_ggplot, #first row with full Panama map
          ggarrange(coiba_map, lasperlas_map, #second row with coiba & las perlas inserts
  ncol = 2,
  widths = c(2, 1),
  labels = c("B", "C"), #labels bottom two plots "B" & "C"
  font.label = list(size = 14, color = "black", face = "bold"),
  align = "hv"),
  nrow = 2,
  heights = c(1,2),
  labels = "A", #labels top plot with "A"
  legend = "none"
)

#attempt 3 at combining
#other test to get figures arranged together using cowplot package function 
ggdraw() +
  draw_plot(pa_map_ggplot, x = 0, y = .6, width = 1, height = 0.3) +
  draw_plot(coiba_map, x = 0, y = 0, width = 0.5, height = 0.6) +
  draw_plot(lasperlas_map, x = 0.5, y = 0, width = 0.5, height = 0.6) +
  draw_plot_label(label = c("A", "B", "C"), size = 15,
                  x = c(0, 0, 0.5), y = c(1, 0.6, 0.6))
```
```{r add GPS coordinates of sampling sites to Panama Map (in progress), echo = FALSE}
####issue with mapping this, likely because the shapefile is in meters, not coordinates
###need to adjust lat/long coordinates of boxes & insets

#convert map shapefile to lat/long
Panama_map_latlong <- st_transform(Panama_map, "+proj=longlat +ellps=WGS84 +datum=WGS84")

head(st_coordinates(Panama_map_latlong))

theme_set(theme_bw())


###GPS coordinates
TEP_sampling_sites <- tibble::tribble( 
  ~sampling_site,   ~latitude,     ~longitude,
  "Pacheca", 8.662814, -79.055719, #Las Perlas
  "Chapera", 8.592016, -79.019052, #Las Perlas
  "Saboga",  8.6284317, -79.055171, #Las Perlas
  "Machete", 7.6381651, -81.737106, #Coiba
  "Cocos", 7.6158149, -81.714212, #Coiba
  "Frijol", 7.6546197,  -81.727973, #Coiba
  "Afuera", 7.695941962,   -81.63292903, #Coiba
  "Granito", 7.5939362,   -81.712836, #Coiba
)

###GPS coordinates
Water_sampling_sites <- tibble::tribble( 
  ~sampling_site,   ~latitude,     ~longitude,
  "Afuera", 7.695941962, -81.63292903, #Coiba
  "Damas", 7.402953962, -81.665283, #Coiba
  "Uvas", 7.815128015, -81.75892202, #Coiba
  "Contadora", 8.63245, -79.02834, #Las Perlas
  "Saboga", 8.62807, -79.05462, #Las Perlas
  "MogoMogo", 8.57442, -79.01646, #Las Perlas
)

site_cols <- c( "#FFAD65", "#FF6E00", "#662700", "#99FFFF", "#009999", "#006666",
                "#00CCCC", "#00FFFF", "#993D00", "#FF8E33", "#CC5500", "#FFCA99")

sites <- c("Afuera", "Damas", "Uvas", "Contadora", "Saboga", "MogoMogo", #water sampling
           "Pacheca", "Chapera", "Machete", "Cocos", "Frijol", "Granito" ) #fish only

names(site_cols) = sites


#new plot with lat/long data
pa_map_ggplot2 <- 
  Panama_map_latlong %>% 
  ggplot() +
  geom_sf() +
  # Prevent ggplot from slightly expanding the map limits beyond the bounding box of the spatial objects
  coord_sf(expand = FALSE) +
  theme(
    legend.justification = c(0, 1),
    legend.position = c(0, .95)
    ) 

pa_map_ggplot2 <- 
  pa_map_ggplot2 +
  geom_rect(
    xmin = -81.951016, #coiba box. adjust to lat/long 
    ymin = 7.200294, #adjust to lat/long
    xmax = -81.498966,
    ymax = 7.897507,
    fill = NA, 
    colour = "#662700",
    linewidth = 0.6,
    linetype = 6 #double dashed line
  ) +
   geom_rect(
    xmin = -79.230405, #las perlas box. adjust to lat/long 
    ymin = 8.079409, #adjust to lat/long
    xmax = -78.772893,
    ymax = 8.773159,
    fill = NA, 
    colour = "#7F7D9C",
    linewidth = 0.6
  )
 
 pa_map_ggplot2
 
#Coiba inset
coiba_map2 <- pa_map_ggplot2 +
  geom_point(data = TEP_sampling_sites, aes(x = longitude, y = latitude), size = 4, 
        shape = 23) +
   geom_point(data = Water_sampling_sites, aes(x = longitude, y = latitude), size = 4, 
        shape = 4) +
   scale_fill_manual(values = site_cols) +
  coord_sf(
    xlim = c(-81.951016, -81.498966), #-81.951016, -81.498966
    ylim = c(7.200294, 7.897507), #7.200294, 7.897507
    expand = FALSE
    ) +
  theme(axis.title.x=element_blank(), #remove x axis labels
        axis.title.y=element_blank(),  #remove y axis labels
        )
coiba_map2

#Las Perlas inset
lasperlas_map2 <- pa_map_ggplot2 +
  geom_point(data = TEP_sampling_sites, aes(x = longitude, y = latitude), size = 4, 
        shape = 23) +
   geom_point(data = Water_sampling_sites, aes(x = longitude, y = latitude), size = 3, 
        shape = 4) +
   scale_fill_manual(values = site_cols) +
  coord_sf(
    xlim = c(-79.230405, -78.772893), #-79.230405, -78.772893
    ylim = c(8.079409, 8.773159), #8.079409, 8.773159
    expand = FALSE
    ) +
  theme(axis.title.x=element_blank(), #remove x axis labels
        axis.title.y=element_blank(),  #remove y axis labels
        )
lasperlas_map2

#attempt 1 at combining
pa_map_sites2 <- grid.arrange( 
             coiba_map2, lasperlas_map2, 
             ncol = 2) # both site maps side-by-side 

```

```{r composite map with Coiba and all of Panama}

ggdraw(pa_map_ggplot) +
  draw_plot(
    {
      pa_map_ggplot + 
        coord_sf(
    xlim = c(397000, 442000), #coiba inset coordinates - to have frame match, need to shrink boxes above for inset as well
    ylim = c(795000, 857000), #coiba inset coordinates
    expand = FALSE
    ) +
        theme(legend.position = "none")
      },
    # The distance along a (0,1) x-axis to draw the left edge of the plot
    x = -0.16, 
    # The distance along a (0,1) y-axis to draw the bottom edge of the plot
    y = 0,
    # The width and height of the plot expressed as proportion of the entire ggdraw object
    width = 0.5, 
    height = 0.46) #######need to fix arrangement of these
```
```{r tmap test map}
# Add fill layer to Panama shape
tm_shape(Panama_map) +
  tm_fill() 
# Add border layer to Panama shape
tm_shape(Panama_map) +
  tm_borders() 
# Add fill and border layers to Panama shape. tm_polygon() can be used to condense both tm_fill & tm_borders
tm_shape(Panama_map) +
  tm_fill() +
  tm_borders() 
```
Section 9.2.7 - Inset maps (Geocomputation with R)

```{r tmap save Panama map w/o boundaries}
pa_map = tm_shape(Panama_map) + tm_fill()
pa_map

pa_map +
  tm_compass(type = "8star", size = 3, position = c("right", "top")) + #add compass to map
  tm_scale_bar(breaks = c(0, 50, 100), text.size = 0.5, position = c("left", "bottom")) #add scale bar
```

```{r inset w/ tmap & GPS coords, echo = FALSE}
###used these plots for poster

# Convert data frame to sf object
TEP_sampling_sites.sf <- st_as_sf(x = TEP_sampling_sites, 
                        coords = c("longitude", "latitude"),
                        crs = "+proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0")
Water_sampling_sites.sf <- st_as_sf(x = Water_sampling_sites, 
                        coords = c("longitude", "latitude"),
                        crs = "+proj=longlat +datum=WGS84 +ellps=WGS84 +towgs84=0,0,0")

#Coiba inset - define area
pa_coiba = st_bbox(c(xmin = -81.951016, xmax = -81.498966, 
                      ymin = 7.200294, ymax = 7.897507),
                    crs = st_crs(Panama_map_latlong)) |> 
  st_as_sfc()
#Create inset w/ area defined above
pa_coiba_map = tm_shape(Panama_map_latlong, bbox = pa_coiba) + tm_fill()
  
pa_coiba_map <- pa_coiba_map +
  tm_scale_bar(breaks = c(0, 10, 20), text.size = 0.75, position = c("right", "bottom")) + #add scale bar
    tm_shape(TEP_sampling_sites.sf) + #add circles for GPS coords of fish sampling sites
  tm_symbols(col = "#CC5500", size = 1, shape = 21)

pa_coiba_map

#Las Perlas inset
pa_lasperlas = st_bbox(c(xmin = -79.230405, xmax = -78.772893, 
                      ymin = 8.079409, ymax = 8.773159),
                    crs = st_crs(Panama_map_latlong)) |> 
  st_as_sfc()

#Create inset w/ area defined above
pa_lasperlas_map = tm_shape(Panama_map, bbox = pa_lasperlas) + tm_fill()

pa_lasperlas_map <- pa_lasperlas_map +
  tm_scale_bar(breaks = c(0, 10, 20), text.size = 0.75, position = c("right", "bottom")) + #add scale bar
  tm_shape(TEP_sampling_sites.sf) + #add circles for GPS coords of fish sampling sites
  tm_symbols(col = "#009999", size = 1, shape = 21)

pa_lasperlas_map

```
```{r save inset maps, echo = FALSE}
## save an image ("plot" mode)
tmap_save(pa_lasperlas_map, filename = "Las Perlas Map Inset w GPS Coords.png")

## save an image ("plot" mode)
tmap_save(pa_coiba_map, filename = "Coiba Map Inset w GPS Coords.png")

```

```{r create tmap with GPS coordinates, echo = FALSE}
###used this plot for poster

#using lat/long map -- Panama_map_latlong

#Integrated map w/ insets marked
pa_map2 = tm_shape(Panama_map_latlong) + tm_fill() +
  tm_shape(pa_coiba) + tm_borders(lwd = 3, col = "#662700", lty = "dotted") + #coiba border
  tm_shape(pa_lasperlas) + tm_borders(lwd = 3, col = "#7F7D9C") + #las perlas border
tm_compass(type = "8star", size = 3, position = c("right", "top")) + #add compass to map
  tm_scale_bar(breaks = c(0, 50, 100), text.size = 0.5, position = c("right", "bottom")) #add scale bar

pa_map2


## save an image ("plot" mode)
tmap_save(pa_map2, filename = "Panama Map w Sites.png")

```

```{r create multi-panel map w/ full panama and insets w/ gridExtra, echo = FALSE}

site_maps <- tmap_arrange(pa_coiba_map, pa_lasperlas_map, ncol = 2)
site_maps #multi-panel w/ both study sites

pa_map2 #full map of panama

#at the moment, it seems that tmap has issues creating anything beyond the most basic multipanel figure and is not compatible with other functions/packages such as cowplot - either edit this outside of R, or potentially save individual images, read into R as images, and work from there
```

All the draft maps above still need some visual improvement - final goal would be to have a map of panama showing boxes around our two TEP study sites with a scale bar, compass, and inset map situating Panama within the broader central american context (given that our Panama shapefile does not include other countries, we will need to import mapping data from OpenStreetMaps or another open source world map - alternatively, given that we do not need high resolution for this, it might be possible to get away with using the built in world map in R and overlaying our more detailed Panama map over it). Below this main map, we would have two panels showing the zoomed-in Las Perlas and Coiba maps with red dots marking GPS coordinate points for each of our sample collections sites (that is - LP: Chapera, Pacheca, Saboga, Coiba: Granito, Machete, Cocos, Afuera, Frijol, Frijolito). Also potentially want labels on map marking Tropical Eastern Pacific, Gulf of Panama, Gulf of Chiriquí, and/or ellipses rather than boxes around the two sampling sites, as these more closely match the shapes of the sampling locations. 

####Section 1b - environmental data --> water temperature in Las Perlas & Coiba

Temperature data have been collected by STRI researchers for many years at several different sampling locations across Panama. During our sample collection, Hobo data loggers were placed by the RRR team - special thanks to Anabel Cornejo for collecting and sharing theses data - including at Afuera (GofC), Bahía Damas (GofC - Coiba), Contadora (GofP - next to Saboga), Mogo Mogo (GofP - next to Chapera), Saboga (GofP), and Uvas (GofC - between Coiba Island & Pixvae), providing high-quality water temperature data (measurements taken at 15-30 min intervals) at various depths. 

#full environmental data code found in "ROHR_06_Environmental_Data.Rmd"

code below produces just the final graph from the source data files, can jump down to code chunk that reads in merged temp dataset if already run

setwd("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/RRR_HOBO_Environmental_data")

Markdown document to read, clean, and plot environmental data (water temperature, for now) collected in 2021-22 in Las Perlas and Coiba using HOBO data loggers. Water temperature measurements were taken continuously during deployment at 15 (Las Perlas) to 30 minute (Coiba) intervals. Metadata file for these data = "RRR_watertemp_metadata". Separate files in "RRR_HOBO_Environmental_data" folder for each site (n = 3 per gulf). Water temperature records were retrieved for ~August/September 2021 through March/April 2022. Sensors were deployed at 15-25 ft (4.5 to 7.6 m) depth. Data courtesy of Anabell Cornejo. 

Coiba (Gulf of Chiriqui) sites:
    Bahia Damas ("Damas_RRR1"), lat: 7.402953962, long: -81.665283, date range: 2021-08-20 to	2022-04-19
    Canal de Afuera ("Afuera_RRR1"), lat: 7.695941962, long:	-81.63292903, date range: 2021-08-19 to	2022-04-19
    Uva ("Uvas_RRR1"), lat: 7.815128015, long:	-81.75892202, date range: 2021-08-22 to	2022-04-21
    
Las Perlas (Gulf of Panama) sites: 
    Contadora ("Contadora_RRR1"), lat: 8.63245, long:	-79.02834, date range: 2021-08-30	to 2022-03-22
    Saboga (Saboga_RRR1"), lat: 8.62807, long:	-79.05462, date range: 2021-08-30	to 2022-04-05
    Mogo Mogo ("MogoM_RRR1"), lat: 8.57442, long:	-79.01646 2021-09-01 to	2022-03-25
    
Headers to files: Gulf (Chiriqui or Panama),	Site (Damas, Afuera, Uvas, Contadora, Saboga, MogoMogo)	Num (measurement #),	Date (YYYY-MM-DD),	Time (GMT-5; 24h time),	Temp (ºC)

```{r load required packages, echo = FALSE}
library(tidyverse) #data manipulation
library(ggplot2) #plotting & visualization
library(lubridate) #for date-time column
library(tidyr) #merging columns and manipulating data
library(colorBlindness) #check color choices for color blind accessibility
library(tidyquant) #smoothing temp data
```

Pre-processing of raw data files included merging the datasets from 2021 & early 2022 with the next 2022 datasets (raw data files denoted by a "_NEW" ending, to distinguish from earlier data files), splitting the date:time column into date and time for ease of reading (but this is a useless step given that we're going to merge them again in R...), making the date format match what R likes (Y%-m%-d%), and adjusting the "Num" column of the new dataset to be consecutive, as it appears that this column is reset for each logger data download (i.e., same number appears multiple times in the two different datasets), thus if the original reference number is needed, we have to go back to the unmodified source file (e.g., "Afuera_RRR1_NEW"). 

```{r read in water temp data, echo = FALSE}
#first check WD set to correct location
print(getwd())
#in this case WD should be:
#"/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/RRR_ HOBO_Environmental_data"

#Coiba water data - measurements taken in 30 min intervals
Afuera <- read.csv("Afuera_RRR1.csv", colClasses = c("character", "character", "character", "Date", "character", "numeric")) # covers 8/19/21 to 11/16/22
Damas <- read.csv("Damas_RRR1.csv", colClasses = c("character", "character", "character", "Date", "character", "numeric")) # covers 8/20/21 to 11/16/22 : small data gap between 4-19-22 9 AM and 4-20-22 8:30 AM)
Uvas <- read.csv("Uvas_RRR1.csv", colClasses = c("character", "character", "character", "Date", "character", "numeric")) # covers 8/22/21 to 4/21/22 (have data for Nov 2022 onwards, missing Apr - Nov 2022)

#Las Perlas water data - measurements taken in 15 min intervals
Contadora <- read.csv("Contadora_RRR1.csv", colClasses = c("character", "character", "character", "Date", "character", "numeric")) # covers 8/30/21 to 8/2/22
Saboga <- read.csv("Saboga_RRR1.csv", colClasses = c("character", "character", "character", "Date", "character", "numeric")) # covers 09/01/2021 to 08-05-2022
MogoMogo <- read.csv("MogoM_RRR1.csv", colClasses = c("character", "character", "character", "Date", "character", "numeric")) # covers 9/1/21 to 8/5/22
```

```{r summary of read data}
summary(Afuera)
```

```{r remove NA rows, echo = FALSE}
#remove extra rows from Contadora dataset
Contadora <- na.omit(Contadora)
```

``` {r create a date:time category for Afuera, echo = FALSE}
#merge date & time into a single, new column
Afuera <- Afuera %>%
  unite("Date_Time", Date:Time, sep = " ", remove = FALSE)

Afuera #check this worked - yes! now have a "Date_Time" column (class = character)

# check data types (and contents) of data frame
# date & time read in as characters, temp as numeric --> need to change these

Afuera$Date_Time <- as.POSIXct(Afuera$Date_Time, tz = "America/Panama", format = "%Y-%m-%d %H:%M")

Afuera$Date <- as.Date(Afuera$Date) #this worked!!

Afuera

```

``` {r create a date:time category for Contadora, echo = FALSE}
#merge date & time into a single, new column
Contadora <- Contadora %>%
  unite("Date_Time", Date:Time, sep = " ", remove = FALSE)

Contadora #check this worked - yes! now have a "Date_Time" column (class = character)

# check data types (and contents) of data frame
# date & time read in as characters, temp as numeric --> need to change these

Contadora$Date_Time <- as.POSIXct(Contadora$Date_Time, tz = "America/Panama", format = "%Y-%m-%d %H:%M")

Contadora$Date <- as.Date(Contadora$Date) #this worked!!

Contadora

```

``` {r create a date:time category for Damas, echo = FALSE}
#merge date & time into a single, new column
Damas <- Damas %>%
  unite("Date_Time", Date:Time, sep = " ", remove = FALSE)

Damas #check this worked - yes! now have a "Date_Time" column (class = character)

# check data types (and contents) of data frame
# date & time read in as characters, temp as numeric --> need to change these

Damas$Date_Time <- as.POSIXct(Damas$Date_Time, tz = "America/Panama", format = "%Y-%m-%d %H:%M")

Damas$Date <- as.Date(Damas$Date) #this worked!!

Damas

```

``` {r create a date:time category for MogoMogo, echo = FALSE}
#merge date & time into a single, new column
MogoMogo <- MogoMogo %>%
  unite("Date_Time", Date:Time, sep = " ", remove = FALSE)

MogoMogo #check this worked - yes! now have a "Date_Time" column (class = character)

# check data types (and contents) of data frame
# date & time read in as characters, temp as numeric --> need to change these

MogoMogo$Date_Time <- as.POSIXct(MogoMogo$Date_Time, tz = "America/Panama", format = "%m/%d/%Y %H:%M") #note that specified format for date/time is different in the MogoMogo source file, but can be standardized by specifying initial format 

MogoMogo$Date <- as.Date(MogoMogo$Date, format = "%m/%d/%Y") #this worked!!

MogoMogo

```

``` {r create a date:time category for Saboga, echo = FALSE}
#merge date & time into a single, new column
Saboga <- Saboga %>%
  unite("Date_Time", Date:Time, sep = " ", remove = FALSE)

Saboga #check this worked - yes! now have a "Date_Time" column (class = character)

# check data types (and contents) of data frame
# date & time read in as characters, temp as numeric --> need to change these

Saboga$Date_Time <- as.POSIXct(Saboga$Date_Time, tz = "America/Panama", format = "%Y-%m-%d %H:%M")

Saboga$Date <- as.Date(Saboga$Date) #this worked!!

Saboga

```

``` {r create a date:time category for Uvas, echo = FALSE}
#merge date & time into a single, new column
Uvas <- Uvas %>%
  unite("Date_Time", Date:Time, sep = " ", remove = FALSE)

Uvas #check this worked - yes! now have a "Date_Time" column (class = character)

# check data types (and contents) of data frame
# date & time read in as characters, temp as numeric --> need to change these

Uvas$Date_Time <- as.POSIXct(Uvas$Date_Time, tz = "America/Panama", format = "%Y-%m-%d %H:%M")

Uvas$Date <- as.Date(Uvas$Date) #this worked!!

Uvas

```

```{r merge all 6 sites temp data, echo = FALSE}
temp <- bind_rows(Afuera, Damas, Uvas, Contadora, Saboga, MogoMogo, .id = NULL)
temp
```

###saving & reading in full merged temperature dataset
```{r save merged temperature dataset, echo = FALSE}
saveRDS(temp, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/RRR_HOBO_Environmental_data/water_temp_data_coiba_lasperlas_21_22.rds")

#read in merged data, if not running code from start
temp <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/RRR_HOBO_Environmental_data/water_temp_data_coiba_lasperlas_21_22.rds")
```

```{r put best graph elements so far together with improvements for poster, echo = FALSE}
#assign colors to sites
Blue2DarkOrange18Steps #color blind friendly palette

site_cols <- c( "#FFAD65", "#FF6E00", "#662700", "#99FFFF", "#009999", "#006666")

names(site_cols) = unique(temp$Site)

site_cols

#set theme
theme_set(theme_bw())

#reorder sites to group by gulf
temp$Site <- factor(temp$Site, levels = c('Afuera', 'Damas', 'Uvas', 'Contadora', 'Saboga', 'MogoMogo'))


#temperature graph with bars for sampling season 
#vertical dashed lines mark start & end dates of sampling (slightly darker for upwelling)
#scales set to months
#simple moving average plotted overtop of raw temperature data from all 6 sites in both gulfs

water_temp_final2 <- ggplot(temp, aes(Date_Time, Temp, colour = Site)) +
  annotate("rect", fill = "#660066", alpha = 0.2, 
        xmin = as.POSIXct("2021-10-15 06:00:00"), xmax = as.POSIXct("2021-12-03 00:00:00"),
        ymin = -Inf, ymax = Inf) +
  annotate("rect", fill = "#336633", alpha = 0.2, 
        xmin = as.POSIXct("2022-03-11 06:00:00"), xmax = as.POSIXct("2022-04-29 00:00:00"),
        ymin = -Inf, ymax = Inf) +
  geom_vline(xintercept = as.POSIXct("2021-10-15 06:00:00", tz = "America/Panama"),
             linetype = 6, color = "#601A35", linewidth = 1) + #add sampling start: non-upwelling
   geom_vline(xintercept = as.POSIXct("2021-12-03 00:00:00", tz = "America/Panama"),
             linetype = 6, color = "#601A35", linewidth = 1) + #add sampling end: non-upwelling
    geom_vline(xintercept = as.POSIXct("2022-03-11 06:00:00", tz = "America/Panama"),
             linetype = 6, color = "#333300", linewidth = 1) + #add sampling start: upwelling
    geom_vline(xintercept = as.POSIXct("2022-04-29 00:00:00", tz = "America/Panama"),
             linetype = 6, color = "#333300", linewidth = 1) + #add sampling end: upwelling - actual end date is 2021-04-29, but current data only runs through 4-21-22
          scale_color_manual(values = site_cols) +
  geom_line(na.rm = TRUE, linewidth = 1, alpha = 0.3) + # define line width and transparency of raw temp data
  geom_ma(ma_fun = SMA, n = 96, linetype = 1, size = 2) +  # Plot SMA (standard moving average) across 96 timepoints (given 15 to 30 min intervals, this means calculating the average over 24 to 48 hours)
  coord_x_datetime(xlim = c("2021-10-01 00:00:00", "2022-04-30 00:00:00")) + # Zoom in
  ylab("Water Temperature (ºC)") +
  xlab("Date (months)") +
  scale_fill_discrete(breaks=c('Afuera', 'Damas', 'Uvas', 'Contadora', 'Saboga', 'MogoMogo')) +
  scale_x_datetime(date_breaks = 'month', date_labels = "%b") +
  theme(axis.text.x = element_text(size = 16), #change linewidth of months (x-axis)
        axis.title.x = element_text(size = 18), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 16), #change size of y-axis #s
        axis.title.y = element_text(size = 18), #change size of y-axis title
        legend.title = element_text(size = 18), #change size of legend title
        legend.text = element_text(size = 16), #change size of legend text
        legend.position = c(0.85, 0.15), #make legend inside plot
        legend.title.align = 0.25) #adjust legend title position

water_temp_final2

```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Water Temperature Data 21-22 V3 pdf", plot = water_temp_final2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

```{r code for exporting plot above as a jpg}
ggsave(filename = "Water Temperature Data 21-22 V3", plot = water_temp_final2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='png', dpi=700)
```

####Section 2 - Species by Species ordination plots

"Using cowplot to construct multi-panel figures entirely via code" by Baliga, Gaede, and Senthivasan (v last updated 2019-11-29) https://beep-boopr.github.io/ubcBIOL548L/articles/group-03_Using-cowplot-multi-panel.html

--> this page provides instructions for creating multi-panel plots with (1) icons imported from PNG files, (2) figure legends, (3) adding labels (e.g., "A") to panels, (4) adjusting font, (5) multi-panel plots with figures, maps, images, etc.

Loading our phyloseq object - filtered & trimmed (not rarefied) with phylogenetic tree - from "ROHR_06_Swab_stats_alpha.Rmd"
```{r read phyloseq object, echo=FALSE}
ROHR_06_obj_phylo_root <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_with_rooted_phylogenetic_tree_updated_names.rds")
```

## Water microbiome samples for comparison to fish skin
Loading our phyloseq object - filtered & trimmed (not rarefied) with phylogenetic tree - from "ROHR_08_Water_stats_alpha.Rmd"
```{r read phyloseq object - water, echo=FALSE}
ROHR_08_obj_phylo_root <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_08_trimmed/ROHR_08_with_phylogenetic_tree_rooted.rds")
```
Loading the split phyloseq objects from the initial ROHR_06_phylo_root --> split by host species (see ROHR_06_Swab_Species_by_Species)
```{r read split phyloseqs by species, echo=FALSE}
#A. concolor
ROHR_06_Aconcolor <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_concolor.rds")

#A. troschelii
ROHR_06_Atroschelii <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_troschelii.rds")

#A. xanthopterus
ROHR_06_Axanthopterus <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_xanthopterus.rds")

#C. colonus
ROHR_06_Ccolonus <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_colonus.rds")

#C. panamensis
ROHR_06_Cpanamensis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_panamensis.rds")

#C. humeralis
ROHR_06_Chumeralis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_humeralis.rds")

#E. labriformis
ROHR_06_Elabriformis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_E_labriformis.rds")

#J. nigrirostris
ROHR_06_Jnigrirostris <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_J_nigrirostris.rds")

#M. dorsalis
ROHR_06_Mdorsalis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_M_dorsalis.rds")

#P. laticlavius
ROHR_06_Platiclavius <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_laticlavius.rds")
```

```{r load packages for multi-panel plots, echo = FALSE}
packages <-
  c("gapminder",
    "ggplot2",
    "tidyr",
    "dplyr",
    "tibble",
    "readr",
    "forcats",
    "readxl",
    "ggthemes",
    "magick",
    "grid",
    "cowplot",
    "phyloseq", #to work with phyloseq objects (microbiome data)
    # ggmap and maps are optional; needed for creating maps
    "ggmap",
    "maps",
    "png" ) #to read in PNG image icons
## Now load or install&load all packages (we've already loaded some of these for maps above)
package.check <- lapply(
  packages,
  FUN = function(x)
  {
    if (!require(x, character.only = TRUE))
    {
      install.packages(x, dependencies = TRUE,
                       repos = "http://cran.us.r-project.org")
      library(x, character.only = TRUE)
    }
  }
)
library(phylosmith) # package to create merged meta-data column
library(ggtext) #improved text rendering for ggplot2 (allows more customization of fonts, text, etc.)
```

#PCoAs 

```{r run species-by-species PCoAs, echo = FALSE}
#take code from "ROHR_06_Swab_Species_by_Species" to save all PCoAs at once before creating multiplot

theme_set(theme_bw())

# set colors for seasons - "#660066" = non-upwelling, "#336633" = upwelling
c2 <- c("#660066", "#336633")

ROHR_06_Aconcolor_phylum <- ROHR_06_Aconcolor %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season) 

unique(ROHR_06_Aconcolor_phylum$season)

seasons <- c("non-upwelling", "upwelling")

names(c2) = seasons

print(c2)


#### Abudefduf concolor ####

##transforming to counts
ROHR_06_Aconcolor_prop = transform_sample_counts(ROHR_06_Aconcolor, function(x) x/sum(x))

# Ordinate
Aconc_pcoa <- ordinate(
  physeq = ROHR_06_Aconcolor_prop, 
  method = "PCoA", 
  distance = "bray"
  )
# Plot - color = season, shape = region (coiba or las perlas)
Aconc <- plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

Aconc

#### Abudefduf troschelii ####

##transforming to counts
ROHR_06_Atroschelii_prop = transform_sample_counts(ROHR_06_Atroschelii, function(x) x/sum(x))

# Ordinate
Atro_pcoa <- ordinate(
  physeq = ROHR_06_Atroschelii_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Atro <- plot_ordination(
  physeq = ROHR_06_Atroschelii_prop,
  ordination = Atro_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
#### Acanthurus xanthopterus ####

##transforming to counts
ROHR_06_Axanthopterus_prop = transform_sample_counts(ROHR_06_Axanthopterus, function(x) x/sum(x))

# Ordinate
Axan_pcoa <- ordinate(
  physeq = ROHR_06_Axanthopterus_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Axan <- plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop,
  ordination = Axan_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Cephalopholis colonus ####

##transforming to counts
ROHR_06_Ccolonus_prop = transform_sample_counts(ROHR_06_Ccolonus, function(x) x/sum(x))

# Ordinate
Ccol_pcoa <- ordinate(
  physeq = ROHR_06_Ccolonus_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Ccol <- plot_ordination(
  physeq = ROHR_06_Ccolonus_prop,
  ordination = Ccol_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
#### Cephalopholis panamensis ####

##transforming to counts
ROHR_06_Cpanamensis_prop = transform_sample_counts(ROHR_06_Cpanamensis, function(x) x/sum(x))

# Ordinate
Cpan_pcoa <- ordinate(
  physeq = ROHR_06_Cpanamensis_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Cpan <- plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop,
  ordination = Cpan_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
#### Chaetodon humeralis ####

##transforming to counts
ROHR_06_Chumeralis_prop = transform_sample_counts(ROHR_06_Chumeralis, function(x) x/sum(x))

# Ordinate
Chum_pcoa <- ordinate(
  physeq = ROHR_06_Chumeralis_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Chum <- plot_ordination(
  physeq = ROHR_06_Chumeralis_prop,
  ordination = Chum_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Epinephelus labriformis ####

##transforming to counts
ROHR_06_Elabriformis_prop = transform_sample_counts(ROHR_06_Elabriformis, function(x) x/sum(x))

# Ordinate
Elab_pcoa <- ordinate(
  physeq = ROHR_06_Elabriformis_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Elab<- plot_ordination(
  physeq = ROHR_06_Elabriformis_prop,
  ordination = Elab_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Johnrandallia nigrirostris ####

##transforming to counts
ROHR_06_Jnigrirostris_prop = transform_sample_counts(ROHR_06_Jnigrirostris, function(x) x/sum(x))

# Ordinate
Jnig_pcoa <- ordinate(
  physeq = ROHR_06_Jnigrirostris_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Jnig <- plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop,
  ordination = Jnig_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Microspathodon dorsalis ####

##transforming to counts
ROHR_06_Mdorsalis_prop = transform_sample_counts(ROHR_06_Mdorsalis, function(x) x/sum(x))

# Ordinate
Mdor_pcoa <- ordinate(
  physeq = ROHR_06_Mdorsalis_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Mdor <- plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop,
  ordination = Mdor_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Prionurus laticlavius ####

##transforming to counts
ROHR_06_Platiclavius_prop = transform_sample_counts(ROHR_06_Platiclavius, function(x) x/sum(x))

# Ordinate
Plat_pcoa <- ordinate(
  physeq = ROHR_06_Platiclavius_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Plat <- plot_ordination(
  physeq = ROHR_06_Platiclavius_prop,
  ordination = Plat_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Water Microbiome ####

####### Transform counts to relative abundance per sample
ROHR_08_obj_clean_prop = transform_sample_counts(ROHR_08_obj_phylo_root, function(x) x/sum(x))

# Ordinate
Water_pcoa <- ordinate(
  physeq = ROHR_08_obj_clean_prop, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Water <- plot_ordination(
  physeq = ROHR_08_obj_clean_prop,
  ordination = Water_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
```

```{r fish icons and color codes for species}
#set working directory to folder where fish images are stored for this chunk
setwd("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft")

## fish icons - B&W - save these as .png files to call
icon_Aconc <- readPNG("Atroschelii.png") #using same genus-level silhouettes for now as placeholders
icon_Atro <- readPNG("Atroschelii.png")
icon_Axan <- readPNG("Axanthopterus.png")
icon_Chum <- readPNG("Chumeralis.png")
icon_Cpan <- readPNG("Cpanamensis.png")
icon_Elab <- readPNG("Cpanamensis.png")
icon_Jnig <- readPNG("Chumeralis.png")
icon_Mdor <- readPNG("Atroschelii.png")
icon_Ccol <- readPNG("Ccolonus.png")
icon_Plat <- readPNG("Axanthopterus.png")
icon_Water <- readPNG("Water1.png")

## fish icons - color-coded by species
icon_Aconc1 <- readPNG("AconcolorC.png") #using same genus-level silhouettes for now as placeholders
icon_Atro1 <- readPNG("AtroscheliiC.png")
icon_Axan1 <- readPNG("AxanthopterusC.png")
icon_Chum1 <- readPNG("ChumeralisC.png")
icon_Cpan1 <- readPNG("CpanamensisC.png")
icon_Elab1 <- readPNG("ElabriformisC.png")
icon_Jnig1 <- readPNG("JnigrirostrisC.png")
icon_Mdor1 <- readPNG("MdorsalisC.png")
icon_Ccol1 <- readPNG("CcolonusC.png")
icon_Plat1 <- readPNG("PlaticlaviusC.png")
icon_Water1 <- readPNG("Water1C.png")

## fish icons - color-coded by species w/ black outlines
icon_Aconc2 <- readPNG("AconcolorC_outline.png") #using same genus-level silhouettes for now as placeholders
icon_Atro2 <- readPNG("AtroscheliiC_outline.png")
icon_Axan2 <- readPNG("AxanthopterusC_outline.png")
icon_Chum2 <- readPNG("ChumeralisC_outline.png")
icon_Cpan2 <- readPNG("CpanamensisC_outline.png")
icon_Elab2 <- readPNG("ElabriformisC_outline.png")
icon_Jnig2 <- readPNG("JnigrirostrisC_outline.png")
icon_Mdor2 <- readPNG("MdorsalisC_outline.png")
icon_Ccol2 <- readPNG("CcolonusC_outline.png")
icon_Plat2 <- readPNG("PlaticlaviusC_outline.png")
icon_Water2 <- readPNG("Water1C_outline.png")

## Colors for the different fish trophic levels - roving herbivore "#354A21", territorial herbivore "#74B72E", invertivore "#1F456E", omnivore "#EC9706", planktivore "#882255", carnivore "#893101", water "#BE93D4"

col_Aconc <- "#74B72E" #territorial herbivore
col_Atro <- "#EC9706" #omnivore
col_Axan <- "#354A21" #roving herbivore
col_Chum <- "#1F456E" #invertivore
col_Cpan <- "#893101" #carnivore
col_Elab <- "#893101" #carnivore
col_Jnig <- "#1F456E" #invertivore
col_Mdor <- "#74B72E" #territorial herbivore
col_Ccol <- "#882255" #planktivore
col_Plat <- "#354A21" #roving herbivore
col_Water <- "#BE93D4" #water

##colors for individual fish species
col_AconcS <- "aquamarine4"
col_AtroS <- "darkgoldenrod1"
col_AxanS <- "chartreuse3"
col_ChumS <- "deepskyblue2"
col_CpanS <- "chocolate4" 
col_ElabS <- "chocolate2"
col_JnigS <- "darkblue" 
col_MdorS <- "turquoise"
col_CcolS <- "darkmagenta"
col_PlatS <- "darkgreen"
col_WaterS <- "#BE93D4"
  
## Rectangles within legends - **have not used this code yet** (as of 3/3/23)
rect_Aconc   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_AconcS))
rect_Acon_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Aconc, alpha = 0.5)
             )
rect_Atro   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_AtroS))
rect_Atro_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Atro, alpha = 0.5)
             )
rect_Axan   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_AxanS))
rect_Axan_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Axan, alpha = 0.5)
             )
rect_Chum   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_ChumS))
rect_Chum_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Chum, alpha = 0.5)
             )
rect_Cpan   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_CpanS))
rect_Cpan_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Cpan, alpha = 0.5)
             )
rect_Elab   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_ElabS))
rect_Elab_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Elab, alpha = 0.5)
             )
rect_Jnig   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_JnigS))
rect_Jnig_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Jnig, alpha = 0.5)
             )
rect_Mdor   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_MdorS))
rect_Mdor_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Mdor, alpha = 0.5)
             )
rect_Ccol   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_CcolS))
rect_Ccol_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Ccol, alpha = 0.5)
             )
rect_Plat   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_PlatS))
rect_Plat_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Plat, alpha = 0.5)
             )
rect_Water   <- rectGrob(width = 2, height = 1, gp = gpar(fill = col_WaterS))
rect_Water_t <- rectGrob(
               width = 2,
               height = 1,
               gp = gpar(fill = col_Water, alpha = 0.5)
             )
#points - **have not used this code yet** (as of 3/3/23)
#point shapes correspond to family (e.g., square = damselfish, triangle = acanthuridae, circle = butterflyfish, diamond = groupers)
point_Aconc  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 15, #filled square
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_AconcS)
             )
point_Atro  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 15, #filled square
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_AtroS)
             )
point_Axan  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 17, #filled triangle
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_AxanS)
             )
point_Chum  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 19, #filled circle
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_ChumS)
             )
point_Cpan  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 18, #filled diamond
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_CpanS)
             )
point_Elab  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 18, #filled diamond
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_ElabS)
             )
point_Jnig  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 19, #filled circle
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_JnigS)
             )
point_Mdor  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 15, #filled square
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_MdorS)
             )
point_Ccol  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 18, #filled diamond
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_CcolS)
             )
point_Plat  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 17, #filled triangle
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_PlatS)
             )
point_Water  <- pointsGrob(
               x = 0,
               y = 0,
               pch = 8, #star for water
               size = unit(0.5, units = "char"),
               gp = gpar(fill = col_WaterS)
             )
```

```{r basic combined PCoA plot, echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water
plot_grid(Aconc, Mdor, Axan, Plat, Jnig, Chum, Ccol, Atro, Cpan, Elab, Water)
```
```{r create 12th plot to contain legend for multipanel figure}

#run one of the ordinations above, including legend
plotlegend <- plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_pcoa,
  color = "season",
  shape = "region",
) +
  scale_color_manual(values = c2,
                     labels = c("Wet", "Dry")) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(color = "grey90", size = 1) +
  labs(shape = "Region", color = "Season") #change names of legend titles to capitalize
# + theme(
   # legend.justification = c("right")) #shift legend to right of panel (to accommodate fish icons)  

legendAC <- cowplot::get_legend(plotlegend)
cowplot::ggdraw(legendAC)
```

```{r combined PCoA plot - labeled w/ legend, echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
plot_grid(Aconc, Mdor, Axan, Plat, Jnig, Chum, Ccol, Atro, Cpan, Elab, Water, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)
```
```{r combined PCoA plot - w/ icons (in progress), echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#note that intervals of 4X3 grids are ~0.25 at this scale size-to-side, & ~0.3 vertically

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA <- plot_grid(Aconc, Mdor, Axan, Plat, Jnig, Chum, Ccol, Atro, Cpan, Elab, Water, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

multi_PCoA_1 <- ggdraw() +
  draw_plot(multi_PCoA) +
  draw_image(image = icon_Aconc, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water, x =  0.070, y = -0.20, scale = 0.060) # Panel K

multi_PCoA_1
```

Paused here 3/2/23
--> Will need to modify initial PCoA plots to adjust individual plot legends and pick easier to distinguish colors/shapes for seasons & regions --> then add in a new legend for the whole plot, as sites & seasons will remain the same across all plots
--> also consider making axes scales match, as they are currently different (range b/w -0.6 and 0.6)

3/3/23 --> added color-coded boxes around plots (less professional, but may help serve as visual link b/w PCoAs and alpha/beta diversity plots shaded in same colors), ran code to adjust axes for all plots to make axis scales match (see chunks below)

```{r add color-coded boxes to each PCoA plot for species, echo = FALSE}
## to change color of axes instead of whole box around plot, use code below
# pcoa_plot + theme(axis.line = element_line(color = "red", linewidth = 1))
# other alternative = changing image icon source files to match colors instead of B&W

AconcC <- Aconc + theme(panel.border = element_rect(color = "aquamarine4", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
AtroC <- Atro + theme(panel.border = element_rect(color = "darkgoldenrod1", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
AxanC <- Axan + theme(panel.border = element_rect(color = "chartreuse3", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
ChumC <- Chum + theme(panel.border = element_rect(color = "deepskyblue2", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
CpanC <- Cpan + theme(panel.border = element_rect(color = "chocolate4", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
ElabC <- Elab + theme(panel.border = element_rect(color = "chocolate2", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
JnigC <- Jnig + theme(panel.border = element_rect(color = "darkblue", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
MdorC <- Mdor + theme(panel.border = element_rect(color = "aquamarine2", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
CcolC <- Ccol + theme(panel.border = element_rect(color = "darkmagenta", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
PlatC <- Plat + theme(panel.border = element_rect(color = "darkgreen", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
WaterC <- Water + theme(panel.border = element_rect(color = "#BE93D4", fill = NA, linewidth = 2),
                   plot.margin = unit(c(.25,0.25,0.25,0.25), "cm"))
```


Alternative to full color box = just coloring axes: less colorful, busy result
```{r add color-coded axes to each PCoA plot for species, echo = FALSE}
## to change color of axes instead of whole box around plot, use code below
# pcoa_plot + theme(axis.line = element_line(color = "red", linewidth = 1))
# other alternative = changing image icon source files to match colors instead of B&W

Aconc1 <- Aconc + theme(axis.line = element_line(color = "aquamarine4", linewidth = 1))
                   
Atro1 <- Atro + theme(axis.line = element_line(color = "darkgoldenrod1", linewidth = 1))
                   
Axan1 <- Axan + theme(axis.line = element_line(color = "chartreuse3", linewidth = 1))
                   
Chum1 <- Chum + theme(axis.line = element_line(color = "deepskyblue2", linewidth = 1))
                   
Cpan1 <- Cpan + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
Elab1 <- Elab + theme(axis.line = element_line(color = "chocolate2", linewidth = 1))
                   
Jnig1 <- Jnig + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
Mdor1 <- Mdor + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
Ccol1 <- Ccol + theme(axis.line = element_line(color = "darkmagenta", linewidth = 1))
                   
Plat1 <- Plat + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
Water1 <- Water + theme(axis.line = element_line(color = "#BE93D4", linewidth = 1))
                   
```

```{r combined PCoA plot - w/ icons & color codes - boxes, echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA <- plot_grid(AconcC, MdorC, AxanC, PlatC, JnigC, ChumC, CcolC, AtroC, CpanC, ElabC, WaterC, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

#add icons to graph
multi_PCoA_1 <- ggdraw() +
  draw_plot(multi_PCoA) +
  draw_image(image = icon_Aconc, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water, x =  0.070, y = -0.20, scale = 0.060) # Panel K

multi_PCoA_1
```
```{r combined PCoA plot - w/ icons & color codes - axis only, echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA_bis <- plot_grid(Aconc1, Mdor1, Axan1, Plat1, Jnig1, Chum1, Ccol1, Atro1, Cpan1, Elab1, Water1, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

#add icons to graph
multi_PCoA_bis2 <- ggdraw() +
  draw_plot(multi_PCoA_bis) +
  draw_image(image = icon_Aconc, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water, x =  0.070, y = -0.20, scale = 0.060) # Panel K

multi_PCoA_bis2
```


```{r adjust axes so that they are consistent across PCoA plots}
#check that this is a valid thing to do and will not alter results shown
# across our plots: x-axis ranges from -0.6 to 0.5, y-axis ranges from -0.6 to ~ 0.6

AconcS <- Aconc + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
AtroS <- Atro + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
AxanS <- Axan + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
ChumS <- Chum + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
CpanS <- Cpan + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
ElabS <- Elab + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
JnigS <- Jnig + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
MdorS <- Mdor + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
CcolS <- Ccol + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
PlatS <- Plat + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
WaterS <- Water + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))

multi_PCoA_2 <- plot_grid(AconcS, MdorS, AxanS, PlatS, JnigS, ChumS, CcolS, AtroS, CpanS, ElabS, WaterS, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

#add icons to graph
multi_PCoA_3 <- ggdraw() +
  draw_plot(multi_PCoA_2) +
  draw_image(image = icon_Aconc2, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor2, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan2, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat2, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol2, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro2, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan2, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab2, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water2, x =  0.070, y = -0.20, scale = 0.060) # Panel K
multi_PCoA_3
```
```{r combined PCoA plot - w/ icons & color-coded fish + axis color codes - axes adjusted, echo = FALSE}
#check that this is a valid thing to do and will not alter results shown
# across our plots: x-axis ranges from -0.6 to 0.5, y-axis ranges from -0.6 to ~ 0.6

Aconc1S <- Aconc1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Atro1S <- Atro1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Axan1S <- Axan1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Chum1S <- Chum1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Cpan1S <- Cpan1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Elab1S <- Elab1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Jnig1S <- Jnig1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Mdor1S <- Mdor1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Ccol1S <- Ccol1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Plat1S <- Plat1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Water1S <- Water1 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))


#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA_tris <- plot_grid(Aconc1S, Mdor1S, Axan1S, Plat1S, Jnig1S, Chum1S, Ccol1S, Atro1S, Cpan1S, Elab1S, Water1S, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

#add icons to graph
multi_PCoA_tris2 <- ggdraw() +
  draw_plot(multi_PCoA_tris) +
  draw_image(image = icon_Aconc2, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor2, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan2, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat2, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol2, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro2, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan2, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab2, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water2, x =  0.070, y = -0.20, scale = 0.060) # Panel K

multi_PCoA_tris2
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by species", plot = multi_PCoA_tris2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

```{r add color-coded axes to each PCoA plot for trophic group rather than species, echo = FALSE}
## to change color of axes instead of whole box around plot, use code below
# pcoa_plot + theme(axis.line = element_line(color = "red", linewidth = 1))
# other alternative = changing image icon source files to match colors instead of B&W

Aconc2 <- Aconc + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
Atro2 <- Atro + theme(axis.line = element_line(color = "darkgoldenrod1", linewidth = 1))
                   
Axan2 <- Axan + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
Chum2 <- Chum + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
Cpan2 <- Cpan + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
Elab2 <- Elab + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
Jnig2 <- Jnig + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
Mdor2 <- Mdor + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
Ccol2 <- Ccol + theme(axis.line = element_line(color = "darkmagenta", linewidth = 1))
                   
Plat2 <- Plat + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
Water2 <- Water + theme(axis.line = element_line(color = "#BE93D4", linewidth = 1))
                   
```

```{r same PCoA plot but with axes coded by trophic group rather than individual species, echo = FALSE}
# across our plots: x-axis ranges from -0.6 to 0.5, y-axis ranges from -0.6 to ~ 0.6

Aconc2S <- Aconc2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Atro2S <- Atro2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Axan2S <- Axan2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Chum2S <- Chum2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Cpan2S <- Cpan2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Elab2S <- Elab2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Jnig2S <- Jnig2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Mdor2S <- Mdor2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Ccol2S <- Ccol2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Plat2S <- Plat2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Water2S <- Water2 + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))


#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA_troph <- plot_grid(Aconc2S, Mdor2S, Axan2S, Plat2S, Jnig2S, Chum2S, Cpan2S, Elab2S, Ccol2S, Atro2S, Water2S, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

#add icons to graph
multi_PCoA_troph2 <- ggdraw() +
  draw_plot(multi_PCoA_troph) +
  draw_image(image = icon_Aconc2, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor2, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan2, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat2, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Cpan2, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Elab2, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Ccol2, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Atro2, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water2, x =  0.070, y = -0.20, scale = 0.060) # Panel K

multi_PCoA_troph2
```

```{r code for exporting plot above as a png}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by trophic group.png", plot = multi_PCoA_troph2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='png', dpi=700)
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by trophic group.pdf", plot = multi_PCoA_troph2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

#### testing vertical 2x6 layout
```{r 2x6 vertical layout}

#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA_troph.1 <- plot_grid(Aconc2S, Mdor2S, Axan2S, Plat2S, Jnig2S, Chum2S, Cpan2S, Elab2S, Ccol2S, Atro2S, Water2S, legendAC,
          ncol = 2,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)
multi_PCoA_troph.1

#add icons to graph
multi_PCoA_troph3 <- ggdraw() +
  draw_plot(multi_PCoA_troph.1) +
  draw_image(image = icon_Aconc2, x = -0.36, y =  0.48, scale = 0.080) +# Panel A
  draw_image(image = icon_Mdor2, x = 0.14, y = 0.48, scale = 0.080) +# Panel B
  draw_image(image = icon_Axan2, x =  -0.36, y =  0.313, scale = 0.087) +# Panel C
  draw_image(image = icon_Plat2, x = 0.14, y = 0.313, scale = 0.087) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.36, y = 0.146, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = 0.14, y = 0.146, scale = 0.060) +# Panel F
  draw_image(image = icon_Cpan2, x =  -0.36, y = -0.021, scale = 0.090) +# Panel G
  draw_image(image = icon_Elab2, x = 0.14, y =  -0.021, scale = 0.080) +# Panel H
  draw_image(image = icon_Ccol2, x = -0.36, y = -0.188, scale = 0.090) +# Panel I
  draw_image(image = icon_Atro2, x = 0.14, y = -0.188, scale = 0.090) +# Panel J
  draw_image(image = icon_Water2, x =  -0.36, y = -0.355, scale = 0.090) # Panel K

multi_PCoA_troph3
```

```{r code for exporting plot above as a png}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by trophic group_VERTICAL2X6.png", plot = multi_PCoA_troph3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8, height = 20, device='png', dpi=700)
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by trophic group_VERTICAL2X6.pdf", plot = multi_PCoA_troph3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8, height = 20, device='pdf', dpi=700)
```

#PCoAs with RAREFIED data (to 2500 reads)
```{r read in rarefied species-by-species data}
#species by species rarefied data split from main rarefied data pool 

#A. concolor
ROHR_06_Aconcolor.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_concolor_rar.rds")

#A. troschelii
ROHR_06_Atroschelii.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_troschelii_rar.rds")

#A. xanthopterus
ROHR_06_Axanthopterus.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_xanthopterus_rar.rds")

#C. colonus
ROHR_06_Ccolonus.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_colonus_rar.rds")

#C. panamensis
ROHR_06_Cpanamensis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_panamensis_rar.rds")

#C. humeralis
ROHR_06_Chumeralis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_humeralis_rar.rds")

#E. labriformis
ROHR_06_Elabriformis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_E_labriformis_rar.rds")

#J. nigrirostris
ROHR_06_Jnigrirostris.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_J_nigrirostris_rar.rds")

#M. dorsalis
ROHR_06_Mdorsalis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_M_dorsalis_rar.rds")

#P. laticlavius
ROHR_06_Platiclavius.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_laticlavius_rar.rds")

#water
ROHR08.2500.rar <- readRDS("~/OneDrive - McGill University/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_08_trimmed/ROHR_08_rarefied_2500.rds")
```

```{r run species-by-species PCoAs, echo = FALSE}

#### Abudefduf concolor ####

##transforming to counts
ROHR_06_Aconcolor_prop.rar = transform_sample_counts(ROHR_06_Aconcolor.rar, function(x) x/sum(x))

# Ordinate
Aconc_pcoa.rar <- ordinate(
  physeq = ROHR_06_Aconcolor_prop.rar, 
  method = "PCoA", 
  distance = "bray"
  )
# Plot - color = season, shape = region (coiba or las perlas)
Aconc.rar <- plot_ordination(
  physeq = ROHR_06_Aconcolor_prop.rar,
  ordination = Aconc_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Abudefduf troschelii ####

##transforming to counts
ROHR_06_Atroschelii_prop.rar = transform_sample_counts(ROHR_06_Atroschelii.rar, function(x) x/sum(x))

# Ordinate
Atro_pcoa.rar <- ordinate(
  physeq = ROHR_06_Atroschelii_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Atro.rar <- plot_ordination(
  physeq = ROHR_06_Atroschelii_prop.rar,
  ordination = Atro_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
#### Acanthurus xanthopterus ####

##transforming to counts
ROHR_06_Axanthopterus_prop.rar = transform_sample_counts(ROHR_06_Axanthopterus.rar, function(x) x/sum(x))

# Ordinate
Axan_pcoa.rar <- ordinate(
  physeq = ROHR_06_Axanthopterus_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Axan.rar <- plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop.rar,
  ordination = Axan_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Cephalopholis colonus ####

##transforming to counts
ROHR_06_Ccolonus_prop.rar = transform_sample_counts(ROHR_06_Ccolonus.rar, function(x) x/sum(x))

# Ordinate
Ccol_pcoa.rar <- ordinate(
  physeq = ROHR_06_Ccolonus_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Ccol.rar <- plot_ordination(
  physeq = ROHR_06_Ccolonus_prop.rar,
  ordination = Ccol_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
#### Cephalopholis panamensis ####

##transforming to counts
ROHR_06_Cpanamensis_prop.rar = transform_sample_counts(ROHR_06_Cpanamensis.rar, function(x) x/sum(x))

# Ordinate
Cpan_pcoa.rar <- ordinate(
  physeq = ROHR_06_Cpanamensis_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Cpan.rar <- plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop.rar,
  ordination = Cpan_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Chaetodon humeralis ####

##transforming to counts
ROHR_06_Chumeralis_prop.rar = transform_sample_counts(ROHR_06_Chumeralis.rar, function(x) x/sum(x))

# Ordinate
Chum_pcoa.rar <- ordinate(
  physeq = ROHR_06_Chumeralis_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Chum.rar <- plot_ordination(
  physeq = ROHR_06_Chumeralis_prop.rar,
  ordination = Chum_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Epinephelus labriformis ####

##transforming to counts
ROHR_06_Elabriformis_prop.rar = transform_sample_counts(ROHR_06_Elabriformis.rar, function(x) x/sum(x))

# Ordinate
Elab_pcoa.rar <- ordinate(
  physeq = ROHR_06_Elabriformis_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Elab.rar <- plot_ordination(
  physeq = ROHR_06_Elabriformis_prop.rar,
  ordination = Elab_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Johnrandallia nigrirostris ####

##transforming to counts
ROHR_06_Jnigrirostris_prop.rar = transform_sample_counts(ROHR_06_Jnigrirostris.rar, function(x) x/sum(x))

# Ordinate
Jnig_pcoa.rar <- ordinate(
  physeq = ROHR_06_Jnigrirostris_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Jnig.rar <- plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop.rar,
  ordination = Jnig_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Microspathodon dorsalis ####

##transforming to counts
ROHR_06_Mdorsalis_prop.rar = transform_sample_counts(ROHR_06_Mdorsalis.rar, function(x) x/sum(x))

# Ordinate
Mdor_pcoa.rar <- ordinate(
  physeq = ROHR_06_Mdorsalis_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Mdor.rar <- plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop.rar,
  ordination = Mdor_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Prionurus laticlavius ####

##transforming to counts
ROHR_06_Platiclavius_prop.rar = transform_sample_counts(ROHR_06_Platiclavius.rar, function(x) x/sum(x))

# Ordinate
Plat_pcoa.rar <- ordinate(
  physeq = ROHR_06_Platiclavius_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Plat.rar <- plot_ordination(
  physeq = ROHR_06_Platiclavius_prop.rar,
  ordination = Plat_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Water Microbiome ####

####### Transform counts to relative abundance per sample
Water_prop.rar = transform_sample_counts(ROHR08.2500.rar, function(x) x/sum(x))

# Ordinate
Water_pcoa.rar <- ordinate(
  physeq = Water_prop.rar, 
  method = "PCoA", 
  distance = "bray"
)
# Plot - color = season, shape = region (coiba or las perlas)
Water.rar <- plot_ordination(
  physeq = Water_prop.rar,
  ordination = Water_pcoa.rar,
  color = "season",
  shape = "region",
) + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")
```

```{r visualize individual rarefied plots}
Aconc.rar
Atro.rar
Axan.rar
Chum.rar
Ccol.rar
Cpan.rar
Elab.rar
Jnig.rar
Mdor.rar
Plat.rar
Water.rar
```

```{r visualize individual non-rarefied plots for comparison}
Aconc
Atro
Axan
Chum
Ccol
Cpan
Elab
Jnig
Mdor
Plat
Water
```

Visually, other than for the water samples, it's really hard to see any clear patterns or clusters within any of these fish, be it with the rarefied or non rarefied data... Worth considering the value of actually including this figure in the main text if it's basically showing 10 panels of non-clustering (although PERMANOVAs do show some significant differences between season/region for some hosts)

```{r add color-coded axes to each PCoA rarefied plot for trophic group rather than species, echo = FALSE}
## to change color of axes instead of whole box around plot, use code below
# pcoa_plot + theme(axis.line = element_line(color = "red", linewidth = 1))
# other alternative = changing image icon source files to match colors instead of B&W

Aconc.rar <- Aconc.rar + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
Atro.rar <- Atro.rar + theme(axis.line = element_line(color = "darkgoldenrod1", linewidth = 1))
                   
Axan.rar <- Axan.rar + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
Chum.rar <- Chum.rar + theme(axis.line = element_line(color = "darkblue", linewidth = 1))

Ccol.rar <- Ccol.rar + theme(axis.line = element_line(color = "darkmagenta", linewidth = 1))
                   
Cpan.rar <- Cpan.rar + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
Elab.rar <- Elab.rar + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
Jnig.rar <- Jnig.rar + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
Mdor.rar <- Mdor.rar + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
Plat.rar <- Plat.rar + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
Water.rar <- Water.rar + theme(axis.line = element_line(color = "#BE93D4", linewidth = 1))
                   
```

```{r standardize axes across plots}
# across our plots: x-axis ranges from -0.6 to 0.5, y-axis ranges from -0.6 to ~ 0.6

Aconc.rar2 <- Aconc.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Atro.rar2 <- Atro.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Axan.rar2 <- Axan.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Chum.rar2 <- Chum.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Cpan.rar2 <- Cpan.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Elab.rar2 <- Elab.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Jnig.rar2 <- Jnig.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Mdor.rar2 <- Mdor.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Ccol.rar2 <- Ccol.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Plat.rar2 <- Plat.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
Water.rar2 <- Water.rar + scale_x_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6")) + 
  scale_y_continuous(limits = c(-0.6, 0.6), breaks = seq(-0.6, 0.6, 0.2), labels = c("-0.6", "-0.4", "-0.2", "0.0", "0.2", "0.4", "0.6"))
```

```{r PCoA RAREFIED data (2500) colors by trophic group, echo = FALSE}

#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA_rar <- plot_grid(Aconc.rar2, Mdor.rar2, Axan.rar2, Plat.rar2, Jnig.rar2, Chum.rar2, Cpan.rar2, Elab.rar2, Ccol.rar2, Atro.rar2, Water.rar2, legendAC,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

#add icons to graph
multi_PCoA_rar2 <- ggdraw() +
  draw_plot(multi_PCoA_rar) +
  draw_image(image = icon_Aconc2, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor2, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan2, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat2, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Cpan2, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Elab2, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Ccol2, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Atro2, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water2, x =  0.070, y = -0.20, scale = 0.060) # Panel K

multi_PCoA_rar2
```
```{r code for exporting plot above as a png}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - RAREFIED 2500.png", plot = multi_PCoA_rar2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='png', dpi=700)
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - RAREFIED 2500.pdf", plot = multi_PCoA_rar2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

#### testing vertical 2x6 layout
```{r 2x6 vertical layout}

#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_PCoA_troph.1.rar <- plot_grid(Aconc.rar2, Mdor.rar2, Axan.rar2, Plat.rar2, Jnig.rar2, Chum.rar2, Cpan.rar2, Elab.rar2, Ccol.rar2, Atro.rar2, Water.rar2, legendAC,
          ncol = 2,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)
multi_PCoA_troph.1.rar

#add icons to graph
multi_PCoA_troph3.rar <- ggdraw() +
  draw_plot(multi_PCoA_troph.1) +
  draw_image(image = icon_Aconc2, x = -0.36, y =  0.48, scale = 0.080) +# Panel A
  draw_image(image = icon_Mdor2, x = 0.14, y = 0.48, scale = 0.080) +# Panel B
  draw_image(image = icon_Axan2, x =  -0.36, y =  0.313, scale = 0.087) +# Panel C
  draw_image(image = icon_Plat2, x = 0.14, y = 0.313, scale = 0.087) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.36, y = 0.146, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = 0.14, y = 0.146, scale = 0.060) +# Panel F
  draw_image(image = icon_Cpan2, x =  -0.36, y = -0.021, scale = 0.090) +# Panel G
  draw_image(image = icon_Elab2, x = 0.14, y =  -0.021, scale = 0.080) +# Panel H
  draw_image(image = icon_Ccol2, x = -0.36, y = -0.188, scale = 0.090) +# Panel I
  draw_image(image = icon_Atro2, x = 0.14, y = -0.188, scale = 0.090) +# Panel J
  draw_image(image = icon_Water2, x =  -0.36, y = -0.355, scale = 0.090) # Panel K

multi_PCoA_troph3.rar
```

```{r code for exporting plot above as a png}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by trophic group_VERTICAL2X6 RAREFIED 2500.png", plot = multi_PCoA_troph3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8, height = 20, device='png', dpi=700)
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin PCoA Plots 21-22 - color coded by trophic group_VERTICAL2X6 RAREFIED 2500.pdf", plot = multi_PCoA_troph3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8, height = 20, device='pdf', dpi=700)
```

#NMDS
```{r run species-by-species NMDS, echo = FALSE}
#can add code chunk below to have ellipses on NMDS plots
# + stat_ellipse(type = "t")

##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

set.seed(1)

#### Abudefduf concolor ####

# Ordinate
Aconc_nmds <- ordinate(
  physeq = ROHR_06_Aconcolor_prop, 
  method = "NMDS", 
  distance = "bray"
)

Aconc_n <- plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Abudefduf troschelii ####

# Ordinate
Atro_nmds <- ordinate(
  physeq = ROHR_06_Atroschelii_prop, 
  method = "NMDS", 
  distance = "bray"
)

Atro_n <- plot_ordination(
  physeq = ROHR_06_Atroschelii_prop,
  ordination = Atro_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Acanthurus xanthopterus ####

# Ordinate
Axan_nmds <- ordinate(
  physeq = ROHR_06_Axanthopterus_prop, 
  method = "NMDS", 
  distance = "bray"
)

Axan_n <- plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop,
  ordination = Axan_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Chaetodon humeralis ####
# Ordinate
Chum_nmds <- ordinate(
  physeq = ROHR_06_Chumeralis_prop, 
  method = "NMDS", 
  distance = "bray"
)

Chum_n <- plot_ordination(
  physeq = ROHR_06_Chumeralis_prop,
  ordination = Chum_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Cephalopholis colonus ####
# Ordinate
Ccol_nmds <- ordinate(
  physeq = ROHR_06_Ccolonus_prop, 
  method = "NMDS", 
  distance = "bray"
)

Ccol_n <- plot_ordination(
  physeq = ROHR_06_Ccolonus_prop,
  ordination = Ccol_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1)+
  theme(legend.position = "none")

#### Cephalopholis panamensis ####
# Ordinate
Cpan_nmds <- ordinate(
  physeq = ROHR_06_Cpanamensis_prop, 
  method = "NMDS", 
  distance = "bray"
)

Cpan_n <- plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop,
  ordination = Cpan_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Microspathodon dorsalis ####
# Ordinate
Mdor_nmds <- ordinate(
  physeq = ROHR_06_Mdorsalis_prop, 
  method = "NMDS", 
  distance = "bray"
)

Mdor_n <- plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop,
  ordination = Mdor_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Johnrandallia nigrirostris ####
# Ordinate
Jnig_nmds <- ordinate(
  physeq = ROHR_06_Jnigrirostris_prop, 
  method = "NMDS", 
  distance = "bray"
)

Jnig_n <- plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop,
  ordination = Jnig_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Epinephelus labriformis ####
# Ordinate
Elab_nmds <- ordinate(
  physeq = ROHR_06_Elabriformis_prop, 
  method = "NMDS", 
  distance = "bray"
)

Elab_n <- plot_ordination(
  physeq = ROHR_06_Elabriformis_prop,
  ordination = Elab_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Prionurus laticlavius ####
# Ordinate
Plat_nmds <- ordinate(
  physeq = ROHR_06_Platiclavius_prop, 
  method = "NMDS", 
  distance = "bray"
)

Plat_n <- plot_ordination(
  physeq = ROHR_06_Platiclavius_prop,
  ordination = Plat_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

#### Water ####
# Ordinate
Water_nmds <- ordinate(
  physeq = ROHR_08_obj_clean_prop, 
  method = "NMDS", 
  distance = "bray"
)

Water_n <- plot_ordination(
  physeq = ROHR_08_obj_clean_prop,
  ordination = Water_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual( values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  theme(legend.position = "none")

```

```{r create 12th NMDS plot to contain legend for multipanel figure, echo = FALSE}
# set colors for seasons - "#660066" = non-upwelling, "#336633" = upwelling
c2 <- c("#660066", "#336633")

#run one of the ordinations above, including legend
plotlegend2 <- plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_nmds,
  color = "season",
  shape = "region") + 
  scale_color_manual(values = c2) +
  geom_point(aes(color = season), alpha = 0.6, size = 4.5) +
  geom_point(colour = "grey90", size = 1) +
  labs(color = "Season", shape = "Region") #change names of legend titles to capitalize
# + theme(
   # legend.justification = c("right")) #shift legend to right of panel (to accommodate fish icons)  

legendAc <- cowplot::get_legend(plotlegend2)
cowplot::ggdraw(legendAc)
```
```{r plot basic NMDS multipanel, echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#note that intervals of 4X3 grids are ~0.25 at this scale size-to-side, & ~0.3 vertically

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_NMDS <- plot_grid(Aconc_n, Mdor_n, Axan_n, Plat_n, Jnig_n, Chum_n, Ccol_n, Atro_n, Cpan_n, Elab_n, Water_n, legendAc,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

multi_NMDS_1 <- ggdraw() +
  draw_plot(multi_NMDS) +
  draw_image(image = icon_Aconc, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water, x =  0.072, y = -0.20, scale = 0.060) # Panel K

multi_NMDS_1
```

```{r add color-coded axes to each NMDS plot for species, echo = FALSE}
## to change color of axes instead of whole box around plot, use code below
# pcoa_plot + theme(axis.line = element_line(color = "red", linewidth = 1))
# other alternative = changing image icon source files to match colors instead of B&W

AconcN <- Aconc_n + theme(axis.line = element_line(color = "aquamarine4", linewidth = 1))
                   
AtroN <- Atro_n + theme(axis.line = element_line(color = "darkgoldenrod1", linewidth = 1))
                   
AxanN <- Axan_n + theme(axis.line = element_line(color = "chartreuse3", linewidth = 1))
                   
ChumN <- Chum_n + theme(axis.line = element_line(color = "deepskyblue2", linewidth = 1))
                   
CpanN <- Cpan_n + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
ElabN <- Elab_n + theme(axis.line = element_line(color = "chocolate2", linewidth = 1))
                   
JnigN <- Jnig_n + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
MdorN <- Mdor_n + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
CcolN <- Ccol_n + theme(axis.line = element_line(color = "darkmagenta", linewidth = 1))
                   
PlatN <- Plat_n + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
WaterN <- Water_n + theme(axis.line = element_line(color = "#BE93D4", linewidth = 1))
                   
```

```{r plot color-coded NMDS, echo = TRUE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#note that intervals of 4X3 grids are ~0.25 at this scale size-to-side, & ~0.3 vertically

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_NMDS2 <- plot_grid(AconcN, MdorN, AxanN, PlatN, JnigN, ChumN, CcolN, AtroN, CpanN, ElabN, WaterN, legendAc,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

multi_NMDS_2 <- ggdraw() +
  draw_plot(multi_NMDS2) +
  draw_image(image = icon_Aconc2, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor2, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan2, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat2, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Ccol2, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Atro2, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Cpan2, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Elab2, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water2, x =  0.072, y = -0.20, scale = 0.060) # Panel K

multi_NMDS_2
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin NMDS Plots 21-22 - color coded by species.pdf", plot = multi_NMDS_2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Water NMDS Plot 21-22.pdf", plot = WaterN, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

```{r add color-coded axes to each NMDS plot for species, echo = FALSE}
## to change color of axes instead of whole box around plot, use code below
# pcoa_plot + theme(axis.line = element_line(color = "red", linewidth = 1))
# other alternative = changing image icon source files to match colors instead of B&W

AconcNt <- Aconc_n + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
AtroNt <- Atro_n + theme(axis.line = element_line(color = "darkgoldenrod1", linewidth = 1))
                   
AxanNt <- Axan_n + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
ChumNt <- Chum_n + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
CpanNt <- Cpan_n + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
ElabNt <- Elab_n + theme(axis.line = element_line(color = "chocolate4", linewidth = 1))
                   
JnigNt <- Jnig_n + theme(axis.line = element_line(color = "darkblue", linewidth = 1))
                   
MdorNt <- Mdor_n + theme(axis.line = element_line(color = "aquamarine2", linewidth = 1))
                   
CcolNt <- Ccol_n + theme(axis.line = element_line(color = "darkmagenta", linewidth = 1))
                   
PlatNt <- Plat_n + theme(axis.line = element_line(color = "darkgreen", linewidth = 1))
                   
WaterNt <- Water_n + theme(axis.line = element_line(color = "#BE93D4", linewidth = 1))
                   
```

```{r plot color-coded NMDS, echo = TRUE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#note that intervals of 4X3 grids are ~0.25 at this scale size-to-side, & ~0.3 vertically

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_NMDS_3 <- plot_grid(AconcNt, MdorNt, AxanNt, PlatNt, JnigNt, ChumNt, CpanNt, ElabNt, CcolNt, AtroNt, WaterNt, legendAc,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)

multi_NMDS_3 <- ggdraw() +
  draw_plot(multi_NMDS_3) +
  draw_image(image = icon_Aconc2, x = -0.430, y =  0.460, scale = 0.060) +# Panel A
  draw_image(image = icon_Mdor2, x = -0.180, y = 0.460, scale = 0.060) +# Panel B
  draw_image(image = icon_Axan2, x =  0.070, y =  0.460, scale = 0.060) +# Panel C
  draw_image(image = icon_Plat2, x = 0.320, y = 0.460, scale = 0.060) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.430, y = 0.118, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = -0.180, y = 0.118, scale = 0.060) +# Panel F
  draw_image(image = icon_Cpan2, x =  0.070, y = 0.130, scale = 0.060) +# Panel G
  draw_image(image = icon_Elab2, x = 0.320, y =  0.130, scale = 0.060) +# Panel H
  draw_image(image = icon_Ccol2, x = -0.430, y = -0.20, scale = 0.060) +# Panel I
  draw_image(image = icon_Atro2, x = -0.180, y = -0.20, scale = 0.060) +# Panel J
  draw_image(image = icon_Water2, x =  0.072, y = -0.20, scale = 0.060) # Panel K

multi_NMDS_3
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin NMDS Plots 21-22 color coded by trophic group.pdf", plot = multi_NMDS_3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

```{r code for exporting plot above as a png}
ggsave(filename = "Fish Skin NMDS Plots 21-22 color coded by trophic group.png", plot = multi_NMDS_3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='png', dpi=700)
```

#### testing vertical 2x6 layout NMDS
```{r 2x6 vertical layout}

#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_NMDS_troph.1 <- plot_grid(AconcNt, MdorNt, AxanNt, PlatNt, JnigNt, ChumNt, CcolNt, AtroNt, CpanNt, ElabNt, WaterNt, legendAc,
          ncol = 2,
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)
multi_NMDS_troph.1

#add icons to graph
multi_NMDS_troph2 <- ggdraw() +
  draw_plot(multi_NMDS_troph.1) +
  draw_image(image = icon_Aconc2, x = -0.36, y =  0.48, scale = 0.080) +# Panel A
  draw_image(image = icon_Mdor2, x = 0.14, y = 0.48, scale = 0.080) +# Panel B
  draw_image(image = icon_Axan2, x =  -0.36, y =  0.313, scale = 0.087) +# Panel C
  draw_image(image = icon_Plat2, x = 0.14, y = 0.313, scale = 0.087) +# Panel D
  draw_image(image = icon_Jnig2, x = -0.36, y = 0.146, scale = 0.060) +# Panel E
  draw_image(image = icon_Chum2, x = 0.14, y = 0.146, scale = 0.060) +# Panel F
  draw_image(image = icon_Cpan2, x =  -0.36, y = -0.021, scale = 0.090) +# Panel G
  draw_image(image = icon_Elab2, x = 0.14, y =  -0.021, scale = 0.080) +# Panel H
  draw_image(image = icon_Ccol2, x = -0.36, y = -0.188, scale = 0.090) +# Panel I
  draw_image(image = icon_Atro2, x = 0.14, y = -0.188, scale = 0.090) +# Panel J
  draw_image(image = icon_Water2, x =  -0.36, y = -0.355, scale = 0.090) # Panel K

multi_NMDS_troph2
```

```{r code for exporting plot above as a png}
ggsave(filename = "Fish Skin NMDS Plots 21-22 - color coded by trophic group_VERTICAL2X6.png", plot = multi_NMDS_troph2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8, height = 20, device='png', dpi=700)
```

```{r code for exporting plot above as a pdf}
ggsave(filename = "Fish Skin NMDS Plots 21-22 - color coded by trophic group_VERTICAL2X6.pdf", plot = multi_NMDS_troph2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8, height = 20, device='pdf', dpi=700)
```

####START HERE 3/4/25 - make rarefied NMDS plots

####Section 3 - Skin microbiome composition stacked barplots

###Jump to here barplots 12/19/24

```{r run code for individual stacked barplots, echo = FALSE}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_phylum <- ROHR_06_obj_phylo_root %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

#pick colors for bars
c23 <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3")
  
unique(ROHR_06_phylum$Phylum)

#rename 2 phyla (v. long names currently)
ROHR_06_phylum["Phylum"][ROHR_06_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_phylum["Phylum"][ROHR_06_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

names(c23) = unique(ROHR_06_phylum$Phylum)

####Abudefduf concolor####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Aconcolor)$RRR_number <- as.character(sample_data(ROHR_06_Aconcolor)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Aconcolor <- merge_treatments(ROHR_06_Aconcolor, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_conc_phylum <- ROHR_06_Aconcolor %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_conc_phylum["Phylum"][ROHR_06_conc_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_conc_phylum["Phylum"][ROHR_06_conc_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_conc_phylum$Phylum) #31 unique phyla (compared to 23 in the filtered full dataset) --> need to extend number of available colors from c25 list: currently "darkorange4" and "brown" are unassigned, thus only need 6 more colors

new_phyla <- c("Calditrichota", "Chloroflexi", "Dadabacteria", "Entotheonellaeota", "Gemmatimonadota", "Latescibacterota", "Margulisbacteria", "NB1-j", "PAUC34f", "WPS-2")

#pick colors for these phyla

c10 <- c(
  "slateblue4", "wheat", "lightseagreen", 
  "mediumorchid4", "burlywood4", "brown", "orange1", 
  "yellow", "lightsteelblue1", "honeydew")
  
names(c10) = new_phyla

#merge old & new color assignments
c33 <- c(c23, c10)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Aconc_bar <- ggplot(ROHR_06_conc_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank())

Aconc_bar
####Abudefduf troschelii####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Atroschelii)$RRR_number <- as.character(sample_data(ROHR_06_Atroschelii)$RRR)

#create new categroy with RRR & region/season merged
ROHR_06_Atroschelii <- merge_treatments(ROHR_06_Atroschelii, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_tro_phylum <- ROHR_06_Atroschelii %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_tro_phylum["Phylum"][ROHR_06_tro_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_tro_phylum["Phylum"][ROHR_06_tro_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

#phyla not present in either A. concolor nor filtered ROHR_06 full dataset:
new_phyla2 <- c("MBNT15")

#pick colors for these phyla

c3 <- c("lightyellow3")
  
names(c3) = new_phyla2
#merge old & new color assignments
c36 <- c(c33, c3)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Atrosch_bar <- ggplot(ROHR_06_tro_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank())

Atrosch_bar

####Acanthurus xanthopterus####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Axanthopterus)$RRR_number <- as.character(sample_data(ROHR_06_Axanthopterus)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Axanthopterus <- merge_treatments(ROHR_06_Axanthopterus, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_xan_phylum <- ROHR_06_Axanthopterus %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_xan_phylum["Phylum"][ROHR_06_xan_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_xan_phylum["Phylum"][ROHR_06_xan_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Axanth_bar <- ggplot(ROHR_06_xan_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Axanth_bar

####Chaetodon humeralis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Chumeralis)$RRR_number <- as.character(sample_data(ROHR_06_Chumeralis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Chumeralis <- merge_treatments(ROHR_06_Chumeralis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_hum_phylum <- ROHR_06_Chumeralis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_hum_phylum["Phylum"][ROHR_06_hum_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_hum_phylum["Phylum"][ROHR_06_hum_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

new_phyla3 <- c("Deferrisomatota", "Dependentiae", "LCP-89")

#pick colors for these phyla

c3bis <- c("navy", "gray40", "hotpink4")

names(c3bis) = new_phyla3

#merge old & new color assignments
c39 <- c(c36, c3bis)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Chum_bar <- ggplot(ROHR_06_hum_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c39, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Chum_bar

####Cephalopholis colonus####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Ccolonus)$RRR_number <- as.character(sample_data(ROHR_06_Ccolonus)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Ccolonus <- merge_treatments(ROHR_06_Ccolonus, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_col_phylum <- ROHR_06_Ccolonus %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_col_phylum["Phylum"][ROHR_06_col_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_col_phylum["Phylum"][ROHR_06_col_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_col_phylum$Phylum) #new phyla c("Sva0485")
new_phylum <- c("Sva0485")

#pick colors for these phyla

c1 <- c("sienna3")

names(c1) = new_phylum

#merge old & new color assignments
c40 <- c(c39, c1)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Ccol_bar <- ggplot(ROHR_06_col_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c40, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
 theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Ccol_bar

####Cephalopholis panamensis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Cpanamensis)$RRR_number <- as.character(sample_data(ROHR_06_Cpanamensis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Cpanamensis <- merge_treatments(ROHR_06_Cpanamensis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_pan_phylum <- ROHR_06_Cpanamensis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_pan_phylum["Phylum"][ROHR_06_pan_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_pan_phylum["Phylum"][ROHR_06_pan_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_pan_phylum$Phylum) #this is the largest number of different phyla in a single fish host species so far --> new_phyla4 <- c("Sumerlaeota")

new_phyla4 <- c("Sumerlaeota", "Deferribacterota")

c1bis <- c("ivory", "mediumblue")

names(c1bis) = new_phyla4

#merge old & new color assignments
c42 <- c(c40, c1bis)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Cpan_bar <- ggplot(ROHR_06_pan_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Cpan_bar

####Microspathodon dorsalis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Mdorsalis)$RRR_number <- as.character(sample_data(ROHR_06_Mdorsalis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Mdorsalis <- merge_treatments(ROHR_06_Mdorsalis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_dor_phylum <- ROHR_06_Mdorsalis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_dor_phylum["Phylum"][ROHR_06_dor_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_dor_phylum["Phylum"][ROHR_06_dor_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_dor_phylum$Phylum) #no new phyla

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Mdor_bar <- ggplot(ROHR_06_dor_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
 theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Mdor_bar

####Johnrandallia nigrirostris####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Jnigrirostris)$RRR_number <- as.character(sample_data(ROHR_06_Jnigrirostris)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Jnigrirostris <- merge_treatments(ROHR_06_Jnigrirostris, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_nig_phylum <- ROHR_06_Jnigrirostris %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_nig_phylum["Phylum"][ROHR_06_nig_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_nig_phylum["Phylum"][ROHR_06_nig_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_nig_phylum$Phylum) #no new phyla

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Jnig_bar <- ggplot(ROHR_06_nig_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
   theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Jnig_bar

####Epinephelus labriformis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Elabriformis)$RRR_number <- as.character(sample_data(ROHR_06_Elabriformis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Elabriformis <- merge_treatments(ROHR_06_Elabriformis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_lab_phylum <- ROHR_06_Elabriformis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_lab_phylum["Phylum"][ROHR_06_lab_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lab_phylum["Phylum"][ROHR_06_lab_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_lab_phylum$Phylum) #no new phyla

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Elab_bar <- ggplot(ROHR_06_lab_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
         axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 

Elab_bar

####Prionurus laticlavius####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Platiclavius)$RRR_number <- as.character(sample_data(ROHR_06_Platiclavius)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Platiclavius <- merge_treatments(ROHR_06_Platiclavius, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_lat_phylum <- ROHR_06_Platiclavius %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_lat_phylum["Phylum"][ROHR_06_lat_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lat_phylum["Phylum"][ROHR_06_lat_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Plat_bar <- ggplot(ROHR_06_lat_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(), 
        axis.title.y = element_blank(),
       axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank()) 
Plat_bar

####Water####


#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_08_obj_phylo_root)$RRR_number <- as.character(sample_data(ROHR_08_obj_phylo_root)$RRR)

#create new category with RRR & region/season merged
ROHR_08_obj_phylo_root <- merge_treatments(ROHR_08_obj_phylo_root, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_08_Water_phylum <- ROHR_08_obj_phylo_root %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_08_Water_phylum["Phylum"][ROHR_08_Water_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_08_Water_phylum["Phylum"][ROHR_08_Water_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_08_Water_phylum$Phylum) # c("AncK6", "Nitrospinota", "Aenigmarchaeota", "Poribacteria")
# Latescibacteria, MBNT15, Deferrisomatota, Deferribacterota, LCP-89, and Sumerlaeota are all phyla that were found in at least one fish specimen but not in our water samples
newwater_phyla <- c("AncK6", "Nitrospinota", "Aenigmarchaeota", "Poribacteria")
c4 <- c("darkorange4", "aquamarine", "darkgoldenrod", "chartreuse3")

names(c4) = newwater_phyla

#merge old & new color assignments
c44 <- c(c42, c4)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
Water_bar <- ggplot(ROHR_08_Water_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
 theme(legend.position = "none",
        axis.text.x = element_blank(), 
        axis.title.x = element_blank(), 
        axis.text.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.title.y = element_blank(),
        axis.ticks.x = element_blank()) 

Water_bar
```

```{r code for exporting individual plots above as pdfs}
ggsave(filename = "Aconc barplot 21-22.pdf", plot = Aconc_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Atros barplot 21-22.pdf", plot = Atrosch_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Axan barplot 21-22.pdf", plot = Axanth_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Chum barplot 21-22.pdf", plot = Chum_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Chum barplot 21-22.pdf", plot = Chum_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Cpan barplot 21-22.pdf", plot = Cpan_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Cpan barplot 21-22.pdf", plot = Cpan_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Ccol barplot 21-22.pdf", plot = Ccol_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Jnig barplot 21-22.pdf", plot = Jnig_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Mdor barplot 21-22.pdf", plot = Mdor_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Elab barplot 21-22.pdf", plot = Elab_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Plat barplot 21-22.pdf", plot = Plat_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Plat barplot 21-22.pdf", plot = Plat_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)

ggsave(filename = "Water barplot 21-22.pdf", plot = Water_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

```{r code for exporting individual plots above as PNG files}
ggsave(filename = "Aconc barplot 21-22.png", plot = Aconc_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Atros barplot 21-22.png", plot = Atrosch_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Axan barplot 21-22.png", plot = Axanth_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Chum barplot 21-22.png", plot = Chum_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Chum barplot 21-22.png", plot = Chum_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Cpan barplot 21-22.png", plot = Cpan_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Cpan barplot 21-22.png", plot = Cpan_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Ccol barplot 21-22.png", plot = Ccol_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Jnig barplot 21-22.png", plot = Jnig_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Mdor barplot 21-22.png", plot = Mdor_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Elab barplot 21-22.png", plot = Elab_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Plat barplot 21-22.png", plot = Plat_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Plat barplot 21-22.png", plot = Plat_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)

ggsave(filename = "Water barplot 21-22.png", plot = Water_bar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)
```

We have 44 unique phyla represented across the different fish and water samples, meaning that we'll need to use the full fish dataset (rather than the subset-by-species ones) to create a legend that encompasses all phyla. Since the water samples ALSO had unique phyla, we'll try to merge the two phyloseq objects and create a new legend from this master dataset

```{r merge ROHR_06 & ROHR_08 datasets}
#read in phyloseq objects w/o tree, as this messes with the merge_phyloseq function (alt = separate out phyloseq objects into their component parts and create new one)

#Skin phyloseq object from A-div analyses, pre-tree calc (not rarefied)
ROHR_06_obj_clean_updated_P = readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_clean_updated_phyla.rds")

#Water phyloseq w/o tree
ROHR_08_obj_clean_updated_P <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_08_trimmed/ROHR_08_obj_clean_updated_phyla.rds")

#merge phyloseq objects into single master file
master_phylo <- merge_phyloseq(ROHR_06_obj_clean_updated_P, ROHR_08_obj_clean_updated_P)

#change RRR numbers (sample ID) from integer to character
sample_data(master_phylo)$RRR_number <- as.character(sample_data(master_phylo)$RRR)

#create new category with RRR & region/season merged
master_phylo <- merge_treatments(master_phylo, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
master_phylum <- master_phylo %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
master_phylum["Phylum"][master_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
master_phylum["Phylum"][master_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(master_phylum$Phylum)

#somehow there are significantly more unique phyla in this master dataset ( 66 vs 44) than in the species-by-species ones...
#check # unique phyla in source datasets

ROHR06_phylum <- ROHR_06_obj_clean_updated_P %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

unique(ROHR06_phylum$Phylum) #40

ROHR08_phylum <- ROHR_08_obj_clean_updated_P %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

unique(ROHR08_phylum$Phylum) #38

#somehow there are still some unique phyla that are only found in the merged master dataset and not in either source one??
```
```{r barchart with all samples}
MASTER_bar <- ggplot(master_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  scale_x_discrete() +
  scale_y_continuous(expand=c(0,0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
 theme() 

MASTER_bar
#ignore awful smear of text at bottom, would need to add shorthand + rotate axis labels to fix
```
```{r create 12th plot to contain legend for multipanel bar plot figure, echo = FALSE}

#settings for this one adjusted for individual pdf/png image

#run "master" dataset to get full legend across all host species
plotlegend_bar <- ggplot(master_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, title.hjust = 0.5, keywidth = 5, keyheight = 3, ncol = 4)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 28), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 18), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 20, face = "bold"))

legendMaster <- cowplot::get_legend(plotlegend_bar)
cowplot::ggdraw(legendMaster)

```
```{r save legend for barplots, echo = FALSE}
ggsave(filename = "fish barplot legend 4 column.pdf", plot = legendMaster, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

```{r save legend for barplots as PNG, echo = FALSE}
ggsave(filename = "fish barplot legend 4 column.png", plot = legendMaster, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='png', dpi=700)
```

```{r legend for multipanel bar plot figure, echo = FALSE}

#run "master" dataset to get full legend across all host species
plotlegend_bar2 <- ggplot(master_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, title.hjust = 0.5, keywidth = 1, keyheight = 1, ncol = 3)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 12), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 8), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 12, face = "bold"))

legendMaster2 <- cowplot::get_legend(plotlegend_bar2)
cowplot::ggdraw(legendMaster2)
```

```{r basic combined bar plot, echo = FALSE}
#place plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water (w/ legend)

#with 2 col, will need to specify view window as the defaults currently distort the panels
multi_bar <- plot_grid(Aconc_bar, Mdor_bar, Axanth_bar, Plat_bar, Jnig_bar, Chum_bar, Ccol_bar, Atrosch_bar, Cpan_bar, Elab_bar, Water_bar, legendMaster2, 
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15)
multi_bar
```
```{r optimizing multibar plot}
#basic gridded plot OK, but could be nicer:
# (1) Make letters not overlap with corner of each chart
# (2) Scale legend properly to match charts
# (3) remove grey space above/below bars - fixed with the addition of: scale_y_continuous(expand=c(0,0))

multi_bar2 <- plot_grid(Aconc_bar, Mdor_bar, Axanth_bar, Plat_bar, Jnig_bar, 
                        Chum_bar, Ccol_bar, Atrosch_bar, Cpan_bar, Elab_bar, 
                        Water_bar, legendMaster2, 
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15, 
          hjust = 0,
          vjust = 2,
          nrow = 3,
          scale = 0.95)
multi_bar2

```
```{r save barchart with basic legend, echo = FALSE}
ggsave(filename = "Fish skin barchart V1.png", plot = multi_bar2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='png', dpi=700)
```

```{r save barchart with basic legend pdf, echo = FALSE}
ggsave(filename = "Fish skin barchart V1.pdf", plot = multi_bar2, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

```{r optimizing multibar plot vertical full page format}
#changing arrangement to better match a vertical page format

multi_bar3 <- plot_grid(Aconc_bar, Mdor_bar, Axanth_bar, Plat_bar, Jnig_bar, 
                        Chum_bar, Ccol_bar, Atrosch_bar, Cpan_bar, Elab_bar, 
                        Water_bar, legendMaster2, 
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15, 
          hjust = 0,
          vjust = 1,
          nrow = 4,
          scale = 0.95)
multi_bar3

###this still needs some tweaking: the size of the plots is not ideal and the legend is doing weird stuff (too big and cut off)

```
```{r save vertical barchart with basic legend pdf, echo = FALSE}
ggsave(filename = "Fish skin barchart vertical V1.pdf", plot = multi_bar3, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 8.5, height = 11, device='pdf', dpi=700)
```

```{r multi combined bar plot highlighting comparisons, echo = FALSE}
#show 3 main plots for comparison: roving herbivore (Plat), invertivore (Chum), carnivore (Cpan)
#place other plots in order by trophic group - territorial herbs, roving herbs, invertivores, planktivore, omnivore, carnivores, water (no legend)

multi_bar_main <- plot_grid(Plat_bar, Chum_bar, Cpan_bar, 
          labels=c("E","F","G"),
          label_size = 15, 
          hjust = 0.2,
          vjust = 1,
          nrow = 1)
multi_bar_main

multi_bar_top <- plot_grid(Aconc_bar, Mdor_bar, Axanth_bar, Jnig_bar,
          labels=c("A","B","C","D"),
          label_size = 15, 
          hjust = 0.2,
          vjust = 1,
          nrow = 1)
multi_bar_top

multi_bar_bottom <- plot_grid(Ccol_bar, Atrosch_bar, Elab_bar, Water_bar,
          labels=c("H","I","J","K"),
          label_size = 15, 
          hjust = 0.2,
          vjust = 1,
          nrow = 1)
multi_bar_bottom

composite_bar <- plot_grid(multi_bar_top, multi_bar_main, multi_bar_bottom, nrow = 3, rel_heights = c(1,2,1))

composite_bar
```

###adjusting plots to add space between gulfs

#test plot with Aconcolor
```{r adjust figures above to add space bw gulfs}

Aconc_bar <- ggplot(
  ROHR_06_conc_phylum,
  aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))
) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_blank(),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.ticks.x = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()    # Remove background around facet labels
  ) +
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Aconc_bar
```
```{r run water}

Water_bar <- ggplot(ROHR_08_Water_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))
) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_blank(),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.ticks.x = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()    # Remove background around facet labels
  ) +
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Water_bar

# Save the plot as Water_bar.png
ggsave("Water_bar.png", plot = Water_bar)
```

###Jump to here for loop to make species-by-species barplot split by region 
```{r loop code above for all fish}

# List of dataset suffixes
datasets <- c("conc", "dor", "xan", "lat", "nig", "hum", "pan", "lab", "col", "tro")

# Initialize an empty list to store plots
plot_list <- list()

# Loop over dataset names and generate plots
for (dataset in datasets) {
  # Construct dataset name dynamically
  data_name <- paste0("ROHR_06_", dataset, "_phylum")
  
  # Dynamically create the plot for each dataset
  plot <- ggplot(get(data_name), aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) +
    geom_bar(stat = "identity", position = "fill") +
    scale_fill_manual(values = c44, name = "Microbial Phyla") +
    scale_x_discrete() +
    scale_y_continuous(expand = c(0, 0)) +
    guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
    theme(
      legend.position = "none",
      axis.text.x = element_blank(),
      axis.title.x = element_blank(),
      axis.text.y = element_blank(),
      axis.title.y = element_blank(),
      axis.ticks.y = element_blank(),
      axis.ticks.x = element_blank(),
      strip.placement = "outside",  # Place the facet strips outside the plot area
      strip.text = element_text(size = 12),  # Customize facet label text size
      strip.background = element_blank()    # Remove background around facet labels
    ) +
    facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom
  
  # Save the plot with a dynamic name
  plot_name <- paste0(dataset, "_bar")
  ggsave(paste0(plot_name, ".png"), plot = plot)

# Append each plot to the list
  plot_list[[dataset]] <- plot
}

# Add the Water_bar plot to the list
plot_list[["water"]] <- Water_bar

# Combine all the plots into one grid
combined_plot <- plot_grid(plotlist = plot_list, ncol = 2)  # Adjust ncol as needed

# Display the combined plot
print(combined_plot)

# Optionally, save the combined plot to a file
ggsave("Combined_plots.png", plot = combined_plot, width = 8, height = 11, units = "in")
```

```{r add labels to see where each season ends}

`# Define colors for the seasons
season_colors <- c("non-upwelling" = "#660066", "upwelling" = "#336633") 

Aconc_bar2 <- ggplot(
  ROHR_06_conc_phylum,
  aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))
) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()    # Remove background around facet labels
  ) +
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Aconc_bar2

```
```{r loop named dataset for all fish}
# Loop over dataset names and generate plots
for (dataset in datasets) {
  # Construct dataset name dynamically
  data_name <- paste0("ROHR_06_", dataset, "_phylum")
  
  # Dynamically create the plot for each dataset
  plot2 <- ggplot(get(data_name), aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) +
    geom_bar(stat = "identity", position = "fill") +
    scale_fill_manual(values = c44, name = "Microbial Phyla") +
    scale_x_discrete() +
    scale_y_continuous(expand = c(0, 0)) +
    guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
    theme(
      legend.position = "none",
      axis.text.x = element_text(angle = 90, hjust = 1),
      axis.title.x = element_blank(),
      axis.text.y = element_blank(),
      axis.title.y = element_blank(),
      axis.ticks.y = element_blank(),
      axis.ticks.x = element_blank(),
      strip.placement = "outside",  # Place the facet strips outside the plot area
      strip.text = element_text(size = 12),  # Customize facet label text size
      strip.background = element_blank()    # Remove background around facet labels
    ) +
    facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom
  
  # Save the plot with a dynamic name
  plot_name2 <- paste0(dataset, "_bar2")
  ggsave(paste0(plot_name2, ".png"), plot = plot2)
}
```

```{r run water}

Water_bar2 <- ggplot(ROHR_08_Water_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))
) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.ticks.x = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()    # Remove background around facet labels
  ) +
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Water_bar2

# Save the plot as Water_bar.png
ggsave("Water_bar2.png", plot = Water_bar2)
```

##Stacked barplots with RAREFIED data (to 2500 reads)
#####IN PROGRESS 03/11/25
```{r microbial taxa color codes}
#use color codes from plots above
# list of taxa IN ORDER from unrarefied skin dataset

c44_taxa <- c("Acidobacteriota","Actinobacteriota","Bacteroidota","Bdellovibrionota","Campylobacterota", "Crenarchaeota", "Cyanobacteria", "Deinococcota", "Desulfobacterota", "Fibrobacterota", "Firmicutes", "Fusobacteriota", "Marinimicrobia", "Myxococcota", "Nanoarchaeota", "Nitrospirota", "Patescibacteria", "Planctomycetota", "Proteobacteria", "SAR324 clade", "Spirochaetota", "Thermoplasmatota", "Verrucomicrobiota",
              "Calditrichota", "Chloroflexi", "Dadabacteria", "Entotheonellaeota", "Gemmatimonadota", "Latescibacterota", "Margulisbacteria", "NB1-j", "PAUC34f", "WPS-2",
              "MBNT15",
              "Deferrisomatota", "Dependentiae", "LCP-89",
              "Sva0485",
              "Sumerlaeota", "Deferribacterota",
              "AncK6", "Nitrospinota", "Aenigmarchaeota", "Poribacteria" #water
              )

#unique colors
c44 <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3",
  "slateblue4", "wheat", "lightseagreen", 
  "mediumorchid4", "burlywood4", "brown", "orange1", 
  "yellow", "lightsteelblue1", "honeydew",
  "lightyellow3",
  "navy", "gray40", "hotpink4",
  "sienna3",
  "ivory", "mediumblue",
  "darkorange4", "aquamarine", "darkgoldenrod", "chartreuse3"
  )
  
names(c44) = unique(c44_taxa)

```

```{r run code for individual stacked barplots with RAREFIED data, echo = FALSE}
####Abudefduf concolor####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Aconcolor.rar)$RRR_number <- as.character(sample_data(ROHR_06_Aconcolor.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Aconcolor.rar <- merge_treatments(ROHR_06_Aconcolor.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_conc_phylum.rar <- ROHR_06_Aconcolor.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_conc_phylum.rar["Phylum"][ROHR_06_conc_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_conc_phylum.rar["Phylum"][ROHR_06_conc_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_conc_phylum.rar$Phylum) #31 unique phyla 

####Abudefduf troschelii####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Atroschelii.rar)$RRR_number <- as.character(sample_data(ROHR_06_Atroschelii.rar)$RRR)

#create new categroy with RRR & region/season merged
ROHR_06_Atroschelii.rar <- merge_treatments(ROHR_06_Atroschelii.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_tro_phylum.rar <- ROHR_06_Atroschelii.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_tro_phylum.rar["Phylum"][ROHR_06_tro_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_tro_phylum.rar["Phylum"][ROHR_06_tro_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Acanthurus xanthopterus####
#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Axanthopterus.rar)$RRR_number <- as.character(sample_data(ROHR_06_Axanthopterus.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Axanthopterus.rar <- merge_treatments(ROHR_06_Axanthopterus.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_xan_phylum.rar <- ROHR_06_Axanthopterus.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_xan_phylum.rar["Phylum"][ROHR_06_xan_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_xan_phylum.rar["Phylum"][ROHR_06_xan_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Chaetodon humeralis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Chumeralis.rar)$RRR_number <- as.character(sample_data(ROHR_06_Chumeralis.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Chumeralis.rar <- merge_treatments(ROHR_06_Chumeralis.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_hum_phylum.rar <- ROHR_06_Chumeralis.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_hum_phylum.rar["Phylum"][ROHR_06_hum_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_hum_phylum.rar["Phylum"][ROHR_06_hum_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Cephalopholis colonus####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Ccolonus.rar)$RRR_number <- as.character(sample_data(ROHR_06_Ccolonus.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Ccolonus.rar <- merge_treatments(ROHR_06_Ccolonus.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_col_phylum.rar <- ROHR_06_Ccolonus.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_col_phylum.rar["Phylum"][ROHR_06_col_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_col_phylum.rar["Phylum"][ROHR_06_col_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Cephalopholis panamensis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Cpanamensis.rar)$RRR_number <- as.character(sample_data(ROHR_06_Cpanamensis.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Cpanamensis.rar <- merge_treatments(ROHR_06_Cpanamensis.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_pan_phylum.rar <- ROHR_06_Cpanamensis.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_pan_phylum.rar["Phylum"][ROHR_06_pan_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_pan_phylum.rar["Phylum"][ROHR_06_pan_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Microspathodon dorsalis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Mdorsalis.rar)$RRR_number <- as.character(sample_data(ROHR_06_Mdorsalis.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Mdorsalis.rar <- merge_treatments(ROHR_06_Mdorsalis.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_dor_phylum.rar <- ROHR_06_Mdorsalis.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_dor_phylum.rar["Phylum"][ROHR_06_dor_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_dor_phylum.rar["Phylum"][ROHR_06_dor_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Johnrandallia nigrirostris####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Jnigrirostris.rar)$RRR_number <- as.character(sample_data(ROHR_06_Jnigrirostris.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Jnigrirostris.rar <- merge_treatments(ROHR_06_Jnigrirostris.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_nig_phylum.rar <- ROHR_06_Jnigrirostris.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_nig_phylum.rar["Phylum"][ROHR_06_nig_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_nig_phylum.rar["Phylum"][ROHR_06_nig_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Epinephelus labriformis####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Elabriformis.rar)$RRR_number <- as.character(sample_data(ROHR_06_Elabriformis.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Elabriformis.rar <- merge_treatments(ROHR_06_Elabriformis.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_lab_phylum.rar <- ROHR_06_Elabriformis.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_lab_phylum.rar["Phylum"][ROHR_06_lab_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lab_phylum.rar["Phylum"][ROHR_06_lab_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Prionurus laticlavius####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Platiclavius.rar)$RRR_number <- as.character(sample_data(ROHR_06_Platiclavius.rar)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Platiclavius.rar <- merge_treatments(ROHR_06_Platiclavius.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_lat_phylum.rar <- ROHR_06_Platiclavius.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_lat_phylum.rar["Phylum"][ROHR_06_lat_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lat_phylum.rar["Phylum"][ROHR_06_lat_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

####Water####

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR08.2500.rar)$RRR_number <- as.character(sample_data(ROHR08.2500.rar)$RRR)

#create new category with RRR & region/season merged
ROHR08.2500.rar <- merge_treatments(ROHR08.2500.rar, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_08_Water_phylum.rar <- ROHR08.2500.rar %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_08_Water_phylum.rar["Phylum"][ROHR_08_Water_phylum.rar["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_08_Water_phylum.rar["Phylum"][ROHR_08_Water_phylum.rar["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

```

##In progress 3/11/25

###Jump to here for loop to make species-by-species RAREFIED 2500 barplot split by region 

```{r run water separately}

Water_bar.rar <- ggplot(ROHR_08_Water_phylum.rar, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))
) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_blank(),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.ticks.x = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()    # Remove background around facet labels
  ) +
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Water_bar.rar

# Save the plot as Water_bar.png
ggsave("Water_bar.rar.png", plot = Water_bar.rar)
```
```{r loop code above for all fish}

# List of dataset suffixes
datasets <- c("conc", "dor", "xan", "lat", "nig", "hum", "pan", "lab", "col", "tro")

# Initialize an empty list to store plots
plot_list <- list()

# Loop over dataset names and generate plots
for (dataset in datasets) {
  # Construct dataset name dynamically
  data_name <- paste0("ROHR_06_", dataset, "_phylum.rar")
  
  # Dynamically create the plot for each dataset
  plot <- ggplot(get(data_name), aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) +
    geom_bar(stat = "identity", position = "fill") +
    scale_fill_manual(values = c44, name = "Microbial Phyla") +
    scale_x_discrete() +
    scale_y_continuous(expand = c(0, 0)) +
    guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
    theme(
      legend.position = "none",
      axis.text.x = element_blank(),
      axis.title.x = element_blank(),
      axis.text.y = element_blank(),
      axis.title.y = element_blank(),
      axis.ticks.y = element_blank(),
      axis.ticks.x = element_blank(),
      strip.placement = "outside",  # Place the facet strips outside the plot area
      strip.text = element_text(size = 12),  # Customize facet label text size
      strip.background = element_blank()    # Remove background around facet labels
    ) +
    facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom
  
  # Save the plot with a dynamic name
  plot_name <- paste0(dataset, "_bar.rar")
  ggsave(paste0(plot_name, ".png"), plot = plot)

# Append each plot to the list
  plot_list[[dataset]] <- plot
}

# Add the Water_bar plot to the list
plot_list[["water"]] <- Water_bar.rar

# Combine all the plots into one grid
combined_plot2 <- plot_grid(plotlist = plot_list, ncol = 2)  # Adjust ncol as needed

# Display the combined plot
print(combined_plot2)

# Optionally, save the combined plot to a file
ggsave("Combined_barplots_rarefied2500.png", plot = combined_plot2, width = 8, height = 11, units = "in")
```

```{r add labels to see where each season ends}

`# Define colors for the seasons
season_colors <- c("non-upwelling" = "#660066", "upwelling" = "#336633") 

Aconc_bar2.rar <- ggplot(
  ROHR_06_conc_phylum.rar,
  aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()) +  # Remove background around facet labels
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Aconc_bar2.rar

```
```{r loop named dataset for all fish}
# Loop over dataset names and generate plots
for (dataset in datasets) {
  # Construct dataset name dynamically
  data_name <- paste0("ROHR_06_", dataset, "_phylum.rar")
  
  # Dynamically create the plot for each dataset
  plot2.rar <- ggplot(get(data_name), aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) +
    geom_bar(stat = "identity", position = "fill") +
    scale_fill_manual(values = c44, name = "Microbial Phyla") +
    scale_x_discrete() +
    scale_y_continuous(expand = c(0, 0)) +
    guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
    theme(
      legend.position = "none",
      axis.text.x = element_text(angle = 90, hjust = 1),
      axis.title.x = element_blank(),
      axis.text.y = element_blank(),
      axis.title.y = element_blank(),
      axis.ticks.y = element_blank(),
      axis.ticks.x = element_blank(),
      strip.placement = "outside",  # Place the facet strips outside the plot area
      strip.text = element_text(size = 12),  # Customize facet label text size
      strip.background = element_blank()    # Remove background around facet labels
    ) +
    facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom
  
  # Save the plot with a dynamic name
  plot_name2 <- paste0(dataset, "_stackedbar2_rarefied2500")
  ggsave(paste0(plot_name2, ".png"), plot = plot2.rar)
}
```

```{r run water}

Water_bar2.rar <- ggplot(ROHR_08_Water_phylum.rar, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))
) +
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name = "Microbial Phyla") +
  scale_x_discrete() +
  scale_y_continuous(expand = c(0, 0)) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  theme(
    legend.position = "none",
    axis.text.x = element_text(angle = 90, hjust = 1),
    axis.title.x = element_blank(),
    axis.text.y = element_blank(),
    axis.title.y = element_blank(),
    axis.ticks.y = element_blank(),
    axis.ticks.x = element_blank(),
    strip.placement = "outside",  # Place the facet strips outside the plot area
    strip.text = element_text(size = 12),  # Customize facet label text size
    strip.background = element_blank()    # Remove background around facet labels
  ) +
  facet_wrap(~ region, scales = "free_x", ncol = 2, strip.position = "bottom")  # Move region labels to bottom

Water_bar2.rar

# Save the plot as Water_bar.png
ggsave("Water_bar2.rar.png", plot = Water_bar2.rar)
```


##CUT BELOW ONCE FIXED ABOVE
```{r RAREFIED multibar plot}

multi_bar_rar <- plot_grid(Aconc_bar.rar, Mdor_bar.rar, Axanth_bar.rar, Plat_bar.rar, Jnig_bar.rar, 
                        Chum_bar.rar, Ccol_bar.rar, Atrosch_bar.rar, Cpan_bar.rar, Elab_bar.rar, 
                        Water_bar.rar, legendMaster2, 
          labels=c("A","B","C","D","E","F","G","H","I","J","K"),
          label_size = 15, 
          hjust = 0,
          vjust = 2,
          nrow = 3,
          scale = 0.95)
multi_bar_rar

```
```{r save barchart with rarefied data png, echo = FALSE}
ggsave(filename = "Fish skin barchart RAREFIED V1.png", plot = multi_bar_rar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='png', dpi=700)
```

```{r save barchart with rarefied data pdf, echo = FALSE}
ggsave(filename = "Fish skin barchart RAREFIED V1.pdf", plot = multi_bar_rar, path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 16, height = 10, device='pdf', dpi=700)
```

