---
title: "Script 8 Core Microbiome Test"
author: "Laura Lardinois"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

--> Script N. 8 - Testing Core Microbiome Analyses

Most recent updates: 
- 2026-02-06 - LLL - create base script

In many host-associated microbiome studies, researchers try to determine a 'core' microbiome, which is meant to reflect truly 'host-associated' rather than transient (i.e., environmental microbes - or ones from the diet or other hosts that may have been in recent contact with the target host) microbial taxa. 

There are many different ways for doing this - for instance, in Clever et al. (2022) they use an Indicator Analysis using the `indval` function of the labdsv package (ordination & multivariate analysis for ecology), which "calculates the indicator value (fidelity and relative abundance) of species in clusters or types" (from the labdsv Rdocumentation page). To do this, they compare microbes in fish, the environment, and potential prey items and they obtain p-values using indicator analysis, correct for multiple tests (Bonferroni correction), then extract the significant (setting a threshold of p < 0.01) ASVs associated with fish. 

Clever et al. (2022) code tutorial: https://bocasbiome.github.io/wf1.html#run-indicator-analysis-to-identify-core-taxa

Another approach for this, as is offered by the microbiome package, would be to transform the data to composition (accounting for differences in sequencing depth), then calculate the 'core' as the taxa that exceed a given prevalence and detection threshold (e.g., relative abundance > 1% (detection), and found in >= 50% of samples). This has the advantage of not needing non-target microbiome samples (e.g. water vs. fish) to calculate a 'core' microbiome, but leaves the decision of what constitutes a reasonable threshold and prevalence up to the researcher (without really having an informative statistical decision-making framework behind this).

See microbiome package 'core' tutorial here: https://microbiome.github.io/tutorials/Core.html

```{r load required packages}
library(phyloseq) #microbiome data handling - phyloseq obj
library(microbiome) #multiple functions for analyzing and visualizing microbiome data
library(labdsv) #ordination & multivariate analysis for ecology, including indicator analysis
library(here) #simplify calls to read/write data to project folder with relative positioning
library(tidyverse) #data manipulation in tidy format
```

We'll load our whole fish dataset to start, but it may make more sense to calculate a core microbiome per host species, given the differences between hosts, and/or merge the fish dataset with the water for these analyses to be able to run the indicator analysis (if I understand what it's doing correctly)

```{r read in fish microbiome data from script 3 b-div}
#I'm being lazy here and just copying intermediate phyloseq objects and other R files generated by these new analyses to the GitHub/git-tracked project

ROHR_06_obj_phylo_root <- readRDS(here("Intermediate_R_objects/ROHR_06_obj_with_rooted_phylogenetic_tree_updated_names.rds"))
```

### CORE MICROBIOME TEST #1 - microbiome package + mods from Bocasbiome scripts

##### NOTE: transformation to compositional is from microbiome workflow

```{r transform data to compositional}
# transform to compositional(relative abundances)
ROHR_06.rel <- microbiome::transform(ROHR_06_obj_phylo_root, "compositional")
```

#### NOTE: data prep steps below are from Indicator Analysis workflow by Jarrod Scott (they don't convert to compositional data...)

```{r bind ASVs and convert tips from sequence to ASV IDs}
#bind ASVs to tax table
tax_table(ROHR_06.rel) <- cbind(tax_table(ROHR_06.rel),
                                           rownames(tax_table(ROHR_06.rel)))

colnames(tax_table(ROHR_06.rel)) <-
        c("Kingdom", "Phylum", "Class",
          "Order", "Family", "Genus", "ASV")

# replace ASV sequences with ASV IDs
taxa_names(ROHR_06.rel) <- 1:ntaxa(ROHR_06.rel)
taxa_names(ROHR_06.rel) <- paste("ASV_",
                                 taxa_names(ROHR_06.rel),
                                 sep = "") #this gives you "ASV_1", "ASV_2", assigned sequentially
```

```{r re-make phyloseq with updated structure}
ROHR_06.rel.all <- data.frame(tax_table(ROHR_06.rel))
ROHR_06.rel.new <- data.frame(ASV_ID = row.names(ROHR_06.rel.all),
                            ROHR_06.rel.all)
ROHR_06.rel.new <- ROHR_06.rel.new[, c(2,3,4,5,6,7,1,8)]
ROHR_06.rel.new2 <- as.matrix(ROHR_06.rel.new)
ROHR_06.rel <- merge_phyloseq(otu_table(ROHR_06.rel),
                                         tax_table(ROHR_06.rel.new2),
                                         sample_data(ROHR_06.rel),
                                         phy_tree(ROHR_06.rel))
saveRDS(ROHR_06.rel, "Intermediate_R_objects/ROHR_06_rel_ASVID.rds")
```

Not all taxa have assignments to genus level, so we'll want to round to most detailed taxonomic level that is not NA

```{r round to non-NA taxonomy}
#rename a copy to mess with taxononomic assignments
ROHR_06_rel_LCA <- ROHR_06.rel

# take highest assigned taxonomic resolution
tax.clean <- data.frame(tax_table(ROHR_06_rel_LCA))
for (i in 1:6){ tax.clean[,i] <- as.character(tax.clean[,i])}
tax.clean[is.na(tax.clean)] <- ""

for (i in 1:nrow(tax.clean)){
    if (tax.clean[i,2] == ""){
        Kingdom <- base::paste("k", tax.clean[i,1], sep = "_")
        tax.clean[i, 2:6] <- Kingdom
    } else if (tax.clean[i,3] == ""){
        Phylum <- base::paste("p", tax.clean[i,2], sep = "_")
        tax.clean[i, 3:6] <- Phylum
    } else if (tax.clean[i,4] == ""){
        Class <- base::paste("c", tax.clean[i,3], sep = "_")
        tax.clean[i, 4:6] <- Class
    } else if (tax.clean[i,5] == ""){
        Order <- base::paste("o", tax.clean[i,4], sep = "_")
        tax.clean[i, 5:6] <- Order
    } else if (tax.clean[i,6] == ""){
        tax.clean$Genus[i] <- base::paste("f",tax.clean$Family[i], sep = "_")
        }
}
rm(Class, Order, Phylum, Kingdom) #if you get an error "...'Kingdom' not found", this likely means all are assigned to kingdom level

#append to ASV names
tax.clean <- tax.clean %>% unite("ASV_IDa", Genus:ASV_ID,
                                 remove = FALSE, sep = "_")
tax.clean <- tax.clean %>% unite("ASV_IDb", ASV_ID:Genus,
                                 remove = FALSE, sep = "_")
tax.clean <- tax.clean[, c(1,2,3,4,5,8,9,6,7,10)]
tax.clean$ASV_IDa <-
  str_replace_all(tax.clean$ASV_IDa,
                  'Clostridium_sensu_stricto_[0-9]',
                  'Clostridium')
tax.clean$ASV_IDb <-
  str_replace_all(tax.clean$ASV_IDb,
                  'Clostridium_sensu_stricto_[0-9]',
                  'Clostridium')
tax.clean$ASV_IDc <- tax.clean$ASV_IDa
tax.clean$ASV_IDc <-
  str_replace_all(tax.clean$ASV_IDc,
                  '_ASV', '')
tax.clean <- tax.clean[, c(1,2,3,4,5,6,7,8,9,11,10)]
write.csv(tax.clean, "Intermediate_R_objects/ROHR_06_tax_no_na.csv")
```

Now, we can add the taxonomy back to the phyloseq object with no NA gaps
```{r save updated phyloseq w ASV IDs and no NAs}
tax_table(ROHR_06_rel_LCA) <- as.matrix(tax.clean)
rank_names(ROHR_06_rel_LCA)
saveRDS(ROHR_06_rel_LCA, "Intermediate_R_objects/ROHR_06_rel_ASVID_no_NA.rds")
```

```{r remove rownames from ASV table}
data.IndVal.env.fish <- data.frame(otu_table(ROHR_06_rel_LCA))
data.IndVal.env.fish
data.IndVal.ASV <- tibble::remove_rownames(data.IndVal.env.fish)
```

# IN PROGRESS 02/06/26 - FOR RUNNING INDICATOR SPECIES ANALYSIS






After these prep step, we can try out running the `core_members` function from the microbiome package!

#### RUN CORE MICROBIOME TEST - 'core_members'

```{r run core taxa function at 1% detection and 50% prevalence}
core.taxa.50 <- core_members(ROHR_06_rel_LCA, detection = 0.0001, prevalence = 50/100)

core.taxa.50
```

We now have a list of 44 ASVs - however, these codes don't tell us much more than the sequences...

```{r }
# 
tax_table(ROHR_06.rel)[, colnames(tax_table(ROHR_06.rel))] <- gsub(tax_table(ROHR_06.rel)[, colnames(tax_table(ROHR_06.rel))],  pattern = "[a-z]__", replacement = "")

# Use the microbiome function add_besthit to get taxonomic identities of ASVs.
ROHR_06.rel.f <- microbiome::add_besthit(ROHR_06.rel, sep = " ")

# Check 
taxa_names(ROHR_06.rel.f)[1:10]
```





