---
title: "ROHR_06_Swab_Species_by_Species"
author: ""
date: "2023-02-13"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Script N. 3.2 - species-by-species beta diversity analyses


Most recent edit: 

02/07/2025 - LLL - adding Jaccard & UniFrac PERMANOVAS

11/14/23

Project summary

- Fish were collected from 2 sites in the Tropical Eastern Pacific: Las Perlas & Coiba. 
- Sampling during both upwelling & non-upwelling seasons (collections done 2021-22). 
- 10 target species
- Swabs = 9 per species.
- Control swabs (swabs exposed to air in field) collected at each site and season
- Swabs were stored in liquid nitrogen for transport from field to lab, then at -80ÂºC
- DNA was extracted using ZymoBIOMICS 96 well DNA kit - 4 plates total
- DNA was amplified following a modified version of the Earth Microbiome Project 16S protocol using phased primers (1 set per plate)
- Negative PCR controls and extraction controls included in each plate
- Index PCR performed to attach unique barcodes, then all four plates (384 samples) pooled for sequencing on the Illumina MiSeq sequencing platform of the Smithsonian Tropical Research Institute's (STRI) Naos facilities in Panama (courtesy of Marta Vargas)
- Sequenced libraries were demultiplexed at STRI by Marta Vargas using the MiSeq Reporter Software
- These libraries were trimmed with Cutadapt to remove primers before being read into R to start the DADA2 pipeline (following tutorial v 1.16) 
- Silva version 138.1 (mar 10, 2021 update) used for taxonomic assignments
- See "Skin Pre Processing RRR Fish 16S Sequences Summer 2022" document for pipeline and processing up to the creation of the initial phyloseq object

- This code was adapted from various source files including Matt Leray's BCS work, the Bocasbiome project, and several tutorial pages

Loading our phyloseq object - filtered & trimmed (not rarefied) with phylogenetic tree 
as well as our rarefied dataset (to 2500 reads) - from "ROHR_06_Swab_stats_alpha.Rmd"
```{r read phyloseq object, echo=FALSE}
ROHR_06_obj_phylo_root <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_obj_with_rooted_phylogenetic_tree_updated_names.rds")

#read rarefied data
ROHR06.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_rarefied_2500_tree.rds")
```


```{r load required packages, echo=FALSE}
library(dada2)
library(ggplot2)
library(ff)
library(phyloseq)
library(gridExtra)
library(dplyr)
library(decontam)
library(ape)
library(reltools)
library(ampvis2)
library(edgeR)
library(PathoStat)
library(ape)
library(phangorn)
library(seqinr)
library(paco)
library(MCMCglmm)
library(phytools)
library(vegan)
library(reshape2)
library(pairwiseAdonis) #for pairwise comparisons following PERMANOVA (adonis2 function in vegan)
library(GUniFrac) #to perform Generalized UniFrac Distance calculation
library(hillR) #hill numbers package - calculate taxonomic, functional, and phylogenetic diversity via Hill's #s
library(forcats) #to reorder samples by trophic level (or other characteristic)
library(microbiome) #to work with microbiome data, including "meta" function used to extract metadata from phyloseq
library(ggtext) #improved text rendering for ggplot2 (allows more customization of fonts, text, etc.)
library(phylosmith) # package to create merged meta-data column
```

```{r subsetting full dataset by species}

####subsetting data by species


####Abudefduf concolor

ROHR_06_Aconcolor <- subset_samples(ROHR_06_obj_phylo_root, species == "Abudefduf concolor")
ROHR_06_Aconcolor # 38 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Aconcolor <- prune_taxa(taxa_sums(ROHR_06_Aconcolor) > 0, ROHR_06_Aconcolor)
ROHR_06_Aconcolor # 38 samples, 3055 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Aconcolor_prop = transform_sample_counts(ROHR_06_Aconcolor, function(x) x/sum(x))


####Abudefduf troschelii

ROHR_06_Atroschelii <- subset_samples(ROHR_06_obj_phylo_root, species == "Abudefduf troschelii")
ROHR_06_Atroschelii #35 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Atroschelii <- prune_taxa(taxa_sums(ROHR_06_Atroschelii) > 0, ROHR_06_Atroschelii)
ROHR_06_Atroschelii #35 samples, 2684 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Atroschelii_prop = transform_sample_counts(ROHR_06_Atroschelii, function(x) x/sum(x))


####Acanthurus xanthopterus

ROHR_06_Axanthopterus <- subset_samples(ROHR_06_obj_phylo_root, species == "Acanthurus xanthopterus")
ROHR_06_Axanthopterus # 35 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Axanthopterus <- prune_taxa(taxa_sums(ROHR_06_Axanthopterus) > 0, ROHR_06_Axanthopterus)
ROHR_06_Axanthopterus # 35 samples, 2846 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Axanthopterus_prop = transform_sample_counts(ROHR_06_Axanthopterus, function(x) x/sum(x))


####Cephalopholis colonus

ROHR_06_Ccolonus <- subset_samples(ROHR_06_obj_phylo_root, species == "Cephalopholis colonus")
ROHR_06_Ccolonus # 30 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Ccolonus <- prune_taxa(taxa_sums(ROHR_06_Ccolonus) > 0, ROHR_06_Ccolonus)
ROHR_06_Ccolonus # 30 samples, 2876 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Ccolonus_prop = transform_sample_counts(ROHR_06_Ccolonus, function(x) x/sum(x))


####Cephalopholis panamensis

ROHR_06_Cpanamensis <- subset_samples(ROHR_06_obj_phylo_root, species == "Cephalopholis panamensis")
ROHR_06_Cpanamensis # 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Cpanamensis <- prune_taxa(taxa_sums(ROHR_06_Cpanamensis) > 0, ROHR_06_Cpanamensis)
ROHR_06_Cpanamensis # 33 samples, 3537 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Cpanamensis_prop = transform_sample_counts(ROHR_06_Cpanamensis, function(x) x/sum(x))


####Chaetodon humeralis

ROHR_06_Chumeralis <- subset_samples(ROHR_06_obj_phylo_root, species == "Chaetodon humeralis")
ROHR_06_Chumeralis # 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Chumeralis <- prune_taxa(taxa_sums(ROHR_06_Chumeralis) > 0, ROHR_06_Chumeralis)
ROHR_06_Chumeralis # 33 samples, 1504 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Chumeralis_prop = transform_sample_counts(ROHR_06_Chumeralis, function(x) x/sum(x))


####Epinephelus labriformis

ROHR_06_Elabriformis <- subset_samples(ROHR_06_obj_phylo_root, species == "Epinephelus labriformis")
ROHR_06_Elabriformis # 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Elabriformis <- prune_taxa(taxa_sums(ROHR_06_Elabriformis) > 0, ROHR_06_Elabriformis)
ROHR_06_Elabriformis # 33 samples, 4121 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Elabriformis_prop = transform_sample_counts(ROHR_06_Elabriformis, function(x) x/sum(x))


####Johnrandallia nigrirostris

ROHR_06_Jnigrirostris <- subset_samples(ROHR_06_obj_phylo_root, species == "Johnrandallia nigrirostris")
ROHR_06_Jnigrirostris # 35 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Jnigrirostris <- prune_taxa(taxa_sums(ROHR_06_Jnigrirostris) > 0, ROHR_06_Jnigrirostris)
ROHR_06_Jnigrirostris # 35 samples, 1889 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Jnigrirostris_prop = transform_sample_counts(ROHR_06_Jnigrirostris, function(x) x/sum(x))


####Microspathodon dorsalis

ROHR_06_Mdorsalis <- subset_samples(ROHR_06_obj_phylo_root, species == "Microspathodon dorsalis")
ROHR_06_Mdorsalis # 36 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Mdorsalis <- prune_taxa(taxa_sums(ROHR_06_Mdorsalis) > 0, ROHR_06_Mdorsalis)
ROHR_06_Mdorsalis # 36 samples, 3558 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Mdorsalis_prop = transform_sample_counts(ROHR_06_Mdorsalis, function(x) x/sum(x))


####Prionurus laticlavius

ROHR_06_Platiclavius <- subset_samples(ROHR_06_obj_phylo_root, species == "Prionurus laticlavius")
ROHR_06_Platiclavius # 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Platiclavius <- prune_taxa(taxa_sums(ROHR_06_Platiclavius) > 0, ROHR_06_Platiclavius)
ROHR_06_Platiclavius # 33 samples, 2930 taxa (full dataset = 6551 taxa, 341 samples)

##transforming to counts
ROHR_06_Platiclavius_prop = transform_sample_counts(ROHR_06_Platiclavius, function(x) x/sum(x))


# Save Phyloseq objects

#A. concolor
saveRDS(ROHR_06_Aconcolor, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_concolor.rds")

#A. troschelii
saveRDS(ROHR_06_Atroschelii, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_troschelii.rds")

#A. xanthopterus
saveRDS(ROHR_06_Axanthopterus, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_xanthopterus.rds")

#C. colonus
saveRDS(ROHR_06_Ccolonus, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_colonus.rds")

#C. panamensis
saveRDS(ROHR_06_Cpanamensis, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_panamensis.rds")

#C. humeralis
saveRDS(ROHR_06_Chumeralis, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_humeralis.rds")

#E. labriformis
saveRDS(ROHR_06_Elabriformis, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_E_labriformis.rds")

#J. nigrirostris
saveRDS(ROHR_06_Jnigrirostris, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_J_nigrirostris.rds")

#M. dorsalis
saveRDS(ROHR_06_Mdorsalis, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_M_dorsalis.rds")

#P. laticlavius
saveRDS(ROHR_06_Platiclavius, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_laticlavius.rds")

```

```{r read subset dataset}
#temporarily set working directory
setwd("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed")

#A. concolor
ROHR_06_Aconcolor <- readRDS( "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_concolor.rds")

#A. troschelii
ROHR_06_Atroschelii <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_troschelii.rds")

#A. xanthopterus
ROHR_06_Axanthopterus <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_xanthopterus.rds")

#C. colonus
ROHR_06_Ccolonus <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_colonus.rds")

#C. panamensis
ROHR_06_Cpanamensis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_panamensis.rds")

#C. humeralis
ROHR_06_Chumeralis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_humeralis.rds")

#E. labriformis
ROHR_06_Elabriformis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_E_labriformis.rds")

#J. nigrirostris
ROHR_06_Jnigrirostris <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_J_nigrirostris.rds")

#M. dorsalis
ROHR_06_Mdorsalis <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_M_dorsalis.rds")

#P. laticlavius
ROHR_06_Platiclavius <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_laticlavius.rds")
```
```{r subsetting rarefied dataset by species}

####subsetting data by species

####Abudefduf concolor

ROHR_06_Aconcolor.rar <- subset_samples(ROHR06.rar, species == "Abudefduf concolor")
ROHR_06_Aconcolor.rar # 32 down from 38 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Aconcolor.rar <- prune_taxa(taxa_sums(ROHR_06_Aconcolor.rar) > 0, ROHR_06_Aconcolor.rar)
ROHR_06_Aconcolor.rar # 32 samples, 2622 down from 3055 taxa

##transforming to counts
ROHR_06_Aconcolor.rar_prop = transform_sample_counts(ROHR_06_Aconcolor.rar, function(x) x/sum(x))


####Abudefduf troschelii

ROHR_06_Atroschelii.rar <- subset_samples(ROHR06.rar, species == "Abudefduf troschelii")
ROHR_06_Atroschelii.rar #24 down from 35 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Atroschelii.rar <- prune_taxa(taxa_sums(ROHR_06_Atroschelii.rar) > 0, ROHR_06_Atroschelii.rar)
ROHR_06_Atroschelii.rar #24 samples, 2190 down from 2684 taxa

##transforming to counts
ROHR_06_Atroschelii.rar_prop = transform_sample_counts(ROHR_06_Atroschelii.rar, function(x) x/sum(x))


####Acanthurus xanthopterus

ROHR_06_Axanthopterus.rar <- subset_samples(ROHR06.rar, species == "Acanthurus xanthopterus")
ROHR_06_Axanthopterus.rar # 33 down from 35 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Axanthopterus.rar <- prune_taxa(taxa_sums(ROHR_06_Axanthopterus.rar) > 0, ROHR_06_Axanthopterus.rar)
ROHR_06_Axanthopterus.rar # 33 samples, 2412 down from 2846 taxa 

##transforming to counts
ROHR_06_Axanthopterus.rar_prop = transform_sample_counts(ROHR_06_Axanthopterus.rar, function(x) x/sum(x))

####Cephalopholis colonus

ROHR_06_Ccolonus.rar <- subset_samples(ROHR06.rar, species == "Cephalopholis colonus")
ROHR_06_Ccolonus.rar # 27 down from 30 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Ccolonus.rar <- prune_taxa(taxa_sums(ROHR_06_Ccolonus.rar) > 0, ROHR_06_Ccolonus.rar)
ROHR_06_Ccolonus.rar # 27 samples, 2519 down from 2876 taxa

##transforming to counts
ROHR_06_Ccolonus.rar_prop = transform_sample_counts(ROHR_06_Ccolonus.rar, function(x) x/sum(x))


####Cephalopholis panamensis

ROHR_06_Cpanamensis.rar <- subset_samples(ROHR06.rar, species == "Cephalopholis panamensis")
ROHR_06_Cpanamensis.rar # 26 down from 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Cpanamensis.rar <- prune_taxa(taxa_sums(ROHR_06_Cpanamensis.rar) > 0, ROHR_06_Cpanamensis.rar)
ROHR_06_Cpanamensis.rar # 26 samples, 3046 down from 3537 taxa

##transforming to counts
ROHR_06_Cpanamensis.rar_prop = transform_sample_counts(ROHR_06_Cpanamensis.rar, function(x) x/sum(x))


####Chaetodon humeralis

ROHR_06_Chumeralis.rar <- subset_samples(ROHR06.rar, species == "Chaetodon humeralis")
ROHR_06_Chumeralis.rar # 24 down from 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Chumeralis.rar <- prune_taxa(taxa_sums(ROHR_06_Chumeralis.rar) > 0, ROHR_06_Chumeralis.rar)
ROHR_06_Chumeralis.rar # 24 samples, 1116 down from 1504 taxa

##transforming to counts
ROHR_06_Chumeralis.rar_prop = transform_sample_counts(ROHR_06_Chumeralis.rar, function(x) x/sum(x))


####Epinephelus labriformis

ROHR_06_Elabriformis.rar <- subset_samples(ROHR06.rar, species == "Epinephelus labriformis")
ROHR_06_Elabriformis.rar # 32 down from 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Elabriformis.rar <- prune_taxa(taxa_sums(ROHR_06_Elabriformis.rar) > 0, ROHR_06_Elabriformis.rar)
ROHR_06_Elabriformis.rar # 32 samples, 3500 down from 4121 taxa

##transforming to counts
ROHR_06_Elabriformis.rar_prop = transform_sample_counts(ROHR_06_Elabriformis.rar, function(x) x/sum(x))


####Johnrandallia nigrirostris

ROHR_06_Jnigrirostris.rar <- subset_samples(ROHR06.rar, species == "Johnrandallia nigrirostris")
ROHR_06_Jnigrirostris.rar # 19 down from 35 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Jnigrirostris.rar <- prune_taxa(taxa_sums(ROHR_06_Jnigrirostris.rar) > 0, ROHR_06_Jnigrirostris.rar)
ROHR_06_Jnigrirostris.rar # 19 samples, 1345 (down from 1889) taxa

##transforming to counts
ROHR_06_Jnigrirostris.rar_prop = transform_sample_counts(ROHR_06_Jnigrirostris.rar, function(x) x/sum(x))


####Microspathodon dorsalis

ROHR_06_Mdorsalis.rar <- subset_samples(ROHR06.rar, species == "Microspathodon dorsalis")
ROHR_06_Mdorsalis.rar # 31 down from 36 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Mdorsalis.rar <- prune_taxa(taxa_sums(ROHR_06_Mdorsalis.rar) > 0, ROHR_06_Mdorsalis.rar)
ROHR_06_Mdorsalis.rar # 31 samples, 2877 down from 3558 taxa

##transforming to counts
ROHR_06_Mdorsalis.rar_prop = transform_sample_counts(ROHR_06_Mdorsalis.rar, function(x) x/sum(x))


####Prionurus laticlavius

ROHR_06_Platiclavius.rar <- subset_samples(ROHR06.rar, species == "Prionurus laticlavius")
ROHR_06_Platiclavius.rar # 31 down from 33 samples

#removings ASVs that are not present in any sample (if any)
ROHR_06_Platiclavius.rar <- prune_taxa(taxa_sums(ROHR_06_Platiclavius.rar) > 0, ROHR_06_Platiclavius.rar)
ROHR_06_Platiclavius.rar # 31 samples, 2580 down from 2930 taxa

##transforming to counts
ROHR_06_Platiclavius.rar_prop = transform_sample_counts(ROHR_06_Platiclavius.rar, function(x) x/sum(x))

```

```{r save rarefied subsets}
# Save Phyloseq objects

#A. concolor
saveRDS(ROHR_06_Aconcolor.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_concolor_rar.rds")

#A. troschelii
saveRDS(ROHR_06_Atroschelii.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_troschelii_rar.rds")

#A. xanthopterus
saveRDS(ROHR_06_Axanthopterus.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_xanthopterus_rar.rds")

#C. colonus
saveRDS(ROHR_06_Ccolonus.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_colonus_rar.rds")

#C. panamensis
saveRDS(ROHR_06_Cpanamensis.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_panamensis_rar.rds")

#C. humeralis
saveRDS(ROHR_06_Chumeralis.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_humeralis_rar.rds")

#E. labriformis
saveRDS(ROHR_06_Elabriformis.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_E_labriformis_rar.rds")

#J. nigrirostris
saveRDS(ROHR_06_Jnigrirostris.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_J_nigrirostris_rar.rds")

#M. dorsalis
saveRDS(ROHR_06_Mdorsalis.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_M_dorsalis_rar.rds")

#P. laticlavius
saveRDS(ROHR_06_Platiclavius.rar, "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_laticlavius_rar.rds")

```

```{r read rarefied 2500 subset dataset}
#temporarily set working directory
setwd("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed")

#A. concolor
ROHR_06_Aconcolor.rar <- readRDS( "/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_concolor_rar.rds")

#A. troschelii
ROHR_06_Atroschelii.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_troschelii_rar.rds")

#A. xanthopterus
ROHR_06_Axanthopterus.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_A_xanthopterus_rar.rds")

#C. colonus
ROHR_06_Ccolonus.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_colonus_rar.rds")

#C. panamensis
ROHR_06_Cpanamensis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_panamensis_rar.rds")

#C. humeralis
ROHR_06_Chumeralis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_C_humeralis_rar.rds")

#E. labriformis
ROHR_06_Elabriformis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_E_labriformis_rar.rds")

#J. nigrirostris
ROHR_06_Jnigrirostris.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_J_nigrirostris_rar.rds")

#M. dorsalis
ROHR_06_Mdorsalis.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_M_dorsalis_rar.rds")

#P. laticlavius
ROHR_06_Platiclavius.rar <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_06_trimmed/ROHR_06_P_laticlavius_rar.rds")
```

###Statistics & visualisations to be run for each fish species
- A-diversity Hill numbers

- PERMANOVA (+ dispersion & pairwise comparisons)
  Distances:
  - Bray-Curtis*
  - Jaccard
  - UniFrac
  - GUniFrac
  - WUniFrac
  
  *principal distance metric (for main body of publication)
  
- PCoAs (using Bray-Curtis distances)

- NMDS

- Stacked barplots of microbiome composition

- Differential abundance analyses w/in host species between seasons/gulfs [to be coded!!]


###Recap of stats tests, assumptions, etc.
PERMANOVA (Permutational Multivariate Analysis of Variance)

source: https://archetypalecology.wordpress.com/2018/02/21/permutational-multivariate-analysis-of-variance-permanova-in-r-preliminary/

- non-parametric, multivariate statistical test
- compares groups of objects
H0 (general) = centroids & dispersion of groups are equivalent for all groups
H0 (specific to this case) = There are no differences in the relative abundances of microbial taxa in the fish skin microbiome across all sites, species and seasons in the Tropical Eastern Pacific. 

-test statistic = pseudo-F ratio
"It compares the total sum of squared dissimilarities (or ranked dissimilarities) among objects belonging to different groups to that of objects belonging to the same group. Larger F-ratios indicate more pronounced group separation, however, the significance of this ratio is usually of more interest than its magnitude." 

Assumptions:
- objects in dataset are exchangeable under the null hypothesis
- exchangeable objects (sites, samples, observations) are independent ***** (need to account for this assumption NOT being met in test setup)
- exchangeable objects have similar multivariate dispersion (betadisper --> multivariate analogue to Levene's test for homogeneity of variances) (see code chunk below)


###Species-by-species analyses

#Abudefduf concolor

# A. concolor Hill numbers
```{r Hill numbers}

```

# A. concolor PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA A. concolor  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Aconc <- phyloseq::distance(ROHR_06_Aconcolor_prop, method = "bray")
sampledf.Aconc <- data.frame(sample_data(ROHR_06_Aconcolor_prop))

####beta dispersion####

###by region
beta.bray.region.Aconc <- betadisper(type.bray.Aconc, sampledf.Aconc$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Aconc, pairwise = TRUE, permutations = 10000) # P =  0.0371 variance is not homogeneous

plot(beta.bray.region.Aconc, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Aconc <- betadisper(type.bray.Aconc, sampledf.Aconc$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.0127 variance is not homogeneous
permutest(beta.bray.season.Aconc, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Aconc, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Aconc <- as(sample_data(ROHR_06_Aconcolor_prop), "data.frame")

####by region####
           
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Aconcolor_prop, method="bray") ~ region,
        data = metadata.Aconc) #significant, but very small R2 --> p = 0.008, R2 = 0.044

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Aconcolor_prop, method="bray") ~ season,
        data = metadata.Aconc) #significant, p = 0.001, R2 = 0.057

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Aconcolor_prop, method="bray") ~ region*season,
        data = metadata.Aconc) #no significant interaction b/w region & season (p = 0.13), but both have significant independent effects for A. concolor (0.004 & 0.001)


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Aconc, factors = sampledf.Aconc$region) #p = 0.006, R2 = 0.0445

###by season
set.seed(1911)
pairwise.adonis(type.bray.Aconc, factors = sampledf.Aconc$season) #p = 0.001, R2 = 0.057
```

```{r PERMANOVA A. concolor  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Aconcolor, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Aconcolor))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P = 0.1718 is homogeneous across regions

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.4158 is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000) # p = 0.0013
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000) # p = 0.00009999
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

#adonis2(formula = type.jaccard ~ region * season, data = sampledf, permutations = 10000)
#              Df SumOfSqs      R2      F     Pr(>F)    
#region         1   0.5503 0.03736 1.4368  0.0006999 ***
#season         1   0.6461 0.04386 1.6869 0.00009999 ***
#region:season  1   0.5120 0.03476 1.3367  0.0030997 ** 
#Residual      34  13.0226 0.88403                      
#Total         37  14.7310 1.00000 

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) # p = 0.002

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) # p = 0.001
```

```{r PERMANOVA A. concolor  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Aconcolor, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Aconcolor))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 0.1648

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.4533
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

#adonis2(formula = type.unifrac ~ region, data = sampledf, permutations = 10000)
#         Df SumOfSqs      R2      F Pr(>F)
#region    1   0.2094 0.02594 0.9586 0.4101
#Residual 36   7.8623 0.97406              
#Total    37   8.0717 1.00000 

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

#adonis2(formula = type.unifrac ~ season, data = sampledf, permutations = 10000)
#         Df SumOfSqs      R2      F Pr(>F)  
#season    1   0.3986 0.04938 1.8702 0.0375 *
#Residual 36   7.6731 0.95062                
#Total    37   8.0717 1.00000  

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

#adonis2(formula = type.unifrac ~ region * season, data = sampledf, permutations = 10000)
#              Df SumOfSqs      R2      F Pr(>F)  
#region         1   0.2094 0.02594 0.9877 0.3761  
#season         1   0.3986 0.04938 1.8805 0.0374 *
#region:season  1   0.2568 0.03182 1.2117 0.2025  
#Residual      34   7.2069 0.89286                
#Total         37   8.0717 1.00000 

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 0.423	

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 0.039
```

# A. concolor PCoAs
```{r PCoA (Bray) A. concolor, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Aconc_pcoa <- ordinate(
  physeq = ROHR_06_Aconcolor_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Abudefduf concolor Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "A concolor Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# A. concolor NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Aconc_nmds <- ordinate(
  physeq = ROHR_06_Aconcolor_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Abudefduf concolor Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Aconcolor_prop,
  ordination = Aconc_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Abudefduf concolor Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#A. concolor stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_phylum <- ROHR_06_obj_phylo_root %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(Phylum)                                      # Sort data frame alphabetically by phylum

#pick colors for bars
c23 <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3")

print(c23)

unique(ROHR_06_phylum$Phylum)

#rename 2 phyla (v. long names currently)
ROHR_06_phylum["Phylum"][ROHR_06_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_phylum["Phylum"][ROHR_06_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

#check this worked
unique(ROHR_06_phylum$Phylum) #yay!

#[1] "Acidobacteriota"   "Actinobacteriota"  "Bacteroidota"      "Bdellovibrionota"  "Campylobacterota" 
# [6] "Crenarchaeota"     "Cyanobacteria"     "Deinococcota"      "Desulfobacterota"  "Fibrobacterota"   
# [11] "Firmicutes"        "Fusobacteriota"    "Marinimicrobia"    "Myxococcota"       "Nanoarchaeota"    
# [16] "Nitrospirota"      "Patescibacteria"   "Planctomycetota"   "Proteobacteria"    "SAR324 clade"     
# [21] "Spirochaetota"     "Thermoplasmatota"  "Verrucomicrobiota"

##assign names to this vector of hex color values: names have to correspond to the labels of the groups that we want to colorize

names(c23) = unique(ROHR_06_phylum$Phylum)
print(c23)

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Aconcolor)$RRR_number <- as.character(sample_data(ROHR_06_Aconcolor)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Aconcolor <- merge_treatments(ROHR_06_Aconcolor, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_conc_phylum <- ROHR_06_Aconcolor %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_conc_phylum["Phylum"][ROHR_06_conc_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_conc_phylum["Phylum"][ROHR_06_conc_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_conc_phylum$Phylum) #31 unique phyla (compared to 23 in the filtered full dataset) --> need to extend number of available colors from c25 list: currently "darkorange4" and "brown" are unassigned, thus only need 6 more colors

#[1] "Acidobacteriota"   "Actinobacteriota"  "Bacteroidota"      "Bdellovibrionota"  "Calditrichota"    
#[6] "Campylobacterota"  "Chloroflexi"       "Crenarchaeota"     "Cyanobacteria"     "Dadabacteria"     
#[11] "Deinococcota"      "Desulfobacterota"  "Entotheonellaeota" "Fibrobacterota"    "Firmicutes"       
#[16] "Fusobacteriota"    "Gemmatimonadota"   "Latescibacterota"  "Margulisbacteria"  "Marinimicrobia"   
#[21] "Myxococcota"       "NB1-j"             "Nanoarchaeota"     "PAUC34f"           "Patescibacteria"  
#[26] "Planctomycetota"   "Proteobacteria"    "SAR324 clade"      "Thermoplasmatota"  "Verrucomicrobiota"
#[31] "WPS-2"  

#of the phyla above, c("Calditrichota", "Chloroflexi", "Dadabacteria", "Entotheonellaeota", "Gemmatimonadota", "Latescibacterota", "Margulisbacteria", "NB1-j", "PAUC34f", "WPS-2") were not present in the filtered ROHR_06_phylum dataset

new_phyla <- c("Calditrichota", "Chloroflexi", "Dadabacteria", "Entotheonellaeota", "Gemmatimonadota", "Latescibacterota", "Margulisbacteria", "NB1-j", "PAUC34f", "WPS-2")

#pick colors for these phyla

c10 <- c(
  "slateblue4", "wheat", "lightseagreen", 
  "mediumorchid4", "burlywood4", "brown", "orange1", 
  "yellow", "lightsteelblue1", "honeydew")
  
names(c10) = new_phyla
print(c10)

#merge old & new color assignments
c33 <- c(c23, c10)
print(c33) 
```

```{r prep data visualize most abundant taxa - A. concolor}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_conc_phylum2 <- ROHR_06_Aconcolor %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_conc_phylum2["Phylum"][ROHR_06_conc_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_conc_phylum2["Phylum"][ROHR_06_conc_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_conc_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - A concolor}


# A. concolor microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Abudefduf concolor"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_conc_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_conc_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_conc_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_conc_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c33, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("15083 C - NU", "8726 C - NU", "8770 C - NU", "8775 C - NU", "8785 C - NU", "8786 C - NU", "8790 C - NU", "8798 C - NU", "8800 C - NU", "7877 LP - NU", "8206 LP - NU", "8212 LP - NU", "8213 LP - NU", "8218 LP - NU", "8220 LP - NU", "8223 LP - NU", "8224 LP - NU", "8226 LP - NU", "8228 LP - NU", "16769 C - U", "16787 C - U", "16788 C - U", "16789 C - U", "16790 C - U", "16791 C - U", "16792 C - U", "16793 C - U", "16794 C - U", "15112 LP - U", "15114 LP - U", "15135 LP - U", "15144 LP - U", "15145 LP - U", "15151 LP - U", "15161 LP - U", "16690 LP - U", "16692 LP - U", "16693 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###A concolor barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - A concolor}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Aconcolor)[, "Phylum"])

#Convert to relative abundance
Aconc_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Aconcolor, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Aconcolor)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Aconc_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Aconc_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c33, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

###Abudefduf troschelii

# A. troschelii PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA A. troschelii  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Atro <- phyloseq::distance(ROHR_06_Atroschelii_prop, method = "bray")
sampledf.Atro <- data.frame(sample_data(ROHR_06_Atroschelii_prop))

####beta dispersion####

###by region
beta.bray.region.Atro <- betadisper(type.bray.Atro, sampledf.Atro$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Atro, pairwise = TRUE, permutations = 10000) # P =   variance is not homogeneous

plot(beta.bray.region.Atro, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Atro <- betadisper(type.bray.Atro, sampledf.Atro$season,
                         type = "centroid", bias.adjust = TRUE) # P =  variance is not homogeneous
permutest(beta.bray.season.Atro, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Atro, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Atro <- as(sample_data(ROHR_06_Atroschelii_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Atroschelii_prop, method="bray") ~ region,
        data = metadata.Atro) # NSD

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Atroschelii_prop, method="bray") ~ season,
        data = metadata.Atro) # NSD

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Atroschelii_prop, method="bray") ~ region*season,
        data = metadata.Atro) # NSD & no interaction b/w region & season


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Atro, factors = sampledf.Atro$region) #p = 0.074, R2 = 0.037 (NSD)

###by season
set.seed(1911)
pairwise.adonis(type.bray.Atro, factors = sampledf.Atro$season) #p = 0.709, R2 = 0.026 (NSD)
```

```{r PERMANOVA A. troschelii  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Atroschelii, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Atroschelii))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA A. troschelii  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Atroschelii, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Atroschelii))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# A. troschelii PCoAs
```{r PCoA (Bray) A. troschelii, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Atro_pcoa <- ordinate(
  physeq = ROHR_06_Atroschelii_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Atroschelii_prop,
  ordination = Atro_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Abudefduf troschelii Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Atroschelii_prop,
  ordination = Atro_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "A troschelii Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# A. troschelii NMDS
```{r NMDS A troschelii, echo=FALSE}
set.seed(1)

# Ordinate
Atro_nmds <- ordinate(
  physeq = ROHR_06_Atroschelii_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Atroschelii_prop,
  ordination = Atro_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Abudefduf troschelii Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Atroschelii_prop,
  ordination = Atro_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Abudefduf troschelii Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#A. troschelii stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Atroschelii)$RRR_number <- as.character(sample_data(ROHR_06_Atroschelii)$RRR)

#create new categroy with RRR & region/season merged
ROHR_06_Atroschelii <- merge_treatments(ROHR_06_Atroschelii, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_tro_phylum <- ROHR_06_Atroschelii %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_tro_phylum["Phylum"][ROHR_06_tro_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_tro_phylum["Phylum"][ROHR_06_tro_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c33 #current list of colors assigned to phyla
unique(ROHR_06_tro_phylum$Phylum)

#phyla not present in either A. concolor nor filtered ROHR_06 full dataset:
new_phyla2 <- c("MBNT15")

#pick colors for these phyla

c3 <- c("lightyellow3")
  
names(c3) = new_phyla2
print(c3)

#merge old & new color assignments
c36 <- c(c33, c3)
print(c36)
```

```{r prep data visualize most abundant taxa - A. troschelii}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_tro_phylum2 <- ROHR_06_Atroschelii %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_tro_phylum2["Phylum"][ROHR_06_tro_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_tro_phylum2["Phylum"][ROHR_06_tro_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_tro_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - A. troschelii}


# A. troschelii microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Abudefduf troschelii"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_tro_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_tro_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_tro_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_tro_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("8703 C - NU", "8707  C - NU", "8724  C - NU", "8732  C - NU", "8778  C - NU", "8781  C - NU", "8783  C - NU", "8784  C - NU", "8793  C - NU", "7849 LP - NU", "7865 LP - NU", "7905 LP - NU", "7917 LP - NU", "7921 LP - NU", "7949 LP - NU", "8214 LP - NU", "8216 LP - NU", "8217 LP - NU", "16704 C - U", "16705 C - U", "16706 C - U", "16710 C - U", "16711 C - U", "16713 C - U", "16730 C - U", "16734 C - U", "16755 C - U", "15121 LP - U", "15129  LP - U", "15130  LP - U", "15132  LP - U", "8049  LP - U", "8077  LP - U", "8081  LP - U", "8145  LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###A troschelii barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - A. troschelii}

###need to find a way to reorder these by relative abundance of phyla
##fix bars - currently have grey color (phylum/phyla not assigned a color?)

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Atroschelii)[, "Phylum"])

#Convert to relative abundance
Aconc_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Atroschelii, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Atroschelii)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Aconc_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance


#Plot - faceted by season & region
phyloseq::plot_bar(Aconc_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c36, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

#Acanthurus xanthopterus

# A. xanthopterus PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA A. xanthopterus  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Axan <- phyloseq::distance(ROHR_06_Axanthopterus_prop, method = "bray")
sampledf.Axan <- data.frame(sample_data(ROHR_06_Axanthopterus_prop))

####beta dispersion####

###by region
beta.bray.region.Axan <- betadisper(type.bray.Axan, sampledf.Axan$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Axan, pairwise = TRUE, permutations = 10000) # P =  0.1018 variance is homogeneous

plot(beta.bray.region.Axan, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Axan <- betadisper(type.bray.Axan, sampledf.Axan$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.2792 variance is homogeneous
permutest(beta.bray.season.Axan, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Axan, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Axan <- as(sample_data(ROHR_06_Axanthopterus_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Axanthopterus_prop, method="bray") ~ region,
        data = metadata.Axan) # P = 0.018, R2 = 0.0527

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Axanthopterus_prop, method="bray") ~ season,
        data = metadata.Axan) # P = 0.055 (almost significant), R2 = 0.04483 (but when modeled region*season, both become significant, with a significant interaction between the two)

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Axanthopterus_prop, method="bray") ~ region*season,
        data = metadata.Axan) # Region = P = 0.011, Season = P = 0.033, Region*Season = P = 0.003


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Axan, factors = sampledf.Axan$region) #p = 0.017, R2 = 0.053

###by season
set.seed(1911)
pairwise.adonis(type.bray.Axan, factors = sampledf.Axan$season) #p = 0.061, R2 = 0.0448
```

```{r PERMANOVA A. xanthopterus  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Axanthopterus, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Axanthopterus))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA A. xanthopterus  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Axanthopterus, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Axanthopterus))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# A. xanthopterus PCoAs
```{r PCoA (Bray) A. xanthopterus, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Axan_pcoa <- ordinate(
  physeq = ROHR_06_Axanthopterus_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop,
  ordination = Axan_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Acanthurus xanthopterus Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop,
  ordination = Axan_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "A xanthopterus Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# A. xanthopterus NMDS
```{r NMDS A xanthopterus, echo=FALSE}
set.seed(1)

# Ordinate
Axan_nmds <- ordinate(
  physeq = ROHR_06_Axanthopterus_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop,
  ordination = Axan_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Acanthurus xanthopterus Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Axanthopterus_prop,
  ordination = Axan_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Acanthurus xanthopterus Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#A. xanthopterus stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Axanthopterus)$RRR_number <- as.character(sample_data(ROHR_06_Axanthopterus)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Axanthopterus <- merge_treatments(ROHR_06_Axanthopterus, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_xan_phylum <- ROHR_06_Axanthopterus %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_xan_phylum["Phylum"][ROHR_06_xan_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_xan_phylum["Phylum"][ROHR_06_xan_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c36 #current list of colors assigned to phyla
unique(ROHR_06_xan_phylum$Phylum) #no new phyla (compared to A. trochelii)

```

```{r prep data visualize most abundant taxa - A. xanthopterus}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_xan_phylum2 <- ROHR_06_Axanthopterus %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_xan_phylum2["Phylum"][ROHR_06_xan_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_xan_phylum2["Phylum"][ROHR_06_xan_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_xan_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - A. xanthopterus}


# A. xanthopterus microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Acanthurus xanthopterus"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_xan_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_xan_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_xan_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_xan_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c36, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("15088 C - NU", "8689 C - NU", "8690 C - NU", "8704 C - NU", "8675 C - NU", "8709 C - NU", "8719 C - NU", "8758 C - NU", "8025 LP - NU", "8229 LP - NU", "8232 LP - NU", "8235 LP - NU", "8238 LP - NU", "8239 LP - NU", "8243 LP - NU", "8260 LP - NU", "8683 LP - NU", "16742 C - U", "16749 C - U", "16772 C - U", "16795 C - U", "16809 C - U", "16812 C - U", "16813 C - U", "16814 C - U", "15115 LP - U", "16695 LP - U", "16697 LP - U", "16698 LP - U", "16701 LP - U", "20319 LP - U", "8065 LP - U", "8097 LP - U", "8181 LP - U", "8189 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###A xanthopterus barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - A. xanthopterus}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Axanthopterus)[, "Phylum"])

#Convert to relative abundance
Axan_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Axanthopterus, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Axanthopterus)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Axan_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Axan_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c36, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```


# Chaetodon humeralis

# C. humeralis PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA C. humeralis  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Chum <- phyloseq::distance(ROHR_06_Chumeralis_prop, method = "bray")
sampledf.Chum <- data.frame(sample_data(ROHR_06_Chumeralis_prop))

####beta dispersion####

###by region
beta.bray.region.Chum <- betadisper(type.bray.Chum, sampledf.Chum$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Chum, pairwise = TRUE, permutations = 10000) # P = 0.113  variance is homogeneous

plot(beta.bray.region.Chum, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Chum <- betadisper(type.bray.Chum, sampledf.Chum$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.849 variance is homogeneous
permutest(beta.bray.season.Chum, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Chum, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Chum <- as(sample_data(ROHR_06_Chumeralis_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Chumeralis_prop, method="bray") ~ region,
        data = metadata.Chum) # NSD --> P = 0.127, R2 = 0.0426

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Chumeralis_prop, method="bray") ~ season,
        data = metadata.Chum) # NSD -->  P = 0.105, R2 = 0.04345

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Chumeralis_prop, method="bray") ~ region*season,
        data = metadata.Chum) # NSD, no significant interactions (p = 0.072)


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Chum, factors = sampledf.Chum$region) #p = 0.116, R2 = 0.0426

###by season
set.seed(1911)
pairwise.adonis(type.bray.Chum, factors = sampledf.Chum$season) #p = 0.119, R2 = 0.0434
```

```{r PERMANOVA C. humeralis  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Chumeralis, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Chumeralis))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA C. humeralis  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Chumeralis, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Chumeralis))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```


# C. humeralis PCoAs
```{r PCoA (Bray) C. humeralis, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Chum_pcoa <- ordinate(
  physeq = ROHR_06_Chumeralis_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Chumeralis_prop,
  ordination = Chum_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Chaetodon humeralis Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Chumeralis_prop,
  ordination = Chum_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "C humeralis Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# C. humeralis NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Chum_nmds <- ordinate(
  physeq = ROHR_06_Chumeralis_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Chumeralis_prop,
  ordination = Chum_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Chaetodon humeralis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Chumeralis_prop,
  ordination = Chum_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Chaetodon humeralis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#C. humeralis stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Chumeralis)$RRR_number <- as.character(sample_data(ROHR_06_Chumeralis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Chumeralis <- merge_treatments(ROHR_06_Chumeralis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_hum_phylum <- ROHR_06_Chumeralis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_hum_phylum["Phylum"][ROHR_06_hum_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_hum_phylum["Phylum"][ROHR_06_hum_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c36 #current list of colors assigned to phyla
unique(ROHR_06_hum_phylum$Phylum) #new phyla = c("Deferrisomatota", "Dependentiae", "LCP-89")

new_phyla3 <- c("Deferrisomatota", "Dependentiae", "LCP-89")

#pick colors for these phyla

c3bis <- c("navy", "gray40", "hotpink4")

names(c3bis) = new_phyla3

print(c3bis)

#merge old & new color assignments
c39 <- c(c36, c3bis)
print(c39) 
```

```{r prep data visualize most abundant taxa - C. humeralis}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_hum_phylum2 <- ROHR_06_Chumeralis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_hum_phylum2["Phylum"][ROHR_06_hum_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_hum_phylum2["Phylum"][ROHR_06_hum_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_hum_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - C. humeralis}


# C. humeralis microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Chaetodon humeralis"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_hum_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c39) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_hum_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c39, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_hum_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c39, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_hum_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c39, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("15086 C - NU", "15100 C - NU", "15101 C - NU", "15104 C - NU", "8695 C - NU", "8711 C - NU", "8725 C - NU", "8753 C - NU", "8799 C - NU", "8205 LP - NU", "8207 LP - NU", "8208 LP - NU", "8210 LP - NU", "8211 LP - NU", "8222 LP - NU", "16798 C - U", "16799 C - U", "16802 C - U", "16803 C - U", "16804 C - U", "16805 C - U", "16807 C - U", "16811 C - U", "15124 LP - U", "15141 LP - U", "16689 LP - U", "16691 LP - U", "20339 LP - U", "8053 LP - U", "8137 LP - U", "8141  LP - U", "8149 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###C humeralis barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - C. humeralis}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Chumeralis)[, "Phylum"])

#Convert to relative abundance
Chum_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Chumeralis, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Chumeralis)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Chum_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Chum_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c36, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

# Cephalopholis colonus

# C. colonus PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA C. colonus  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Ccol <- phyloseq::distance(ROHR_06_Ccolonus_prop, method = "bray")
sampledf.Ccol <- data.frame(sample_data(ROHR_06_Ccolonus_prop))

####beta dispersion####

###by region
beta.bray.region.Ccol <- betadisper(type.bray.Ccol, sampledf.Ccol$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Ccol, pairwise = TRUE, permutations = 10000) # P =  variance is homogeneous

plot(beta.bray.region.Ccol, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Ccol <- betadisper(type.bray.Ccol, sampledf.Ccol$season,
                         type = "centroid", bias.adjust = TRUE) # P =  variance is homogeneous
permutest(beta.bray.season.Ccol, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Ccol, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Ccol <- as(sample_data(ROHR_06_Ccolonus_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Ccolonus_prop, method="bray") ~ region,
        data = metadata.Ccol) # P = , R2 = 

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Ccolonus_prop, method="bray") ~ season,
        data = metadata.Ccol) # P = 

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Ccolonus_prop, method="bray") ~ region*season,
        data = metadata.Ccol) # Region = P = , Season = P = , Region*Season = P = 


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Ccol, factors = sampledf.Ccol$region) #p = , R2 = 

###by season
set.seed(1911)
pairwise.adonis(type.bray.Ccol, factors = sampledf.Ccol$season) #p = , R2 = 
```

```{r PERMANOVA C. colonus  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Ccolonus, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Ccolonus))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA C. colonus  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Ccolonus, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Ccolonus))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```


# C. colonus PCoAs
```{r PCoA (Bray) C. colonus, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Ccol_pcoa <- ordinate(
  physeq = ROHR_06_Ccolonus_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Ccolonus_prop,
  ordination = Ccol_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Cephalopholis colonus Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Ccolonus_prop,
  ordination = Ccol_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "P colonus Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# C. colonus NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Ccol_nmds <- ordinate(
  physeq = ROHR_06_Ccolonus_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Ccolonus_prop,
  ordination = Ccol_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Cephalopholis colonus Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Ccolonus_prop,
  ordination = Ccol_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Cephalopholis colonus Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#C. colonus stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Ccolonus)$RRR_number <- as.character(sample_data(ROHR_06_Ccolonus)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Ccolonus <- merge_treatments(ROHR_06_Ccolonus, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_col_phylum <- ROHR_06_Ccolonus %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_col_phylum["Phylum"][ROHR_06_col_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_col_phylum["Phylum"][ROHR_06_col_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c39
unique(ROHR_06_col_phylum$Phylum) #new phyla c("Sva0485")
new_phylum <- c("Sva0485")

#pick colors for these phyla

c1 <- c("sienna3")

names(c1) = new_phylum

print(c1)

#merge old & new color assignments
c40 <- c(c39, c1)
print(c40) 

```

```{r prep data visualize most abundant taxa - C. colonus}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_col_phylum2 <- ROHR_06_Ccolonus %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_col_phylum2["Phylum"][ROHR_06_col_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_col_phylum2["Phylum"][ROHR_06_col_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_col_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - C. colonus}


# C. colonus microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Cephalopholis colonus"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_col_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c40) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_col_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c40, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_col_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c40, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_col_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c40, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("8696 C - NU", "8697 C - NU", "8749 C - NU", "8755 C - NU", "8761 C - NU", "8765 C - NU", "2771 C - NU", "8777 C - NU", "8782 C - NU", "7809 LP - NU", "7853 LP - NU", "7985 LP - NU", "8209 LP - NU", "8250 LP - NU", "8252 LP - NU", "8678 LP - NU", "8679 LP - NU", "16747 C - U", "16750 C - U", "16753 C - U", "16780 C - U", "16781 C - U", "16783 C - U", "16784 C - U", "15137 LP - U", "15138 LP - U", "16694 LP - U", "16696 LP - U", "20307 LP - U", "20355 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
### C colonus barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - C. colonus}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Ccolonus)[, "Phylum"])

#Convert to relative abundance
Ccol_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Ccolonus, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Ccolonus)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Ccol_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Ccol_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c40, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

# Cephalopholis panamensis

# C. panamensis PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA C. panamensis  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Cpan <- phyloseq::distance(ROHR_06_Cpanamensis_prop, method = "bray")
sampledf.Cpan <- data.frame(sample_data(ROHR_06_Cpanamensis_prop))

####beta dispersion####

###by region
beta.bray.region.Cpan <- betadisper(type.bray.Cpan, sampledf.Cpan$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Cpan, pairwise = TRUE, permutations = 10000) # P = 0.08 variance is homogeneous

plot(beta.bray.region.Cpan, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Cpan <- betadisper(type.bray.Cpan, sampledf.Cpan$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.34 variance is homogeneous
permutest(beta.bray.season.Cpan, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Cpan, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Cpan <- as(sample_data(ROHR_06_Cpanamensis_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Cpanamensis_prop, method="bray") ~ region,
        data = metadata.Cpan) # P = 0.002, R2 = 0.056

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Cpanamensis_prop, method="bray") ~ season,
        data = metadata.Cpan) # P = 0.004, R2 = 0.047

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Cpanamensis_prop, method="bray") ~ region*season,
        data = metadata.Cpan) # Region = P = 0.001, Season = P = 0.005, Region*Season = P = 0.013
#significant interaction between region and season (as seen with C. colonus and A. xanthopterus as well)


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Cpan, factors = sampledf.Cpan$region) #p = 0.001, R2 = 0.056

###by season
set.seed(1911)
pairwise.adonis(type.bray.Cpan, factors = sampledf.Cpan$season) #p = 0.004, R2 = 0.047
```

```{r PERMANOVA C. panamensis  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Cpanamensis, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Cpanamensis))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA C. panamensis  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Cpanamensis, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Cpanamensis))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# C. panamensis PCoAs
```{r PCoA (Bray) C. panamensis, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Cpan_pcoa <- ordinate(
  physeq = ROHR_06_Cpanamensis_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop,
  ordination = Cpan_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Cephalopholis panamensis Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop,
  ordination = Cpan_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "C panamensis Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# C. panamensis NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Cpan_nmds <- ordinate(
  physeq = ROHR_06_Cpanamensis_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop,
  ordination = Cpan_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Cephalopholis panamensis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Cpanamensis_prop,
  ordination = Cpan_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Cephalopholis panamensis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#C. panamensis stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Cpanamensis)$RRR_number <- as.character(sample_data(ROHR_06_Cpanamensis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Cpanamensis <- merge_treatments(ROHR_06_Cpanamensis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_pan_phylum <- ROHR_06_Cpanamensis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_pan_phylum["Phylum"][ROHR_06_pan_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_pan_phylum["Phylum"][ROHR_06_pan_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

unique(ROHR_06_pan_phylum$Phylum) #this is the largest number of different phyla in a single fish host species so far --> new_phyla4 <- c("Sumerlaeota")
c40

##code below does not run (copied from print of c40) - used to visualize currently-used colors and pick supplemental ones that don't overlap

  Acidobacteriota  Actinobacteriota      Bacteroidota  Bdellovibrionota  Campylobacterota 
    "dodgerblue2"         "#E31A1C"          "green4"         "#6A3D9A"         "#FF7F00" 
    Crenarchaeota     Cyanobacteria      Deinococcota  Desulfobacterota    Fibrobacterota 
          "black"           "gold1"        "skyblue2"         "#FB9A99"      "palegreen2" 
       Firmicutes    Fusobacteriota    Marinimicrobia       Myxococcota     Nanoarchaeota 
        "#CAB2D6"         "#FDBF6F"          "gray70"          "khaki2"          "maroon" 
     Nitrospirota   Patescibacteria   Planctomycetota    Proteobacteria      SAR324 clade 
        "orchid1"       "deeppink1"           "blue1"      "steelblue4"   "darkturquoise" 
    Spirochaetota  Thermoplasmatota Verrucomicrobiota     Calditrichota       Chloroflexi 
         "green1"         "yellow4"         "yellow3"      "slateblue4"           "wheat" 
     Dadabacteria Entotheonellaeota   Gemmatimonadota  Latescibacterota  Margulisbacteria 
  "lightseagreen"   "mediumorchid4"      "burlywood4"           "brown"         "orange1" 
            NB1-j           PAUC34f             WPS-2            MBNT15   Deferrisomatota 
         "yellow" "lightsteelblue1"        "honeydew"    "lightyellow3"            "navy" 
     Dependentiae            LCP-89           Sva0485 
         "gray40"        "hotpink4"         "sienna3"
     
new_phyla4 <- c("Sumerlaeota", "Deferribacterota")

c1bis <- c("ivory", "mediumblue")

names(c1bis) = new_phyla4

print(c1bis)

#merge old & new color assignments
c42 <- c(c40, c1bis)
print(c42) 

```

```{r prep data visualize most abundant taxa - C. panamensis}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_pan_phylum2 <- ROHR_06_Cpanamensis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_pan_phylum2["Phylum"][ROHR_06_pan_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_pan_phylum2["Phylum"][ROHR_06_pan_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_pan_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - C. panamensis}


# C. panamensis microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Cephalopholis panamensis"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_pan_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_pan_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_pan_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_pan_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("8693 C - NU", "8717C - NU", "8720 C - NU", "8723 C - NU", "8730 C - NU", "8745 C - NU", "8748 C - NU", "8774 C - NU", "7841 LP - NU", "7929 LP - NU", "7941 LP - NU", "8045 LP - NU", "8225 LP - NU", "8680 LP - NU", "8681 LP - NU", "16724 C - U", "16726 C - U", "16731 C - U", "16737 C - U", "16738 C - U", "16745 C - U", "16754 C - U", "16760 C - U", "16774 C - U", "15113 LP - U", "16700  LP - U", "20311 LP - U", "20363 LP - U", "20367 LP - U", "8033 LP - U", "8117 LP - U", "8173 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###C panamensis barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - C. panamensis}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Cpanamensis)[, "Phylum"])

#Convert to relative abundance
Cpan_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Cpanamensis, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Cpanamensis)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Cpan_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Cpan_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c42, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

# Microspathodon dorsalis

# M. dorsalis PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA M. dorsalis  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Mdor <- phyloseq::distance(ROHR_06_Mdorsalis_prop, method = "bray")
sampledf.Mdor <- data.frame(sample_data(ROHR_06_Mdorsalis_prop))

####beta dispersion####

###by region
beta.bray.region.Mdor <- betadisper(type.bray.Mdor, sampledf.Mdor$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Mdor, pairwise = TRUE, permutations = 10000) # P = 0.142  variance is homogeneous

plot(beta.bray.region.Mdor, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Mdor <- betadisper(type.bray.Mdor, sampledf.Mdor$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.391 variance is homogeneous
permutest(beta.bray.season.Mdor, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Mdor, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Mdor <- as(sample_data(ROHR_06_Mdorsalis_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Mdorsalis_prop, method="bray") ~ region,
        data = metadata.Mdor) # P = 0.001, R2 = 0.055

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Mdorsalis_prop, method="bray") ~ season,
        data = metadata.Mdor) # P = 0.003, R2 = 0.056

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Mdorsalis_prop, method="bray") ~ region*season,
        data = metadata.Mdor) # Region = P = 0.001, Season = P = 0.001, Region*Season = P = 0.084
#interaction is not significant, but close


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Mdor, factors = sampledf.Mdor$region) #p = 0.003, R2 = 0.055

###by season
set.seed(1911)
pairwise.adonis(type.bray.Mdor, factors = sampledf.Mdor$season) #p = 0.001, R2 = 0.056
```

```{r PERMANOVA M. dorsalis  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Mdorsalis, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Mdorsalis))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA M. dorsalis  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Mdorsalis, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Mdorsalis))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# M. dorsalis PCoAs
```{r PCoA (Bray) M. dorsalis, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Mdor_pcoa <- ordinate(
  physeq = ROHR_06_Mdorsalis_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop,
  ordination = Mdor_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Microspathodon dorsalis Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop,
  ordination = Mdor_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "M dorsalis Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# M. dorsalis NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Mdor_nmds <- ordinate(
  physeq = ROHR_06_Mdorsalis_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop,
  ordination = Mdor_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Microspathodon dorsalis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Mdorsalis_prop,
  ordination = Mdor_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Microspathodon dorsalis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#M. dorsalis stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Mdorsalis)$RRR_number <- as.character(sample_data(ROHR_06_Mdorsalis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Mdorsalis <- merge_treatments(ROHR_06_Mdorsalis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_dor_phylum <- ROHR_06_Mdorsalis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_dor_phylum["Phylum"][ROHR_06_dor_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_dor_phylum["Phylum"][ROHR_06_dor_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c42
unique(ROHR_06_dor_phylum$Phylum) #no new phyla

```

```{r prep data visualize most abundant taxa - M. dorsalis}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_dor_phylum2 <- ROHR_06_Mdorsalis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_dor_phylum2["Phylum"][ROHR_06_dor_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_dor_phylum2["Phylum"][ROHR_06_dor_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_dor_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - M. dorsalis}


# M. dorsalis microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Microspathodon dorsalis"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_dor_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_dor_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_dor_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_dor_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("8688 C - NU", "8694 C - NU", "8700 C - NU", "8710 C - NU", "8722 C - NU", "8746 C - NU", "8752 C - NU", "8766 C - NU", "8779 C - NU", "7805 LP - NU", "7889 LP - NU", "7961 LP - NU", "7965 LP - NU", "7969 LP - NU", "8244 LP - NU", "8246 LP - NU", "8248 LP - NU", "8249 LP - NU", "16717 C - U", "16725  C - U", "16728 C - U", "16729 C - U", "16735 C - U", "16766 C - U", "16768 C - U", "16782 C - U", "16785 C - U", "15122 LP - U", "15133 LP - U", "15134 LP - U", "15140 LP - U", "15152 LP - U", "16688 LP - U", "2032 LP - U", "8061 LP - U", "8157 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###M dorsalis barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - M. dorsalis}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Mdorsalis)[, "Phylum"])

#Convert to relative abundance
Mdor_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Mdorsalis, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Mdorsalis)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Mdor_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Mdor_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c42, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

# Johnrandallia nigrirostris

# J. nigrirostris PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA J. nigrirostris  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Jnig <- phyloseq::distance(ROHR_06_Jnigrirostris_prop, method = "bray")
sampledf.Jnig <- data.frame(sample_data(ROHR_06_Jnigrirostris_prop))

####beta dispersion####

###by region
beta.bray.region.Jnig <- betadisper(type.bray.Jnig, sampledf.Jnig$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Jnig, pairwise = TRUE, permutations = 10000) # P =  0.712 variance is homogeneous

plot(beta.bray.region.Jnig, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Jnig <- betadisper(type.bray.Jnig, sampledf.Jnig$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.158 variance is homogeneous
permutest(beta.bray.season.Jnig, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Jnig, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Jnig <- as(sample_data(ROHR_06_Jnigrirostris_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Jnigrirostris_prop, method="bray") ~ region,
        data = metadata.Jnig) # NSD --> P = 0.129, R2 = 0.035

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Jnigrirostris_prop, method="bray") ~ season,
        data = metadata.Jnig) # NSD P = 0.093, R2 = 0.036

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Jnigrirostris_prop, method="bray") ~ region*season,
        data = metadata.Jnig) # Region = P = 0.119, Season = P = 0.089, Region*Season = P = 0.128


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Jnig, factors = sampledf.Jnig$region) # NSD p = 0.125, R2 = 0.035

###by season
set.seed(1911)
pairwise.adonis(type.bray.Jnig, factors = sampledf.Jnig$season) # NSD p = 0.1, R2 = 0.036
```

```{r PERMANOVA J. nigrirostris  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Jnigrirostris, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Jnigrirostris))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA J. nigrirostris  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Jnigrirostris, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Jnigrirostris))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# J. nigrirostris PCoAs
```{r PCoA (Bray) J. nigrirostris, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Jnig_pcoa <- ordinate(
  physeq = ROHR_06_Jnigrirostris_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop,
  ordination = Jnig_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Johnrandallia nigrirostris Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop,
  ordination = Jnig_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "J nigrirostris Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# J. nigrirostris NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Jnig_nmds <- ordinate(
  physeq = ROHR_06_Jnigrirostris_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop,
  ordination = Jnig_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Johnrandallia nigrirostris Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Jnigrirostris_prop,
  ordination = Jnig_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Johnrandallia nigrirostris Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#J. nigrirostris stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Jnigrirostris)$RRR_number <- as.character(sample_data(ROHR_06_Jnigrirostris)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Jnigrirostris <- merge_treatments(ROHR_06_Jnigrirostris, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_nig_phylum <- ROHR_06_Jnigrirostris %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_nig_phylum["Phylum"][ROHR_06_nig_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_nig_phylum["Phylum"][ROHR_06_nig_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c42
unique(ROHR_06_nig_phylum$Phylum) #no new phyla

```

```{r prep data visualize most abundant taxa - J. nigrirostris}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_nig_phylum2 <- ROHR_06_Jnigrirostris %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_nig_phylum2["Phylum"][ROHR_06_nig_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_nig_phylum2["Phylum"][ROHR_06_nig_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_nig_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - J. nigrirostris}


# J. nigrirostris microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Johnrandallia nigrirostris"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_nig_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_nig_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_nig_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_nig_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("8691 C - NU", "8701 C - NU", "8716 C - NU", "8718 C - NU", "8747 C - NU", "8751 C - NU", "8760 C - NU", "8764 C - NU", "8768 C - NU", "7909 LP - NU", "7977 LP - NU", "7981 LP - NU", "8219 LP - NU", "8221 LP - NU", "8227 LP - NU", "8233 LP - NU", "8236 LP - NU", "8237 LP - NU", "8685 LP - NU", "16733 C - U", "16748 C - U", "16752 C - U", "16757 C - U", "16763 C - U", "16767 C - U", "16770 C - U", "16778 C - U", "15117 LP - U", "15127 LP - U", "15148 LP - U", "8089 LP - U", "8101 LP - U", "8105 LP - U", "8125 LP - U", "8197 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###J nigrirostris barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - J. nigrirostris}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Jnigrirostris)[, "Phylum"])

#Convert to relative abundance
Jnig_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Jnigrirostris, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Jnigrirostris)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Jnig_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Jnig_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c42, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

# Epinephelus labriformis

# E. labriformis PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA E. labriformis  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Elab <- phyloseq::distance(ROHR_06_Elabriformis_prop, method = "bray")
sampledf.Elab <- data.frame(sample_data(ROHR_06_Elabriformis_prop))

####beta dispersion####

###by region
beta.bray.region.Elab <- betadisper(type.bray.Elab, sampledf.Elab$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Elab, pairwise = TRUE, permutations = 10000) # P = 0.901 variance is homogeneous

plot(beta.bray.region.Elab, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Elab <- betadisper(type.bray.Elab, sampledf.Elab$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.975 variance is homogeneous
permutest(beta.bray.season.Elab, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Elab, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Elab <- as(sample_data(ROHR_06_Elabriformis_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Elabriformis_prop, method="bray") ~ region,
        data = metadata.Elab) # P = 0.014, R2 = 0.045

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Elabriformis_prop, method="bray") ~ season,
        data = metadata.Elab) # P = 0.031, R2 = 0.044

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Elabriformis_prop, method="bray") ~ region*season,
        data = metadata.Elab) # Region = P = 0.010, Season = P = 0.022, Region*Season = P = 0.157


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Elab, factors = sampledf.Elab$region) #p = 0.014, R2 = 0.045

###by season
set.seed(1911)
pairwise.adonis(type.bray.Elab, factors = sampledf.Elab$season) #p = 0.02, R2 = 0.044
```

```{r PERMANOVA E. labriformis  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Elabriformis, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Elabriformis))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA E. labriformis  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Elabriformis, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Elabriformis))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# E. labriformis PCoAs
```{r PCoA (Bray) E. labriformis, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Elab_pcoa <- ordinate(
  physeq = ROHR_06_Elabriformis_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Elabriformis_prop,
  ordination = Elab_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Epinephelus labriformis Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Elabriformis_prop,
  ordination = Elab_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "E labriformis Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# E. labriformis NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Elab_nmds <- ordinate(
  physeq = ROHR_06_Elabriformis_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Elabriformis_prop,
  ordination = Elab_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Epinephelus labriformis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Elabriformis_prop,
  ordination = Elab_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Epinephelus labriformis Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#E. labriformis stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Elabriformis)$RRR_number <- as.character(sample_data(ROHR_06_Elabriformis)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Elabriformis <- merge_treatments(ROHR_06_Elabriformis, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_lab_phylum <- ROHR_06_Elabriformis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_lab_phylum["Phylum"][ROHR_06_lab_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lab_phylum["Phylum"][ROHR_06_lab_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c42
unique(ROHR_06_lab_phylum$Phylum) #no new phyla

```

```{r prep data visualize most abundant taxa - E. labriformis}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_lab_phylum2 <- ROHR_06_Elabriformis %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_lab_phylum2["Phylum"][ROHR_06_lab_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lab_phylum2["Phylum"][ROHR_06_lab_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_lab_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - E. labriformis}


# E. labriformis microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Epinephelus labriformis"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_lab_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_lab_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_lab_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_lab_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("8699 C - NU", "8706  C - NU", "8708 C - NU", "8726 C - NU", "8728 C - NU", "8750 C - NU", "8756 C - NU", "8767 C - NU", "8776 C - NU", "7813 LP - NU", "7821 LP - NU", "7893 LP - NU", "7957 LP - NU", "8001 LP - NU", "8234 LP - NU", "8677 LP - NU", "16712 C - U", "16741 C - U", "16771 C - U", "16786 C - U", "16796 C - U", "16797 C - U", "16806 C - U", "16808 C - U", "16810 C - U", "15126 LP - U", "15139  LP - U", "15143 LP - U", "15147 LP - U", "16686 LP - U", "8093 LP - U", "8133 LP - U", "8185 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###E labriformis barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - E. labriformis}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Elabriformis)[, "Phylum"])

#Convert to relative abundance
Elab_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Elabriformis, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Elabriformis)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Elab_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Elab_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c42, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```

#Prionurus laticlavius

# P. laticlavius PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA P. laticlavius  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Plat <- phyloseq::distance(ROHR_06_Platiclavius_prop, method = "bray")
sampledf.Plat <- data.frame(sample_data(ROHR_06_Platiclavius_prop))

####beta dispersion####

###by region
beta.bray.region.Plat <- betadisper(type.bray.Plat, sampledf.Plat$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Plat, pairwise = TRUE, permutations = 10000) # P = 0.95 variance is homogeneous

plot(beta.bray.region.Plat, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Plat <- betadisper(type.bray.Plat, sampledf.Plat$season,
                         type = "centroid", bias.adjust = TRUE) # P = 0.76 variance is homogeneous
permutest(beta.bray.season.Plat, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Plat, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Plat <- as(sample_data(ROHR_06_Platiclavius_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_06_Platiclavius_prop, method="bray") ~ region,
        data = metadata.Plat) # P = 0.001, R2 = 0.067

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_06_Platiclavius_prop, method="bray") ~ season,
        data = metadata.Plat) # P = 0.008, R2 = 0.059

####testing for interactions between region & season####
adonis2(distance(ROHR_06_Platiclavius_prop, method="bray") ~ region*season,
        data = metadata.Plat) # Region = P = 0.001, Season = P = 0.002, Region*Season = P = 0.001


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Plat, factors = sampledf.Plat$region) #p = 0.001, R2 = 0.067

###by season
set.seed(1911)
pairwise.adonis(type.bray.Plat, factors = sampledf.Plat$season) #p = 0.003, R2 = 0.059
```

```{r PERMANOVA P. laticlavius  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_06_Platiclavius, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_06_Platiclavius))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA P. laticlavius  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_06_Platiclavius, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_06_Platiclavius))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```

# P. laticlavius PCoAs
```{r PCoA (Bray) P. laticlavius, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Plat_pcoa <- ordinate(
  physeq = ROHR_06_Platiclavius_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_06_Platiclavius_prop,
  ordination = Plat_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Prionurus laticlavius Skin Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_06_Platiclavius_prop,
  ordination = Plat_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "Water samples Skin Microbiome PCoA Host Species 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# P. laticlavius NMDS
```{r NMDS A concolor, echo=FALSE}
set.seed(1)

# Ordinate
Plat_nmds <- ordinate(
  physeq = ROHR_06_Platiclavius_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_06_Platiclavius_prop,
  ordination = Plat_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Prionurus laticlavius Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_06_Platiclavius_prop,
  ordination = Plat_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Prionurus laticlavius Skin Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```

#P. laticlavius stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_06_Platiclavius)$RRR_number <- as.character(sample_data(ROHR_06_Platiclavius)$RRR)

#create new category with RRR & region/season merged
ROHR_06_Platiclavius <- merge_treatments(ROHR_06_Platiclavius, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_06_lat_phylum <- ROHR_06_Platiclavius %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_06_lat_phylum["Phylum"][ROHR_06_lat_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lat_phylum["Phylum"][ROHR_06_lat_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c42
unique(ROHR_06_lat_phylum$Phylum) #no new

```

```{r prep data visualize most abundant taxa - P. laticlavius}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_06_lat_phylum2 <- ROHR_06_Platiclavius %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_06_lat_phylum2["Phylum"][ROHR_06_lat_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_06_lat_phylum2["Phylum"][ROHR_06_lat_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_06_lat_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - P. laticlavius}


# P. laticlavius microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of ", italic ("Prionurus laticlavius"), " Skin Microbial Communities"))

# Basic plot

ggplot(ROHR_06_lat_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_06_lat_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_06_lat_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_06_lat_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c42, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("15085 C - NU", "15087 C - NU", "8727 C - NU", "8731 C - NU", "8787 C - NU", "8788 C - NU", "8792 C - NU", "8794 C - NU", "7993 LP - NU", "8009 LP - NU", "8013 LP - NU", "8017 LP - NU", "8021 LP - NU", "8247 LP - NU", "8675 LP - NU", "8676 LP - NU", "8682 LP - NU", "16707 C - U", "16708 C - U", "16718 C - U", "16720 C - U", "16756 C - U", "16759 C - U", "16761 C - U", "16762 C - U", "16764 C - U", "15156 LP - U", "15160 LP - U", "16773 LP - U", "20343 LP - U", "20347 LP - U", "20371 LP - U", "8169 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###P laticlavius barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - P. laticlavius}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_06_Platiclavius)[, "Phylum"])

#Convert to relative abundance
Plat_rel_abund = phyloseq::transform_sample_counts(ROHR_06_Platiclavius, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_06_Platiclavius)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Plat_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Plat_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c42, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```
## Water microbiome samples for comparison to fish skin
Loading our phyloseq object - filtered & trimmed (not rarefied) with phylogenetic tree - from "ROHR_08_Water_stats_alpha.Rmd"
```{r read phyloseq object - water, echo=FALSE}
ROHR_08_obj_phylo_root <- readRDS("/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_08_trimmed/ROHR_08_with_phylogenetic_tree_rooted.rds")
```
```{r counts to relative abundances - water}
####### Transform counts to relative abundance per sample
ROHR_08_obj_clean_prop = transform_sample_counts(ROHR_08_obj_phylo_root, function(x) x/sum(x))
```

# Water samples PERMANOVAs (Bray-Curtis, Jaccard, UniFrac)
```{r PERMANOVA Water samples  - Bray-Curtis Distances, echo = FALSE}
####dissimilarity matrix####
set.seed(1911)
type.bray.Water <- phyloseq::distance(ROHR_08_obj_clean_prop, method = "bray")
sampledf.Water <- data.frame(sample_data(ROHR_08_obj_clean_prop))

####beta dispersion####

###by region
beta.bray.region.Water <- betadisper(type.bray.Water, sampledf.Water$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.bray.region.Water, pairwise = TRUE, permutations = 10000) # P = 0.287 variance is homogeneous

plot(beta.bray.region.Water, hull=FALSE, ellipse=TRUE)

###by season
beta.bray.season.Water <- betadisper(type.bray.Water, sampledf.Water$season,
                         type = "centroid", bias.adjust = TRUE) # P < 0.001 variance is NOT homogeneous (which makes sense given that we are looking across upwelling & non-upwelling seasons)
permutest(beta.bray.season.Water, pairwise = TRUE, permutations = 10000)

plot(beta.bray.season.Water, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

####running the PERMANOVA
metadata.Water <- as(sample_data(ROHR_08_obj_clean_prop), "data.frame")

####by region####
            
#region (collection region - Coiba or Las Perlas)
adonis2(distance(ROHR_08_obj_clean_prop, method="bray") ~ region,
        data = metadata.Water) # P = 0.006, R2 = 0.19

####by season####
#season = upwelling/non-upwelling
adonis2(distance(ROHR_08_obj_clean_prop, method="bray") ~ season,
        data = metadata.Water) # P = 0.001, R2 = 0.313

####testing for interactions between region & season####
adonis2(distance(ROHR_08_obj_clean_prop, method="bray") ~ region*season,
        data = metadata.Water) # Region = P = .001, Season = P = 0.00, Region*Season = P = 0.001
#singificant interaction between region and season (most significant seen so far, much more than fish skin microbiome data)


####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.bray.Water, factors = sampledf.Water$region) #p = 0.005, R2 = 0.19

###by season
set.seed(1911)
pairwise.adonis(type.bray.Water, factors = sampledf.Water$season) #p = 0.001, R2 = 0.31
```

```{r PERMANOVA Water samples  - Jaccard Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1911)
type.jaccard <- phyloseq::distance(ROHR_08_obj_phylo_root, method = "jaccard", binary = T)
sampledf <- data.frame(sample_data(ROHR_08_obj_phylo_root))

####beta dispersion####

###by region
beta.jaccard.region <- betadisper(type.jaccard, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.jaccard.region, pairwise = TRUE, permutations = 10000) # P =  is homogeneous across regions

plot(beta.jaccard.region, hull=FALSE, ellipse=TRUE)

###by season
beta.jaccard.season <- betadisper(type.jaccard, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P =  is homogeneous across seasons
permutest(beta.jaccard.season, pairwise = TRUE, permutations = 10000)

####PERMANOVAs####

###by region
adonis.jaccard.region <- adonis2(type.jaccard ~ region, data = sampledf,
                          permutations = 10000)
adonis.jaccard.region

#adonis2(formula = type.jaccard ~ region, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)    

###by season
adonis.jaccard.season <- adonis2(type.jaccard ~ season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.season

#adonis2(formula = type.jaccard ~ season, data = sampledf, permutations = 10000)
#          Df SumOfSqs      R2      F     Pr(>F)  

###by region*season
adonis.jaccard.regseason <- adonis2(type.jaccard ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.jaccard.regseason

####pairwise comparisons#### 

###by region
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$region) 

###by season
set.seed(1911)
pairwise.adonis(type.jaccard, factors = sampledf$season) 
```

```{r PERMANOVA Water samples  - UniFrac Distances, echo = FALSE}
####dissimilarity matrix####

set.seed(1900)
type.unifrac <- phyloseq::distance(ROHR_08_phylo_root, method = "unifrac", weighted = F)
sampledf <- data.frame(sample_data(ROHR_08_phylo_root))

####beta dispersion####
###by region
beta.unifrac.region <- betadisper(type.unifrac, sampledf$region,
                         type = "centroid", bias.adjust = TRUE)
permutest(beta.unifrac.region, pairwise = TRUE, permutations = 10000) # P = 

plot(beta.unifrac.region, hull=FALSE, ellipse=TRUE)

###by season
beta.unifrac.season <- betadisper(type.unifrac, sampledf$season,
                         type = "centroid", bias.adjust = TRUE) # P = 
permutest(beta.unifrac.season, pairwise = TRUE, permutations = 10000)

plot(beta.unifrac.season, hull=FALSE, ellipse=TRUE)

####PERMANOVAs####

###by region
adonis.unifrac.region <- adonis2(type.unifrac ~ region, data = sampledf,
                          permutations = 10000)
adonis.unifrac.region

###by season
adonis.unifrac.season <- adonis2(type.unifrac ~ season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.season

###by region*season
adonis.unifrac.regseason <- adonis2(type.unifrac ~ region*season, data = sampledf,
                          permutations = 10000)
adonis.unifrac.regseason

####pairwise comparisons#### 
###by region
set.seed(1924)
pairwise.adonis(type.unifrac, factors = sampledf$region) # p-adj = 

###by season
set.seed(1925)
pairwise.adonis(type.unifrac, factors = sampledf$season) # p-adj = 
```


# Water samples PCoAs
```{r PCoA (Bray) Water samples, echo = FALSE}
theme_set(theme_bw())

# Ordinate
Water_pcoa <- ordinate(
  physeq = ROHR_08_obj_clean_prop, 
  method = "PCoA", 
  distance = "bray"
)

# Plot - color = season, shape = region (coiba or las perlas)
plot_ordination(
  physeq = ROHR_08_obj_clean_prop,
  ordination = Water_pcoa,
  color = "season",
  shape = "region",
  title = "PCoA of Water Microbiome Communities"
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

# Same plot as above, except w/o title
plot_ordination(
  physeq = ROHR_08_obj_clean_prop,
  ordination = Water_pcoa,
  color = "season",
  shape = "region",
) + 
  scale_color_brewer(palette = "Paired" ) +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1)

```
```{r code for exporting plot above as a pdf}
ggsave(filename = "TEP Water Microbiome PCoA 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```

# Water samples NMDS
```{r NMDS Water, echo=FALSE}
set.seed(1)

# Ordinate
Water_nmds <- ordinate(
  physeq = ROHR_08_obj_clean_prop, 
  method = "NMDS", 
  distance = "bray"
)

plot_ordination(
  physeq = ROHR_08_obj_clean_prop,
  ordination = Water_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Water Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 3) +
  geom_point(colour = "grey90", size = 1) 

#adding ellipses (student's t-distribution) 
##alternative ellipses using normal distribution = stat_ellipse(type = "norm", linetype = 2)

plot_ordination(
  physeq = ROHR_08_obj_clean_prop,
  ordination = Water_nmds,
  color = "season",
  shape = "region",
  title = "NMDS of Water Microbial Communities") + 
  scale_color_brewer(palette = "Paired") +
  geom_point(aes(color = season), alpha = 0.7, size = 4) +
  geom_point(colour = "grey90", size = 1) +
  stat_ellipse(type = "t")
    
```
```{r code for exporting plot above as a pdf}
ggsave(filename = "Water Microbiome NMDS 21-22", path = '/Volumes/LLL_one/McGill/Lab_Docs/Fish_Data/Fish_2022/2021-2022_TEP_Fish/16S_Summer_2022/ROHR_Figures_Draft', width = 14, height = 10, device='pdf', dpi=700)
```
#Water samples stacked barplots

```{r prep data visualize most abundant taxa - setup colors to match full dataset analyses}

#change RRR numbers (sample ID) from integer to character
sample_data(ROHR_08_obj_phylo_root)$RRR_number <- as.character(sample_data(ROHR_08_obj_phylo_root)$RRR)

#create new category with RRR & region/season merged
ROHR_08_obj_phylo_root <- merge_treatments(ROHR_08_obj_phylo_root, c("RRR_number", "region_season"))

#extend color palette from code above to encompass full set of phyla represented across samples (unfiltered)
ROHR_08_Water_phylum <- ROHR_08_obj_phylo_root %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  arrange(region_season)                              #arrange samples by region & season

#rename 2 phyla (v. long names currently) - to match terms used for full dataset
ROHR_08_Water_phylum["Phylum"][ROHR_08_Water_phylum["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_08_Water_phylum["Phylum"][ROHR_08_Water_phylum["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

c42
unique(ROHR_08_Water_phylum$Phylum) # c("AncK6", "Nitrospinota", "Aenigmarchaeota", "Poribacteria")
# Latescibacteria, MBNT15, Deferrisomatota, Deferribacterota, LCP-89, and Sumerlaeota are all phyla that were found in at least one fish specimen but not in our water samples
newwater_phyla <- c("AncK6", "Nitrospinota", "Aenigmarchaeota", "Poribacteria")
c4 <- c("darkorange4", "aquamarine", "darkgoldenrod", "chartreuse3")

names(c4) = newwater_phyla

print(c4)

#merge old & new color assignments
c44 <- c(c42, c4)
print(c44) 
```

```{r prep data visualize most abundant taxa - Water samples}

# melt to long format (for ggploting) 
# prune out phyla below 2% in each sample

ROHR_08_Water_phylum2 <- ROHR_08_obj_phylo_root %>%
  tax_glom(taxrank = "Phylum") %>%                     # agglomerate at phylum level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% # Transform to rel. abundance
  psmelt() %>%                                         # Melt to long format
  filter(Abundance > 0.02) %>%                         # Filter out low abundance taxa (remove those that contribute less                                                          than 2% of relative abundance of each sample)
  arrange(region_season)                                      # Sort data frame by region & season 

#rename 2 phyla (v. long names currently)
ROHR_08_Water_phylum2["Phylum"][ROHR_08_Water_phylum2["Phylum"] == "Marinimicrobia (SAR406 clade)"] <- "Marinimicrobia"
ROHR_08_Water_phylum2["Phylum"][ROHR_08_Water_phylum2["Phylum"] == "SAR324 clade(Marine group B)"] <- "SAR324 clade"

# check phyla in this subsetted dataset to ensure a color is assigned for each one
unique(ROHR_08_Water_phylum2$Phylum)

```

```{r stacked barplot to visualize most abundant taxa - broken into regions - Water samples}


# Water samples microbial communities broken down by region & season

#make species name in italics for title
plot_title <- expression(paste("Phylum Composition of TEP Water Microbial Communities"))

# Basic plot

ggplot(ROHR_08_Water_phylum2, aes(x = region_season, y = Abundance, fill = Phylum)) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44) + #using colors from full dataset run above, so that colors match across plots
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season")+
  ggtitle(plot_title)

#Optimized plot
ggplot(ROHR_08_Water_phylum2, aes(x = region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  scale_x_discrete(
    breaks = c("Coiba non-upwelling", "Las Perlas non-upwelling", "Coiba upwelling", "Las Perlas upwelling"),
    labels = c("Coiba Non-upwelling", "Las Perlas Non-upwelling", "Coiba Upwelling", "Las Perlas Upwelling"), 
    drop = FALSE
  ) +
  #
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance (Phyla > 2%) \n") +
  xlab("Sampling Gulf and Season") +
  theme(axis.text.x = element_text(size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label (sampling gulfs & seasons)
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region
ggplot(ROHR_08_Water_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)


#########need to find way to fully sort by region & season and mark where each region/season begins/ends####

#Optimized plot - not filtered by abundance (all phyla included) & split by sample rather than region w/ labels for RRR # and season/region
ggplot(ROHR_08_Water_phylum, aes(x = RRR_number_region_season, y = Abundance, fill = fct_reorder(Phylum, Abundance))) + 
  geom_bar(stat = "identity", position = "fill") +
  scale_fill_manual(values = c44, name="Microbial Phyla")+
  scale_x_discrete(
    labels = c("6005 C - NU", "6010 C - NU", "6015 C - NU", "6020 C - NU", "6025 LP - NU", "6030 LP - NU", "6035 LP - NU", "6040 LP - NU", "6045 LP - NU", "6050 LP - NU", "21585 C - U", "21589 C - U", "21593 C - U", "21597 C - U", "21601 C - U", "6055 LP - U", "6060 LP - U", "6065 LP - U", "6070 LP - U")
  ) +
  guides(fill = guide_legend(reverse = TRUE, keywidth = 1, keyheight = 1, ncol = 1)) +
  ylab("Relative Abundance") +
  xlab("Sample ID") +
  theme(axis.text.x = element_text(angle = 90, size = 14), #change size of location names (x-axis)
        axis.title.x = element_text(size = 16), #change size of x-axis label 
        axis.text.y = element_text(size = 10), #change size of y-axis #s
        axis.title.y = element_text(size = 16), #change size of y-axis title
        legend.title = element_text(size = 16), #change size of legend title (microbial phyla)
        legend.text = element_text(size = 10), #change size of legend text (microbial taxa)
        strip.text.y = element_text(size = 13), #change size of each box's label (e.g., Coiba upwelling)
        title = element_markdown(size = 16, face = "bold"))+
  ggtitle(plot_title)

```
###Water samples barplots - sample-by-sample
--> following helpful workflow by Nicholas Ollberding - "Introduction to the Statistical Analysis of Microbiome Data in R"
--> also worth reading his introduction as he links many other useful resources for further reference
https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/

```{r stacked barplots of microbial phyla - sample-by-sample - Water samples}

###need to find a way to reorder these by relative abundance of phyla

#Get count of phyla
table(phyloseq::tax_table(ROHR_08_obj_phylo_root)[, "Phylum"])

#Convert to relative abundance
Water_rel_abund = phyloseq::transform_sample_counts(ROHR_08_obj_phylo_root, function(x){x / sum(x)})

phyloseq::otu_table(ROHR_08_obj_phylo_root)[1:5, 1:5] #visualize OTU table pre-transformation
phyloseq::otu_table(Water_rel_abund)[1:5, 1:5] #visualize OTU table after converting to relative abundance



#Plot - faceted by season & region
phyloseq::plot_bar(Water_rel_abund, fill = "Phylum") +
  geom_bar(aes(fill = fct_reorder(Phylum, Abundance)), stat = "identity", position = "stack") +
  scale_fill_manual(values = c44, name="Microbial Phyla") +
  labs(x = "", y = "Relative Abundance\n") +
  facet_wrap(~ region_season, scales = "free") + #facet by season & region
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank())

```
```{r code to check read counts per sample, echo = FALSE}
# extract individual read counts and sample IDs
sample_counts <- sample_sums(ROHR_06_obj_phylo_root)
sample_ids <- sample_names(ROHR_06_obj_phylo_root)

# create a data frame with sample IDs and individual read counts
sample_table <- data.frame(sample_id = sample_ids, read_count = sample_counts)

# display the sample table
sample_table
```
